{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    }
   ],
   "source": [
    "%pylab inline\n",
    "import scipy.stats as stats\n",
    "import pandas as pd\n",
    "import gseapy\n",
    "rcParams['font.size']=12\n",
    "rcParams['pdf.fonttype']=42\n",
    "import networkx as nx\n",
    "import seaborn as sb\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>pDNA_batch</th>\n",
       "      <th>passes_QC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>replicate_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>CCLFUPGI0005T-311cas9-RepB-p6_batch4</td>\n",
       "      <td>ACH-002446</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>SUPT1-311Cas9_RepB_p6_batch3</td>\n",
       "      <td>ACH-000953</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>OVCAR-5-311Cas9_RepB_p6_batch3</td>\n",
       "      <td>ACH-001151</td>\n",
       "      <td>3</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       DepMap_ID  pDNA_batch  passes_QC\n",
       "replicate_ID                                                           \n",
       "CCLFUPGI0005T-311cas9-RepB-p6_batch4  ACH-002446           4       True\n",
       "SUPT1-311Cas9_RepB_p6_batch3          ACH-000953           3       True\n",
       "OVCAR-5-311Cas9_RepB_p6_batch3        ACH-001151           3       True"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read replicate info\n",
    "\n",
    "replicatemap = pd.read_csv('Achilles_replicate_map.csv',index_col=0,header=0)\n",
    "replicatemap['DepMap_ID'].fillna('pDNA',inplace=True)\n",
    "replicatemap.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['21NT-311Cas9-RepAB-p6_batch4' 'C125PM-311cas9_RepA_p6_batch3'\n",
      " 'C125PM-311cas9_RepB_p6_batch3' 'CMS-311cas9-RepAB-p6_batch4'\n",
      " 'CORL105-311Cas9-RepA-p6_batch4' 'Colo699-311cas9-RepA-p6_batch4'\n",
      " 'Colo699-311cas9-RepB-p6_batch4' 'ESO51-311cas9_RepA_p6_batch3'\n",
      " 'ESO51-311cas9_RepB_p6_batch3' 'FARAGE-311cas9-RepAB-p6_batch4'\n",
      " 'G415-311cas9-RepA-p6_batch4' 'G415-311cas9-RepB-p6_batch4'\n",
      " 'HDLM2-311Cas9_RepA_p4_batch3' 'HEC151-311Cas9_RepA_p3_batch3'\n",
      " 'HEC151-311Cas9_RepB_p3_batch3' 'HSPSS-311cas9-RepA-p6_batch4'\n",
      " 'HSPSS-311cas9-RepB-p6_batch4' 'HUTU-80-311cas9_RepA_p6_batch3'\n",
      " 'HUTU-80-311cas9_RepB_p6_batch3' 'IA-LM-311Cas9_RepA_p4_batch3'\n",
      " 'IA-LM-311Cas9_RepB_p4_batch3' 'ICC6-311cas9-RepA-p6_batch4'\n",
      " 'ICC6-311cas9-RepB-p6_batch4' 'J82-311Cas9_RepA_p5_batch3'\n",
      " 'J82-311Cas9_RepB_p5_batch3' 'JeKo1-311Cas9_RepA_p6_batch3'\n",
      " 'JeKo1-311Cas9_RepB_p6_batch3' 'M040416-311cas9-RepA-p6_batch4'\n",
      " 'M040416-311cas9-RepB-p6_batch4' 'M07E-311cas9-RepA-p6_batch4'\n",
      " 'M07E-311cas9-RepB-p6_batch4' 'MEG01-311cas9_RepA_p6_batch3'\n",
      " 'MEG01-311cas9_RepB_p6_batch3' 'MERO25-311cas9-RepA-p6_batch4'\n",
      " 'MERO25-311cas9-RepB-p6_batch4' 'MFM223-311cas9-RepA-p6_batch4'\n",
      " 'MFM223-311cas9-RepB-p6_batch4' 'MOLM14-311cas9-RepA-p6_batch4'\n",
      " 'MOLM14-311cas9-RepB-p6_batch4' 'NALM1-311cas9_RepA_p6_batch3'\n",
      " 'NALM1-311cas9_RepB_p6_batch3' 'NALM16-311CAS9-RepA-p6_batch4'\n",
      " 'NALM16-311CAS9-RepB-p6_batch4' 'NCC-StC-K14-311Cas9_RepA_p3_batch3'\n",
      " 'NCC-StC-K14-311Cas9_RepB_p3_batch3' 'NCI-H2009-311CAS9-RepA-p6_batch4'\n",
      " 'NCI-H2009-311CAS9-RepB-p6_batch4' 'NCIH1373-311cas9-RepA-p6_batch4'\n",
      " 'NCIH1373-311cas9-RepB-p6_batch4' 'NCIH1395-311Cas9_RepA_p5_batch3'\n",
      " 'NCIH1395-311Cas9_RepB_p5_batch3' 'NCIH2004RT-311cas9-RepA-p6_batch4'\n",
      " 'NCIH2004RT-311cas9-RepB-p6_batch4' 'NCIH522-311Cas9_RepA_p6_batch4'\n",
      " 'OCIAML4-311cas9-RepA-p6_batch4' 'OCIAML4-311cas9-RepB-p6_batch4'\n",
      " 'OCILY19-311cas9-RepA-p6_batch4' 'OCILY19-311cas9-RepB-p6_batch4'\n",
      " 'PACADD135-311Cas9_RepA_p6_batch3' 'PACADD135-311Cas9_RepB_p6_batch3'\n",
      " 'PACADD159-311cas9_RepA_p5_batch3' 'PACADD159-311cas9_RepB_p5_batch3'\n",
      " 'PC3-311cas9_RepA_p6_batch3' 'PC3-311cas9_RepB_p6_batch3'\n",
      " 'RCK8-311cas9-RepA-p6_batch4' 'RCK8-311cas9-RepB-p6_batch4'\n",
      " 'RCM-1-311cas9_RepA_p6_batch3' 'RCM-1-311cas9_RepB_p6_batch3'\n",
      " 'RERF-GC-1B-311Cas9_RepA_p6_batch3' 'RERF-GC-1B-311Cas9_RepB_p6_batch3'\n",
      " 'RPE1SS111-311Cas9-RepA-p6_batch4' 'RPE1SS111-311Cas9-RepB-p6_batch4'\n",
      " 'RPE1SS48-311CAS9-RepA-p6_batch4' 'RPE1SS48-311CAS9-RepB-p6_batch4'\n",
      " 'RPE1SS51_1RepB-p6_batch4' 'RPE1SS6-311Cas9-RepA-p4_batch4'\n",
      " 'RPE1SS6-311Cas9-RepBp4_batch4' 'RPE1SS77-311Cas9-RepA-p5_batch4'\n",
      " 'RPE1SS77-311Cas9-RepB-p5_batch4' 'SF126-311Cas9_RepA_p6_batch3'\n",
      " 'SF126-311Cas9_RepB_p6_batch3' 'SNU-1040-311Cas9_RepA_p4_batch3'\n",
      " 'SNU-1040-311Cas9_RepB_p4_batch3' 'SNU-245-311Cas9_RepA_p6_batch3'\n",
      " 'SNU-245-311Cas9_RepB_p6_batch3' 'SNU175-311Cas9_RepA_p5_batch3'\n",
      " 'SNU175-311Cas9_RepB_p5_batch3' 'SNU407-311Cas9_RepB_p6_batch4'\n",
      " 'SUM185PE-311cas9_RepA_p6_batch3' 'SUM185PE-311cas9_RepB_p6_batch3'\n",
      " 'TE-15-311cas9_RepA_p6_batch3' 'TE-15-311cas9_RepB_p6_batch3'\n",
      " 'TGBC11TKB-311cas9_RepA_p4_batch3' 'TGBC11TKB-311cas9_RepB_p4_batch3'\n",
      " 'UACC812-311Cas9_RepA_p4_batch3' 'UACC812-311Cas9_RepB_p4_batch3'\n",
      " 'WM3211-311cas9-RepA-p6_batch4' 'WM3211-311cas9-RepB-p6_batch4'\n",
      " 'pDNA 10 pg/well_batch4' 'pDNA 100 pg/well_batch4'\n",
      " 'pDNA 100pg/well_batch4' 'pDNA 50 pg/well_batch4']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DepMap_ID</th>\n",
       "      <th>pDNA_batch</th>\n",
       "      <th>passes_QC</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>replicate_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>21NT-311Cas9-RepAB-p6_batch4</td>\n",
       "      <td>ACH-002399</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C125PM-311cas9_RepA_p6_batch3</td>\n",
       "      <td>ACH-001456</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>C125PM-311cas9_RepB_p6_batch3</td>\n",
       "      <td>ACH-001456</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CMS-311cas9-RepAB-p6_batch4</td>\n",
       "      <td>ACH-002709</td>\n",
       "      <td>4</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>CORL105-311Cas9-RepA-p6_batch4</td>\n",
       "      <td>ACH-000161</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>WM3211-311cas9-RepB-p6_batch4</td>\n",
       "      <td>ACH-002508</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pDNA 10 pg/well_batch4</td>\n",
       "      <td>pDNA</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pDNA 100 pg/well_batch4</td>\n",
       "      <td>pDNA</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pDNA 100pg/well_batch4</td>\n",
       "      <td>pDNA</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>pDNA 50 pg/well_batch4</td>\n",
       "      <td>pDNA</td>\n",
       "      <td>4</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>102 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 DepMap_ID  pDNA_batch  passes_QC\n",
       "replicate_ID                                                     \n",
       "21NT-311Cas9-RepAB-p6_batch4    ACH-002399           4       True\n",
       "C125PM-311cas9_RepA_p6_batch3   ACH-001456           3      False\n",
       "C125PM-311cas9_RepB_p6_batch3   ACH-001456           3      False\n",
       "CMS-311cas9-RepAB-p6_batch4     ACH-002709           4      False\n",
       "CORL105-311Cas9-RepA-p6_batch4  ACH-000161           4       True\n",
       "...                                    ...         ...        ...\n",
       "WM3211-311cas9-RepB-p6_batch4   ACH-002508           4       True\n",
       "pDNA 10 pg/well_batch4                pDNA           4       True\n",
       "pDNA 100 pg/well_batch4               pDNA           4       True\n",
       "pDNA 100pg/well_batch4                pDNA           4       True\n",
       "pDNA 50 pg/well_batch4                pDNA           4       True\n",
       "\n",
       "[102 rows x 3 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check increment in comparison with previous version\n",
    "replicatemap_previous = pd.read_csv('../avana_2020Q3/Achilles_replicate_map.csv',index_col=0,header=0)\n",
    "replicatemap_previous['DepMap_ID'].fillna('pDNA',inplace=True)\n",
    "\n",
    "print(setdiff1d(replicatemap.index,replicatemap_previous.index))\n",
    "replicatemap.loc[setdiff1d(replicatemap.index,replicatemap_previous.index)]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'ACH-000845', 'ACH-000124', 'ACH-001556', 'ACH-000343', 'ACH-002399', 'ACH-002462', 'ACH-000999', 'ACH-002709', 'ACH-002508', 'ACH-000462', 'ACH-001850', 'ACH-002510', 'ACH-000602', 'ACH-001497', 'ACH-000110', 'ACH-000144', 'ACH-000609', 'ACH-000949', 'ACH-000533', 'ACH-000268', 'ACH-000565', 'ACH-001041', 'ACH-002464', 'ACH-000090', 'ACH-001846', 'ACH-000072', 'pDNA', 'ACH-000161', 'ACH-001574', 'ACH-000886', 'ACH-000267', 'ACH-000357', 'ACH-001993', 'ACH-000396', 'ACH-001638', 'ACH-001392', 'ACH-002463', 'ACH-000672', 'ACH-000972', 'ACH-000568', 'ACH-001498', 'ACH-000989', 'ACH-000538', 'ACH-001456', 'ACH-001376', 'ACH-001819', 'ACH-002467', 'ACH-000327', 'ACH-001613', 'ACH-001378', 'ACH-002466', 'ACH-000955', 'ACH-000353', 'ACH-002065'}\n",
      "New screens: 54\n"
     ]
    }
   ],
   "source": [
    "# need to analyze (new screens)\n",
    "newscreens = set(replicatemap.loc[setdiff1d(replicatemap.index,replicatemap_previous.index),'DepMap_ID'].unique())\n",
    "print(newscreens)\n",
    "print(\"New screens:\",len(newscreens))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "880"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(replicatemap['DepMap_ID'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([4, 3, 2])"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "replicatemap['pDNA_batch'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def id_converter(string):\n",
    "    return string.replace(\" \",\"_\").replace(\",\",\"_\").replace(\"/\",\"\").replace(\"(\",\"_\").replace(\")\",\"_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "controls = dict()  # [batch] list[repid]\n",
    "samples = dict()  # [batch][cell] list[repid]\n",
    "rep2cell = dict()\n",
    "for repid in replicatemap.index:\n",
    "    cell = replicatemap.loc[repid,'DepMap_ID']\n",
    "    batch = replicatemap.loc[repid,'pDNA_batch']\n",
    "    if cell == 'pDNA':\n",
    "        if batch not in controls:\n",
    "            controls[batch] = list()\n",
    "        controls[batch].append(id_converter(repid))\n",
    "    else:\n",
    "        if batch not in samples:\n",
    "            samples[batch] = dict()\n",
    "        if cell not in samples[batch]:\n",
    "            samples[batch][cell] = list()\n",
    "        samples[batch][cell].append(id_converter(repid))\n",
    "        rep2cell[id_converter(repid)] = cell\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{2: ['Avana4pDNA20160601-311cas9_RepG10_batch2',\n",
       "  'Avana4pDNA20160601-311cas9_RepG12_batch2',\n",
       "  'Avana4pDNA20160601-311cas9_RepG11_batch2',\n",
       "  'Avana4pDNA20160601-311cas9_RepG09_batch2'],\n",
       " 4: ['pDNA_10_pgwell_batch4',\n",
       "  'pDNA_100ng_batch4',\n",
       "  'pDNA_50ng_batch4',\n",
       "  'pDNA_100_pgwell_batch4',\n",
       "  'pDNA_100pgwell_batch4',\n",
       "  'pDNA_10ng_batch4',\n",
       "  'pDNA_50_pgwell_batch4'],\n",
       " 3: ['Avana_4+_Hu_pDNA__M-AA40__93015__batch3',\n",
       "  'Avana_4+_Hu_pDNA__M-AF34__112718__0.2pguL_batch3',\n",
       "  'Avana_4+_Hu_pDNA__M-AA40__93015___0.2pguL__batch3',\n",
       "  'Avana_4+_Hu_pDNA__M-AA40__93015__0.2pguL_batch3']}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "controls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1846"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(rep2cell)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# read readcount"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "readcount = pd.read_csv('Achilles_raw_readcounts.csv',index_col=0,header=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "repid_xref = {a:id_converter(a) for a in readcount.columns}\n",
    "readcount = readcount.rename(columns=repid_xref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>143B-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>21NT-311Cas9-RepAB-p6_batch4</th>\n",
       "      <th>2313287-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>2313287-311Cas9_RepB_p5_batch3</th>\n",
       "      <th>253J-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>59M-311Cas9_RepA_p4_batch3</th>\n",
       "      <th>...</th>\n",
       "      <th>YSCCC-311cas9-RepB-p6_batch3</th>\n",
       "      <th>ZR-75-1-311Cas9_RepA_p5_batch2</th>\n",
       "      <th>ZR-75-1-311Cas9_RepB_p5_batch2</th>\n",
       "      <th>pDNA_10_pgwell_batch4</th>\n",
       "      <th>pDNA_100_pgwell_batch4</th>\n",
       "      <th>pDNA_100ng_batch4</th>\n",
       "      <th>pDNA_100pgwell_batch4</th>\n",
       "      <th>pDNA_10ng_batch4</th>\n",
       "      <th>pDNA_50_pgwell_batch4</th>\n",
       "      <th>pDNA_50ng_batch4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Construct Barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>833</td>\n",
       "      <td>542</td>\n",
       "      <td>327</td>\n",
       "      <td>188</td>\n",
       "      <td>546</td>\n",
       "      <td>213</td>\n",
       "      <td>349</td>\n",
       "      <td>280</td>\n",
       "      <td>489</td>\n",
       "      <td>683</td>\n",
       "      <td>...</td>\n",
       "      <td>421</td>\n",
       "      <td>226</td>\n",
       "      <td>386</td>\n",
       "      <td>1228</td>\n",
       "      <td>2329</td>\n",
       "      <td>5950</td>\n",
       "      <td>392</td>\n",
       "      <td>5437</td>\n",
       "      <td>2001</td>\n",
       "      <td>5662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>1089</td>\n",
       "      <td>925</td>\n",
       "      <td>383</td>\n",
       "      <td>391</td>\n",
       "      <td>1151</td>\n",
       "      <td>736</td>\n",
       "      <td>727</td>\n",
       "      <td>565</td>\n",
       "      <td>770</td>\n",
       "      <td>1279</td>\n",
       "      <td>...</td>\n",
       "      <td>583</td>\n",
       "      <td>632</td>\n",
       "      <td>673</td>\n",
       "      <td>1859</td>\n",
       "      <td>3453</td>\n",
       "      <td>8484</td>\n",
       "      <td>543</td>\n",
       "      <td>7769</td>\n",
       "      <td>2802</td>\n",
       "      <td>8054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>95</td>\n",
       "      <td>108</td>\n",
       "      <td>32</td>\n",
       "      <td>69</td>\n",
       "      <td>72</td>\n",
       "      <td>46</td>\n",
       "      <td>60</td>\n",
       "      <td>99</td>\n",
       "      <td>102</td>\n",
       "      <td>18</td>\n",
       "      <td>...</td>\n",
       "      <td>42</td>\n",
       "      <td>44</td>\n",
       "      <td>39</td>\n",
       "      <td>314</td>\n",
       "      <td>594</td>\n",
       "      <td>1600</td>\n",
       "      <td>103</td>\n",
       "      <td>1397</td>\n",
       "      <td>440</td>\n",
       "      <td>1528</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1707 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      143B-311Cas9_RepA_p6_batch3  \\\n",
       "Construct Barcode                                   \n",
       "AAAAAAATCCAGCAATGCAG                          833   \n",
       "AAAAAACCCGTAGATAGCCT                         1089   \n",
       "AAAAAAGAAGAAAAAACCAG                           95   \n",
       "\n",
       "                      21NT-311Cas9-RepAB-p6_batch4  \\\n",
       "Construct Barcode                                    \n",
       "AAAAAAATCCAGCAATGCAG                           542   \n",
       "AAAAAACCCGTAGATAGCCT                           925   \n",
       "AAAAAAGAAGAAAAAACCAG                           108   \n",
       "\n",
       "                      2313287-311Cas9_RepA_p5_batch3  \\\n",
       "Construct Barcode                                      \n",
       "AAAAAAATCCAGCAATGCAG                             327   \n",
       "AAAAAACCCGTAGATAGCCT                             383   \n",
       "AAAAAAGAAGAAAAAACCAG                              32   \n",
       "\n",
       "                      2313287-311Cas9_RepB_p5_batch3  \\\n",
       "Construct Barcode                                      \n",
       "AAAAAAATCCAGCAATGCAG                             188   \n",
       "AAAAAACCCGTAGATAGCCT                             391   \n",
       "AAAAAAGAAGAAAAAACCAG                              69   \n",
       "\n",
       "                      253J-311Cas9_RepA_p5_batch3  \\\n",
       "Construct Barcode                                   \n",
       "AAAAAAATCCAGCAATGCAG                          546   \n",
       "AAAAAACCCGTAGATAGCCT                         1151   \n",
       "AAAAAAGAAGAAAAAACCAG                           72   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepA_p6_batch3  \\\n",
       "Construct Barcode                                       \n",
       "AAAAAAATCCAGCAATGCAG                              213   \n",
       "AAAAAACCCGTAGATAGCCT                              736   \n",
       "AAAAAAGAAGAAAAAACCAG                               46   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepB_p6_batch3  \\\n",
       "Construct Barcode                                       \n",
       "AAAAAAATCCAGCAATGCAG                              349   \n",
       "AAAAAACCCGTAGATAGCCT                              727   \n",
       "AAAAAAGAAGAAAAAACCAG                               60   \n",
       "\n",
       "                      5637-311Cas9_RepA_p6_batch3  \\\n",
       "Construct Barcode                                   \n",
       "AAAAAAATCCAGCAATGCAG                          280   \n",
       "AAAAAACCCGTAGATAGCCT                          565   \n",
       "AAAAAAGAAGAAAAAACCAG                           99   \n",
       "\n",
       "                      5637-311Cas9_RepB_p6_batch3  59M-311Cas9_RepA_p4_batch3  \\\n",
       "Construct Barcode                                                               \n",
       "AAAAAAATCCAGCAATGCAG                          489                         683   \n",
       "AAAAAACCCGTAGATAGCCT                          770                        1279   \n",
       "AAAAAAGAAGAAAAAACCAG                          102                          18   \n",
       "\n",
       "                      ...  YSCCC-311cas9-RepB-p6_batch3  \\\n",
       "Construct Barcode     ...                                 \n",
       "AAAAAAATCCAGCAATGCAG  ...                           421   \n",
       "AAAAAACCCGTAGATAGCCT  ...                           583   \n",
       "AAAAAAGAAGAAAAAACCAG  ...                            42   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepA_p5_batch2  \\\n",
       "Construct Barcode                                      \n",
       "AAAAAAATCCAGCAATGCAG                             226   \n",
       "AAAAAACCCGTAGATAGCCT                             632   \n",
       "AAAAAAGAAGAAAAAACCAG                              44   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepB_p5_batch2  pDNA_10_pgwell_batch4  \\\n",
       "Construct Barcode                                                             \n",
       "AAAAAAATCCAGCAATGCAG                             386                   1228   \n",
       "AAAAAACCCGTAGATAGCCT                             673                   1859   \n",
       "AAAAAAGAAGAAAAAACCAG                              39                    314   \n",
       "\n",
       "                      pDNA_100_pgwell_batch4  pDNA_100ng_batch4  \\\n",
       "Construct Barcode                                                 \n",
       "AAAAAAATCCAGCAATGCAG                    2329               5950   \n",
       "AAAAAACCCGTAGATAGCCT                    3453               8484   \n",
       "AAAAAAGAAGAAAAAACCAG                     594               1600   \n",
       "\n",
       "                      pDNA_100pgwell_batch4  pDNA_10ng_batch4  \\\n",
       "Construct Barcode                                               \n",
       "AAAAAAATCCAGCAATGCAG                    392              5437   \n",
       "AAAAAACCCGTAGATAGCCT                    543              7769   \n",
       "AAAAAAGAAGAAAAAACCAG                    103              1397   \n",
       "\n",
       "                      pDNA_50_pgwell_batch4  pDNA_50ng_batch4  \n",
       "Construct Barcode                                              \n",
       "AAAAAAATCCAGCAATGCAG                   2001              5662  \n",
       "AAAAAACCCGTAGATAGCCT                   2802              8054  \n",
       "AAAAAAGAAGAAAAAACCAG                    440              1528  \n",
       "\n",
       "[3 rows x 1707 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "readcount.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Avana4pDNA20160601-311cas9_RepG09_batch2',\n",
       "       'Avana4pDNA20160601-311cas9_RepG10_batch2',\n",
       "       'Avana4pDNA20160601-311cas9_RepG11_batch2',\n",
       "       'Avana4pDNA20160601-311cas9_RepG12_batch2',\n",
       "       'Avana_4+_Hu_pDNA__M-AA40__93015__0.2pguL_batch3',\n",
       "       'Avana_4+_Hu_pDNA__M-AA40__93015___0.2pguL__batch3',\n",
       "       'Avana_4+_Hu_pDNA__M-AA40__93015__batch3',\n",
       "       'Avana_4+_Hu_pDNA__M-AF34__112718__0.2pguL_batch3',\n",
       "       'pDNA_100_pgwell_batch4', 'pDNA_100ng_batch4',\n",
       "       'pDNA_100pgwell_batch4', 'pDNA_10_pgwell_batch4',\n",
       "       'pDNA_10ng_batch4', 'pDNA_50_pgwell_batch4', 'pDNA_50ng_batch4'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "setdiff1d(readcount.columns,list(rep2cell.keys()))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# write screen info for bagel pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "headertext = \"\"\"#All delimiter is tab\n",
    "#*: Replicates/Controls should be identical as headers of the readcount file.\n",
    "#[Screen Name/ID]       [Replicates*]   [Controls*]\n",
    "\"\"\"\n",
    "\n",
    "with open(\"DepMap_SCREENINFO.txt\",'w') as fout:\n",
    "    with open(\"DepMap_SCREENINFO_NEW_ONLY.txt\",'w') as fout2:\n",
    "        fout.write(headertext)\n",
    "        fout2.write(headertext)\n",
    "        for batch in samples:\n",
    "            ctrls = \",\".join(controls[batch])\n",
    "            for depmapid in samples[batch]:\n",
    "                temp = intersect1d(readcount.columns,samples[batch][depmapid])\n",
    "                if len(temp) > 0:\n",
    "                    reps = \",\".join(temp)\n",
    "\n",
    "                    fout.write(f\"{depmapid}\\t{reps}\\t{ctrls}\\n\")\n",
    "                    if depmapid in newscreens:\n",
    "                        fout2.write(f\"{depmapid}\\t{reps}\\t{ctrls}\\n\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#All delimiter is tab\r\n",
      "#*: Replicates/Controls should be identical as headers of the readcount file.\r\n",
      "#[Screen Name/ID]       [Replicates*]   [Controls*]\r\n",
      "ACH-002446\tCCLFUPGI0005T-311cas9-RepA-p6_batch4,CCLFUPGI0005T-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-001795\t94T778-311cas9-RepA-p6_batch4,94T778-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-001556\tMERO25-311cas9-RepA-p6_batch4,MERO25-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-000645\tJL-1-311Cas9_RepA_p6_batch4,JL-1-311Cas9_RepB_p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-002462\tRPE1SS48-311CAS9-RepA-p6_batch4,RPE1SS48-311CAS9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-002508\tWM3211-311cas9-RepA-p6_batch4,WM3211-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-001145\tOC316-311cas9-RepA-p6_batch4,OC316-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n"
     ]
    }
   ],
   "source": [
    "!head DepMap_SCREENINFO.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#All delimiter is tab\r\n",
      "#*: Replicates/Controls should be identical as headers of the readcount file.\r\n",
      "#[Screen Name/ID]       [Replicates*]   [Controls*]\r\n",
      "ACH-001556\tMERO25-311cas9-RepA-p6_batch4,MERO25-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-002462\tRPE1SS48-311CAS9-RepA-p6_batch4,RPE1SS48-311CAS9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-002508\tWM3211-311cas9-RepA-p6_batch4,WM3211-311cas9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-000161\tCORL105-311Cas9-RepA-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-000886\tNCI-H2009-311CAS9-RepA-p6_batch4,NCI-H2009-311CAS9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-001498\tFARAGE-311cas9-RepAB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n",
      "ACH-001993\tNALM16-311CAS9-RepA-p6_batch4,NALM16-311CAS9-RepB-p6_batch4\tpDNA_10_pgwell_batch4,pDNA_100ng_batch4,pDNA_50ng_batch4,pDNA_100_pgwell_batch4,pDNA_100pgwell_batch4,pDNA_10ng_batch4,pDNA_50_pgwell_batch4\r\n"
     ]
    }
   ],
   "source": [
    "!head DepMap_SCREENINFO_NEW_ONLY.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#module load R/3.4.4\n",
    "#for all\n",
    "#python run_bagel_crisprcleanr.py -i Achilles_raw_readcounts_filtered_singletarget_CCDS.tsv -s DepMap_SCREENINFO.txt  -o Avana_CCDS_filtering\n",
    "#for newly sequenced screens\n",
    "#python run_bagel_crisprcleanr.py -i Achilles_raw_readcounts_filtered_singletarget_CCDS.tsv -s DepMap_SCREENINFO_NEW_ONLY.txt  -o Avana_CCDS_filtering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# modify readcount"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgrna   genome_alignment             gene  n_alignments\n",
       "0  AAAAAAATCCAGCAATGCAG  chr10_110964620_+     SHOC2 (8036)             1\n",
       "1  AAAAAACCCGTAGATAGCCT   chr12_95003615_+  NDUFA12 (55967)             1\n",
       "2  AAAAAAGAAGAAAAAACCAG    chr4_75970356_-    SDAD1 (55153)             1"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# guidemap \n",
    "\n",
    "guidegenemap = pd.read_csv(\"Achilles_guide_map.csv\")\n",
    "guidegenemap.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap['sgrna'].is_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GeneID\n",
      "1     A1BG\n",
      "2      A2M\n",
      "3    A2MP1\n",
      "Name: Symbol, dtype: object\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>8036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>55967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>55153</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgrna   genome_alignment             gene  n_alignments  \\\n",
       "0  AAAAAAATCCAGCAATGCAG  chr10_110964620_+     SHOC2 (8036)             1   \n",
       "1  AAAAAACCCGTAGATAGCCT   chr12_95003615_+  NDUFA12 (55967)             1   \n",
       "2  AAAAAAGAAGAAAAAACCAG    chr4_75970356_-    SDAD1 (55153)             1   \n",
       "\n",
       "  official_symbol  entrez_id  \n",
       "0           SHOC2       8036  \n",
       "1         NDUFA12      55967  \n",
       "2           SDAD1      55153  "
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# id convert to new official symbol\n",
    "entrez2symbol = pd.read_csv(\"Homo_sapiens.gene_info.id2symbol\",sep=\"\\t\",index_col=0)['Symbol']\n",
    "print(entrez2symbol.head(3))\n",
    "\n",
    "\n",
    "genexref = dict()\n",
    "genexref_entrez = dict()\n",
    "for i in guidegenemap.index:\n",
    "    gene = guidegenemap.loc[i,'gene']\n",
    "    symbol = gene.split(\" \")[0]\n",
    "    entrezid = int(gene.split(\" \")[1].replace(\"(\",\"\").replace(\")\",\"\"))\n",
    "    genexref_entrez[i] = entrezid\n",
    "    if entrezid in entrez2symbol:\n",
    "        genexref[i] = entrez2symbol[entrezid] # use new\n",
    "    else:\n",
    "        genexref[i] = symbol # use old\n",
    "        \n",
    "guidegenemap['official_symbol'] = pd.Series(genexref)\n",
    "guidegenemap['entrez_id'] = pd.Series(genexref_entrez)\n",
    "guidegenemap.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# multi-protein check - CCDS + guidemap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"\\narray(['Withdrawn', 'Public', 'Reviewed, update pending',\\n       'Under review, update', 'Withdrawn, inconsistent annotation',\\n       'Under review, withdrawal'], dtype=object)\\n\""
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# CCDS + guidemap\n",
    "ccdsdata = pd.read_table('CCDS.20180614.txt',index_col=None,header=0)\n",
    "\n",
    "ccdsdata = ccdsdata[(ccdsdata['ccds_status'] == 'Public') | (ccdsdata['ccds_status'] == 'Reviewed, update pending')]\n",
    "'''\n",
    "array(['Withdrawn', 'Public', 'Reviewed, update pending',\n",
    "       'Under review, update', 'Withdrawn, inconsistent annotation',\n",
    "       'Under review, withdrawal'], dtype=object)\n",
    "'''\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chromosome</th>\n",
       "      <th>nc_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>ccds_id</th>\n",
       "      <th>ccds_status</th>\n",
       "      <th>cds_strand</th>\n",
       "      <th>cds_from</th>\n",
       "      <th>cds_to</th>\n",
       "      <th>cds_locations</th>\n",
       "      <th>match_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>SAMD11</td>\n",
       "      <td>148398</td>\n",
       "      <td>CCDS2.2</td>\n",
       "      <td>Public</td>\n",
       "      <td>+</td>\n",
       "      <td>925941</td>\n",
       "      <td>944152</td>\n",
       "      <td>[925941-926012, 930154-930335, 931038-931088, ...</td>\n",
       "      <td>Identical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>NOC2L</td>\n",
       "      <td>26155</td>\n",
       "      <td>CCDS3.1</td>\n",
       "      <td>Public</td>\n",
       "      <td>-</td>\n",
       "      <td>944693</td>\n",
       "      <td>959239</td>\n",
       "      <td>[944693-944799, 945056-945145, 945517-945652, ...</td>\n",
       "      <td>Identical</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>NC_000001.11</td>\n",
       "      <td>PLEKHN1</td>\n",
       "      <td>84069</td>\n",
       "      <td>CCDS4.1</td>\n",
       "      <td>Public</td>\n",
       "      <td>+</td>\n",
       "      <td>966531</td>\n",
       "      <td>974574</td>\n",
       "      <td>[966531-966613, 966703-966802, 970276-970422, ...</td>\n",
       "      <td>Identical</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #chromosome  nc_accession     gene  gene_id  ccds_id ccds_status cds_strand  \\\n",
       "1           1  NC_000001.11   SAMD11   148398  CCDS2.2      Public          +   \n",
       "2           1  NC_000001.11    NOC2L    26155  CCDS3.1      Public          -   \n",
       "3           1  NC_000001.11  PLEKHN1    84069  CCDS4.1      Public          +   \n",
       "\n",
       "  cds_from  cds_to                                      cds_locations  \\\n",
       "1   925941  944152  [925941-926012, 930154-930335, 931038-931088, ...   \n",
       "2   944693  959239  [944693-944799, 945056-945145, 945517-945652, ...   \n",
       "3   966531  974574  [966531-966613, 966703-966802, 970276-970422, ...   \n",
       "\n",
       "  match_type  \n",
       "1  Identical  \n",
       "2  Identical  \n",
       "3  Identical  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccdsdata.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#chromosome</th>\n",
       "      <th>nc_accession</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>ccds_id</th>\n",
       "      <th>ccds_status</th>\n",
       "      <th>cds_strand</th>\n",
       "      <th>cds_from</th>\n",
       "      <th>cds_to</th>\n",
       "      <th>cds_locations</th>\n",
       "      <th>match_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>16268</td>\n",
       "      <td>16</td>\n",
       "      <td>NC_000016.10</td>\n",
       "      <td>AARS</td>\n",
       "      <td>16</td>\n",
       "      <td>CCDS32474.1</td>\n",
       "      <td>Public</td>\n",
       "      <td>-</td>\n",
       "      <td>70252720</td>\n",
       "      <td>70282762</td>\n",
       "      <td>[70252720-70252905, 70253267-70253380, 7025371...</td>\n",
       "      <td>Identical</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      #chromosome  nc_accession  gene  gene_id      ccds_id ccds_status  \\\n",
       "16268          16  NC_000016.10  AARS       16  CCDS32474.1      Public   \n",
       "\n",
       "      cds_strand  cds_from    cds_to  \\\n",
       "16268          -  70252720  70282762   \n",
       "\n",
       "                                           cds_locations match_type  \n",
       "16268  [70252720-70252905, 70253267-70253380, 7025371...  Identical  "
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccdsdata[ccdsdata['gene_id']==16]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>8036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>55967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>55153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>chr2_33588446_-</td>\n",
       "      <td>FAM98A (25940)</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM98A</td>\n",
       "      <td>25940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>chr19_19891600_+</td>\n",
       "      <td>ZNF253 (56242)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF253</td>\n",
       "      <td>56242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74357</td>\n",
       "      <td>TGTACAAAGTCCTCTGAGCG</td>\n",
       "      <td>chr19_53149299_+</td>\n",
       "      <td>ZNF347 (84671)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF347</td>\n",
       "      <td>84671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74358</td>\n",
       "      <td>TGTAGTCCCATACACAAAAG</td>\n",
       "      <td>chr14_52514943_+</td>\n",
       "      <td>TXNDC16 (57544)</td>\n",
       "      <td>1</td>\n",
       "      <td>TXNDC16</td>\n",
       "      <td>57544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74359</td>\n",
       "      <td>TTTGTTTCCTCTGAAGTAGT</td>\n",
       "      <td>chr10_78035398_-</td>\n",
       "      <td>RPS24 (6229)</td>\n",
       "      <td>4</td>\n",
       "      <td>RPS24</td>\n",
       "      <td>6229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74360</td>\n",
       "      <td>TTTGTTTCCTCTTCTCGAGG</td>\n",
       "      <td>chr8_75016590_-</td>\n",
       "      <td>CRISPLD1 (83690)</td>\n",
       "      <td>1</td>\n",
       "      <td>CRISPLD1</td>\n",
       "      <td>83690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74361</td>\n",
       "      <td>TTTGTTTGTCTCATTCATGG</td>\n",
       "      <td>chr1_43283269_+</td>\n",
       "      <td>C1orf210 (149466)</td>\n",
       "      <td>1</td>\n",
       "      <td>C1orf210</td>\n",
       "      <td>149466</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74331 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna   genome_alignment               gene  \\\n",
       "0      AAAAAAATCCAGCAATGCAG  chr10_110964620_+       SHOC2 (8036)   \n",
       "1      AAAAAACCCGTAGATAGCCT   chr12_95003615_+    NDUFA12 (55967)   \n",
       "2      AAAAAAGAAGAAAAAACCAG    chr4_75970356_-      SDAD1 (55153)   \n",
       "3      AAAAAAGCTCAAGAAGGAGG    chr2_33588446_-     FAM98A (25940)   \n",
       "4      AAAAAAGGCTGTAAAAGCGT   chr19_19891600_+     ZNF253 (56242)   \n",
       "...                     ...                ...                ...   \n",
       "74357  TGTACAAAGTCCTCTGAGCG   chr19_53149299_+     ZNF347 (84671)   \n",
       "74358  TGTAGTCCCATACACAAAAG   chr14_52514943_+    TXNDC16 (57544)   \n",
       "74359  TTTGTTTCCTCTGAAGTAGT   chr10_78035398_-       RPS24 (6229)   \n",
       "74360  TTTGTTTCCTCTTCTCGAGG    chr8_75016590_-   CRISPLD1 (83690)   \n",
       "74361  TTTGTTTGTCTCATTCATGG    chr1_43283269_+  C1orf210 (149466)   \n",
       "\n",
       "       n_alignments official_symbol  entrez_id  \n",
       "0                 1           SHOC2       8036  \n",
       "1                 1         NDUFA12      55967  \n",
       "2                 1           SDAD1      55153  \n",
       "3                 1          FAM98A      25940  \n",
       "4                 1          ZNF253      56242  \n",
       "...             ...             ...        ...  \n",
       "74357             1          ZNF347      84671  \n",
       "74358             1         TXNDC16      57544  \n",
       "74359             4           RPS24       6229  \n",
       "74360             1        CRISPLD1      83690  \n",
       "74361             1        C1orf210     149466  \n",
       "\n",
       "[74331 rows x 6 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = guidegenemap[guidegenemap['entrez_id'].isin(ccdsdata['gene_id'].unique())]\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "b = a.groupby('sgrna').size()\n",
    "single_targeting_guides_CCDS = b[b==1].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "70981"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(single_targeting_guides_CCDS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>8036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>55967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>55153</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgrna   genome_alignment             gene  n_alignments  \\\n",
       "0  AAAAAAATCCAGCAATGCAG  chr10_110964620_+     SHOC2 (8036)             1   \n",
       "1  AAAAAACCCGTAGATAGCCT   chr12_95003615_+  NDUFA12 (55967)             1   \n",
       "2  AAAAAAGAAGAAAAAACCAG    chr4_75970356_-    SDAD1 (55153)             1   \n",
       "\n",
       "  official_symbol  entrez_id  \n",
       "0           SHOC2       8036  \n",
       "1         NDUFA12      55967  \n",
       "2           SDAD1      55153  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS = a[a['sgrna'].isin(single_targeting_guides_CCDS)]\n",
    "guidegenemap_single_CCDS.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>3183</td>\n",
       "      <td>ACAGAGCGCGAACTGCGCGT</td>\n",
       "      <td>chr6_166167463_-</td>\n",
       "      <td>TBXT (6862)</td>\n",
       "      <td>1</td>\n",
       "      <td>TBXT</td>\n",
       "      <td>6862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27196</td>\n",
       "      <td>CTTCCTGCTGGACTTCGTGG</td>\n",
       "      <td>chr6_166166779_-</td>\n",
       "      <td>TBXT (6862)</td>\n",
       "      <td>1</td>\n",
       "      <td>TBXT</td>\n",
       "      <td>6862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30002</td>\n",
       "      <td>GAGATGATCGTGACCAAGAA</td>\n",
       "      <td>chr6_166167394_-</td>\n",
       "      <td>TBXT (6862)</td>\n",
       "      <td>1</td>\n",
       "      <td>TBXT</td>\n",
       "      <td>6862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61373</td>\n",
       "      <td>TCACCAACAAGCTCAACGGA</td>\n",
       "      <td>chr6_166166603_-</td>\n",
       "      <td>TBXT (6862)</td>\n",
       "      <td>1</td>\n",
       "      <td>TBXT</td>\n",
       "      <td>6862</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment         gene  n_alignments  \\\n",
       "3183   ACAGAGCGCGAACTGCGCGT  chr6_166167463_-  TBXT (6862)             1   \n",
       "27196  CTTCCTGCTGGACTTCGTGG  chr6_166166779_-  TBXT (6862)             1   \n",
       "30002  GAGATGATCGTGACCAAGAA  chr6_166167394_-  TBXT (6862)             1   \n",
       "61373  TCACCAACAAGCTCAACGGA  chr6_166166603_-  TBXT (6862)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "3183             TBXT       6862  \n",
       "27196            TBXT       6862  \n",
       "30002            TBXT       6862  \n",
       "61373            TBXT       6862  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='TBXT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(70981, 6)"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS['sgrna'].is_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>143B-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>21NT-311Cas9-RepAB-p6_batch4</th>\n",
       "      <th>2313287-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>2313287-311Cas9_RepB_p5_batch3</th>\n",
       "      <th>253J-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>...</th>\n",
       "      <th>YSCCC-311cas9-RepB-p6_batch3</th>\n",
       "      <th>ZR-75-1-311Cas9_RepA_p5_batch2</th>\n",
       "      <th>ZR-75-1-311Cas9_RepB_p5_batch2</th>\n",
       "      <th>pDNA_10_pgwell_batch4</th>\n",
       "      <th>pDNA_100_pgwell_batch4</th>\n",
       "      <th>pDNA_100ng_batch4</th>\n",
       "      <th>pDNA_100pgwell_batch4</th>\n",
       "      <th>pDNA_10ng_batch4</th>\n",
       "      <th>pDNA_50_pgwell_batch4</th>\n",
       "      <th>pDNA_50ng_batch4</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sgrna</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>833</td>\n",
       "      <td>542</td>\n",
       "      <td>327</td>\n",
       "      <td>188</td>\n",
       "      <td>546</td>\n",
       "      <td>213</td>\n",
       "      <td>349</td>\n",
       "      <td>280</td>\n",
       "      <td>489</td>\n",
       "      <td>...</td>\n",
       "      <td>421</td>\n",
       "      <td>226</td>\n",
       "      <td>386</td>\n",
       "      <td>1228</td>\n",
       "      <td>2329</td>\n",
       "      <td>5950</td>\n",
       "      <td>392</td>\n",
       "      <td>5437</td>\n",
       "      <td>2001</td>\n",
       "      <td>5662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>1089</td>\n",
       "      <td>925</td>\n",
       "      <td>383</td>\n",
       "      <td>391</td>\n",
       "      <td>1151</td>\n",
       "      <td>736</td>\n",
       "      <td>727</td>\n",
       "      <td>565</td>\n",
       "      <td>770</td>\n",
       "      <td>...</td>\n",
       "      <td>583</td>\n",
       "      <td>632</td>\n",
       "      <td>673</td>\n",
       "      <td>1859</td>\n",
       "      <td>3453</td>\n",
       "      <td>8484</td>\n",
       "      <td>543</td>\n",
       "      <td>7769</td>\n",
       "      <td>2802</td>\n",
       "      <td>8054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>95</td>\n",
       "      <td>108</td>\n",
       "      <td>32</td>\n",
       "      <td>69</td>\n",
       "      <td>72</td>\n",
       "      <td>46</td>\n",
       "      <td>60</td>\n",
       "      <td>99</td>\n",
       "      <td>102</td>\n",
       "      <td>...</td>\n",
       "      <td>42</td>\n",
       "      <td>44</td>\n",
       "      <td>39</td>\n",
       "      <td>314</td>\n",
       "      <td>594</td>\n",
       "      <td>1600</td>\n",
       "      <td>103</td>\n",
       "      <td>1397</td>\n",
       "      <td>440</td>\n",
       "      <td>1528</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1708 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         GENE  143B-311Cas9_RepA_p6_batch3  \\\n",
       "sgrna                                                        \n",
       "AAAAAAATCCAGCAATGCAG    SHOC2                          833   \n",
       "AAAAAACCCGTAGATAGCCT  NDUFA12                         1089   \n",
       "AAAAAAGAAGAAAAAACCAG    SDAD1                           95   \n",
       "\n",
       "                      21NT-311Cas9-RepAB-p6_batch4  \\\n",
       "sgrna                                                \n",
       "AAAAAAATCCAGCAATGCAG                           542   \n",
       "AAAAAACCCGTAGATAGCCT                           925   \n",
       "AAAAAAGAAGAAAAAACCAG                           108   \n",
       "\n",
       "                      2313287-311Cas9_RepA_p5_batch3  \\\n",
       "sgrna                                                  \n",
       "AAAAAAATCCAGCAATGCAG                             327   \n",
       "AAAAAACCCGTAGATAGCCT                             383   \n",
       "AAAAAAGAAGAAAAAACCAG                              32   \n",
       "\n",
       "                      2313287-311Cas9_RepB_p5_batch3  \\\n",
       "sgrna                                                  \n",
       "AAAAAAATCCAGCAATGCAG                             188   \n",
       "AAAAAACCCGTAGATAGCCT                             391   \n",
       "AAAAAAGAAGAAAAAACCAG                              69   \n",
       "\n",
       "                      253J-311Cas9_RepA_p5_batch3  \\\n",
       "sgrna                                               \n",
       "AAAAAAATCCAGCAATGCAG                          546   \n",
       "AAAAAACCCGTAGATAGCCT                         1151   \n",
       "AAAAAAGAAGAAAAAACCAG                           72   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepA_p6_batch3  \\\n",
       "sgrna                                                   \n",
       "AAAAAAATCCAGCAATGCAG                              213   \n",
       "AAAAAACCCGTAGATAGCCT                              736   \n",
       "AAAAAAGAAGAAAAAACCAG                               46   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepB_p6_batch3  \\\n",
       "sgrna                                                   \n",
       "AAAAAAATCCAGCAATGCAG                              349   \n",
       "AAAAAACCCGTAGATAGCCT                              727   \n",
       "AAAAAAGAAGAAAAAACCAG                               60   \n",
       "\n",
       "                      5637-311Cas9_RepA_p6_batch3  \\\n",
       "sgrna                                               \n",
       "AAAAAAATCCAGCAATGCAG                          280   \n",
       "AAAAAACCCGTAGATAGCCT                          565   \n",
       "AAAAAAGAAGAAAAAACCAG                           99   \n",
       "\n",
       "                      5637-311Cas9_RepB_p6_batch3  ...  \\\n",
       "sgrna                                              ...   \n",
       "AAAAAAATCCAGCAATGCAG                          489  ...   \n",
       "AAAAAACCCGTAGATAGCCT                          770  ...   \n",
       "AAAAAAGAAGAAAAAACCAG                          102  ...   \n",
       "\n",
       "                      YSCCC-311cas9-RepB-p6_batch3  \\\n",
       "sgrna                                                \n",
       "AAAAAAATCCAGCAATGCAG                           421   \n",
       "AAAAAACCCGTAGATAGCCT                           583   \n",
       "AAAAAAGAAGAAAAAACCAG                            42   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepA_p5_batch2  \\\n",
       "sgrna                                                  \n",
       "AAAAAAATCCAGCAATGCAG                             226   \n",
       "AAAAAACCCGTAGATAGCCT                             632   \n",
       "AAAAAAGAAGAAAAAACCAG                              44   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepB_p5_batch2  pDNA_10_pgwell_batch4  \\\n",
       "sgrna                                                                         \n",
       "AAAAAAATCCAGCAATGCAG                             386                   1228   \n",
       "AAAAAACCCGTAGATAGCCT                             673                   1859   \n",
       "AAAAAAGAAGAAAAAACCAG                              39                    314   \n",
       "\n",
       "                      pDNA_100_pgwell_batch4  pDNA_100ng_batch4  \\\n",
       "sgrna                                                             \n",
       "AAAAAAATCCAGCAATGCAG                    2329               5950   \n",
       "AAAAAACCCGTAGATAGCCT                    3453               8484   \n",
       "AAAAAAGAAGAAAAAACCAG                     594               1600   \n",
       "\n",
       "                      pDNA_100pgwell_batch4  pDNA_10ng_batch4  \\\n",
       "sgrna                                                           \n",
       "AAAAAAATCCAGCAATGCAG                    392              5437   \n",
       "AAAAAACCCGTAGATAGCCT                    543              7769   \n",
       "AAAAAAGAAGAAAAAACCAG                    103              1397   \n",
       "\n",
       "                      pDNA_50_pgwell_batch4  pDNA_50ng_batch4  \n",
       "sgrna                                                          \n",
       "AAAAAAATCCAGCAATGCAG                   2001              5662  \n",
       "AAAAAACCCGTAGATAGCCT                   2802              8054  \n",
       "AAAAAAGAAGAAAAAACCAG                    440              1528  \n",
       "\n",
       "[3 rows x 1708 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#save filtered\n",
    "guidegenemap_single_CCDS.to_csv(\"Achilles_guide_map_filtered_singletarget_CCDS.csv\",index=False)\n",
    "cells = ['GENE'] + list(readcount.columns)\n",
    "readcount_mod = readcount.loc[single_targeting_guides_CCDS].copy()\n",
    "readcount_mod['GENE'] = pd.Series(index=guidegenemap_single_CCDS['sgrna'],data=guidegenemap_single_CCDS['official_symbol'].values)\n",
    "readcount_mod[cells].to_csv(\"Achilles_raw_readcounts_filtered_singletarget_CCDS.tsv\",sep=\"\\t\")\n",
    "readcount_mod[cells].head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>8036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>55967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>55153</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  sgrna   genome_alignment             gene  n_alignments  \\\n",
       "0  AAAAAAATCCAGCAATGCAG  chr10_110964620_+     SHOC2 (8036)             1   \n",
       "1  AAAAAACCCGTAGATAGCCT   chr12_95003615_+  NDUFA12 (55967)             1   \n",
       "2  AAAAAAGAAGAAAAAACCAG    chr4_75970356_-    SDAD1 (55153)             1   \n",
       "\n",
       "  official_symbol  entrez_id  \n",
       "0           SHOC2       8036  \n",
       "1         NDUFA12      55967  \n",
       "2           SDAD1      55153  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>chr2_33588446_-</td>\n",
       "      <td>FAM98A (25940)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>chr19_19891600_+</td>\n",
       "      <td>ZNF253 (56242)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74356</td>\n",
       "      <td>TTTGTTGGAGAGATGTACGA</td>\n",
       "      <td>chr3_185533596_-</td>\n",
       "      <td>LIPH (200879)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74357</td>\n",
       "      <td>TTTGTTGGCACAAATACGGG</td>\n",
       "      <td>chr14_46651531_+</td>\n",
       "      <td>RPL10L (140801)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74358</td>\n",
       "      <td>TTTGTTTCCTCTGAAGTAGT</td>\n",
       "      <td>chr10_78035398_-</td>\n",
       "      <td>RPS24 (6229)</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74359</td>\n",
       "      <td>TTTGTTTCCTCTTCTCGAGG</td>\n",
       "      <td>chr8_75016590_-</td>\n",
       "      <td>CRISPLD1 (83690)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74360</td>\n",
       "      <td>TTTGTTTGTCTCATTCATGG</td>\n",
       "      <td>chr1_43283269_+</td>\n",
       "      <td>C1orf210 (149466)</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74361 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna   genome_alignment               gene  \\\n",
       "0      AAAAAAATCCAGCAATGCAG  chr10_110964620_+       SHOC2 (8036)   \n",
       "1      AAAAAACCCGTAGATAGCCT   chr12_95003615_+    NDUFA12 (55967)   \n",
       "2      AAAAAAGAAGAAAAAACCAG    chr4_75970356_-      SDAD1 (55153)   \n",
       "3      AAAAAAGCTCAAGAAGGAGG    chr2_33588446_-     FAM98A (25940)   \n",
       "4      AAAAAAGGCTGTAAAAGCGT   chr19_19891600_+     ZNF253 (56242)   \n",
       "...                     ...                ...                ...   \n",
       "74356  TTTGTTGGAGAGATGTACGA   chr3_185533596_-      LIPH (200879)   \n",
       "74357  TTTGTTGGCACAAATACGGG   chr14_46651531_+    RPL10L (140801)   \n",
       "74358  TTTGTTTCCTCTGAAGTAGT   chr10_78035398_-       RPS24 (6229)   \n",
       "74359  TTTGTTTCCTCTTCTCGAGG    chr8_75016590_-   CRISPLD1 (83690)   \n",
       "74360  TTTGTTTGTCTCATTCATGG    chr1_43283269_+  C1orf210 (149466)   \n",
       "\n",
       "       n_alignments  \n",
       "0                 1  \n",
       "1                 1  \n",
       "2                 1  \n",
       "3                 1  \n",
       "4                 1  \n",
       "...             ...  \n",
       "74356             1  \n",
       "74357             1  \n",
       "74358             4  \n",
       "74359             1  \n",
       "74360             1  \n",
       "\n",
       "[74361 rows x 4 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# check difference between previsous version and current version\n",
    "temp = pd.read_csv(\"../avana_2020Q3/Achilles_guide_map_filtered_singletarget_CCDS.csv\",index_col=None)\n",
    "temp2 = pd.read_csv(\"../avana_2020Q3/Achilles_guide_map.csv\",index_col=None)\n",
    "temp2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>63976</td>\n",
       "      <td>TCTCAGCTAGGTACAACACA</td>\n",
       "      <td>chr15_45120684_+</td>\n",
       "      <td>DUOXA1 (90527)</td>\n",
       "      <td>1</td>\n",
       "      <td>DUOXA1</td>\n",
       "      <td>90527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73328</td>\n",
       "      <td>TTGTTGCTAGTAGAACAACG</td>\n",
       "      <td>chr1_88760262_-</td>\n",
       "      <td>PKN2 (5586)</td>\n",
       "      <td>1</td>\n",
       "      <td>PKN2</td>\n",
       "      <td>5586</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment            gene  n_alignments  \\\n",
       "63976  TCTCAGCTAGGTACAACACA  chr15_45120684_+  DUOXA1 (90527)             1   \n",
       "73328  TTGTTGCTAGTAGAACAACG   chr1_88760262_-     PKN2 (5586)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "63976          DUOXA1      90527  \n",
       "73328            PKN2       5586  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# new sgRNA in current version\n",
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['sgrna'].isin(set(guidegenemap_single_CCDS['sgrna'] ) - set(temp['sgrna']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sgrna, genome_alignment, gene, n_alignments]\n",
       "Index: []"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp2[temp2['sgrna'].isin(set(guidegenemap_single_CCDS['sgrna'] ) - set(temp['sgrna']))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0           SHOC2\n",
       "1         NDUFA12\n",
       "2           SDAD1\n",
       "3          FAM98A\n",
       "4          ZNF253\n",
       "           ...   \n",
       "74357      ZNF347\n",
       "74358     TXNDC16\n",
       "74359       RPS24\n",
       "74360    CRISPLD1\n",
       "74361    C1orf210\n",
       "Name: official_symbol, Length: 70981, dtype: object"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS['official_symbol']"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# run bagel pipeline\n",
    "#module load R/3.4.4\n",
    "#!python run_bagel_crisprcleanr.py -i Achilles_raw_readcounts_filtered_singletarget_CCDS.tsv -s DepMap_SCREENINFO_NEW_ONLY.txt -o Avana_CCDS_filtering > log_bagel\n",
    "#!python run_bagel_crisprcleanr.py -i Achilles_raw_readcounts_filtered_singletarget_CCDS.tsv -s DepMap_SCREENINFO.txt -o Avana_CCDS_filtering > log_bagel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load pr files\n",
    "\n",
    "def check_stat(prdata,bfthre):\n",
    "    a = prdata[prdata['BF']>=bfthre].sort_values('BF')\n",
    "    r = a.iloc[0]['Recall']\n",
    "    p = a.iloc[0]['Precision']\n",
    "    fmeasure = 2*r*p / (r+p)\n",
    "    return(len(a.index),fmeasure)\n",
    "    \n",
    "\n",
    "bfdata_single_CCDS = pd.DataFrame(index = sorted(guidegenemap_single_CCDS['official_symbol'].unique()))\n",
    "stats_single_CCDS = pd.DataFrame(columns=['Fmeasure','BF5_n'])\n",
    "for prfile in glob.glob(\"*.pr\"):\n",
    "    cell = prfile.split(\".\")[1]  #Avana_2020Q1_CCDS_filtering.ACH-002204.pr\n",
    "    prdata = pd.read_csv(prfile,sep=\"\\t\",index_col=0,header=0)\n",
    "    esscount, f = check_stat(prdata,5)\n",
    "    \n",
    "    stats_single_CCDS.loc[cell] = [f,esscount]\n",
    "    \n",
    "    bfdata_single_CCDS[cell] = prdata['BF']\n",
    "    \n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(18111, 808)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-001172</th>\n",
       "      <th>ACH-002017</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-000995</th>\n",
       "      <th>ACH-000391</th>\n",
       "      <th>ACH-001674</th>\n",
       "      <th>ACH-001795</th>\n",
       "      <th>ACH-002018</th>\n",
       "      <th>ACH-000172</th>\n",
       "      <th>ACH-000645</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001819</th>\n",
       "      <th>ACH-001041</th>\n",
       "      <th>ACH-002399</th>\n",
       "      <th>ACH-002464</th>\n",
       "      <th>ACH-000845</th>\n",
       "      <th>ACH-001574</th>\n",
       "      <th>ACH-002467</th>\n",
       "      <th>ACH-002463</th>\n",
       "      <th>ACH-000602</th>\n",
       "      <th>ACH-002510</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>A1BG</td>\n",
       "      <td>-24.971</td>\n",
       "      <td>-5.503</td>\n",
       "      <td>-21.171</td>\n",
       "      <td>-6.261</td>\n",
       "      <td>-17.224</td>\n",
       "      <td>-18.186</td>\n",
       "      <td>-21.752</td>\n",
       "      <td>-26.337</td>\n",
       "      <td>-2.121</td>\n",
       "      <td>-23.069</td>\n",
       "      <td>...</td>\n",
       "      <td>-31.675</td>\n",
       "      <td>-10.494</td>\n",
       "      <td>-12.439</td>\n",
       "      <td>-22.723</td>\n",
       "      <td>-10.896</td>\n",
       "      <td>-13.444</td>\n",
       "      <td>-10.789</td>\n",
       "      <td>-20.144</td>\n",
       "      <td>-21.737</td>\n",
       "      <td>-12.560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A1CF</td>\n",
       "      <td>-31.667</td>\n",
       "      <td>-0.874</td>\n",
       "      <td>-14.970</td>\n",
       "      <td>-10.729</td>\n",
       "      <td>-14.296</td>\n",
       "      <td>-24.983</td>\n",
       "      <td>-35.797</td>\n",
       "      <td>-24.425</td>\n",
       "      <td>-5.806</td>\n",
       "      <td>-29.454</td>\n",
       "      <td>...</td>\n",
       "      <td>-23.541</td>\n",
       "      <td>-29.790</td>\n",
       "      <td>-11.197</td>\n",
       "      <td>-24.534</td>\n",
       "      <td>-15.655</td>\n",
       "      <td>-23.113</td>\n",
       "      <td>-12.627</td>\n",
       "      <td>-23.998</td>\n",
       "      <td>-16.273</td>\n",
       "      <td>-15.189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A2M</td>\n",
       "      <td>-22.225</td>\n",
       "      <td>-3.130</td>\n",
       "      <td>-21.386</td>\n",
       "      <td>-2.465</td>\n",
       "      <td>-7.090</td>\n",
       "      <td>-13.480</td>\n",
       "      <td>-26.093</td>\n",
       "      <td>-15.808</td>\n",
       "      <td>-5.056</td>\n",
       "      <td>-19.130</td>\n",
       "      <td>...</td>\n",
       "      <td>-14.210</td>\n",
       "      <td>-12.345</td>\n",
       "      <td>-7.700</td>\n",
       "      <td>-15.693</td>\n",
       "      <td>-9.705</td>\n",
       "      <td>-11.160</td>\n",
       "      <td>-9.164</td>\n",
       "      <td>-16.173</td>\n",
       "      <td>-13.433</td>\n",
       "      <td>-9.381</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 808 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ACH-001172  ACH-002017  ACH-001145  ACH-000995  ACH-000391  ACH-001674  \\\n",
       "A1BG     -24.971      -5.503     -21.171      -6.261     -17.224     -18.186   \n",
       "A1CF     -31.667      -0.874     -14.970     -10.729     -14.296     -24.983   \n",
       "A2M      -22.225      -3.130     -21.386      -2.465      -7.090     -13.480   \n",
       "\n",
       "      ACH-001795  ACH-002018  ACH-000172  ACH-000645  ...  ACH-001819  \\\n",
       "A1BG     -21.752     -26.337      -2.121     -23.069  ...     -31.675   \n",
       "A1CF     -35.797     -24.425      -5.806     -29.454  ...     -23.541   \n",
       "A2M      -26.093     -15.808      -5.056     -19.130  ...     -14.210   \n",
       "\n",
       "      ACH-001041  ACH-002399  ACH-002464  ACH-000845  ACH-001574  ACH-002467  \\\n",
       "A1BG     -10.494     -12.439     -22.723     -10.896     -13.444     -10.789   \n",
       "A1CF     -29.790     -11.197     -24.534     -15.655     -23.113     -12.627   \n",
       "A2M      -12.345      -7.700     -15.693      -9.705     -11.160      -9.164   \n",
       "\n",
       "      ACH-002463  ACH-000602  ACH-002510  \n",
       "A1BG     -20.144     -21.737     -12.560  \n",
       "A1CF     -23.998     -16.273     -15.189  \n",
       "A2M      -16.173     -13.433      -9.381  \n",
       "\n",
       "[3 rows x 808 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(bfdata_single_CCDS.shape)\n",
    "bfdata_single_CCDS.head(3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fmeasure</th>\n",
       "      <th>BF5_n</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ACH-001344</td>\n",
       "      <td>0.041075</td>\n",
       "      <td>73.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000581</td>\n",
       "      <td>0.044899</td>\n",
       "      <td>98.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000498</td>\n",
       "      <td>0.073112</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-002017</td>\n",
       "      <td>0.111267</td>\n",
       "      <td>136.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000680</td>\n",
       "      <td>0.160860</td>\n",
       "      <td>202.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-001959</td>\n",
       "      <td>0.955614</td>\n",
       "      <td>1896.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-001852</td>\n",
       "      <td>0.957494</td>\n",
       "      <td>1829.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000406</td>\n",
       "      <td>0.960189</td>\n",
       "      <td>1853.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000947</td>\n",
       "      <td>0.962265</td>\n",
       "      <td>1766.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-001664</td>\n",
       "      <td>0.962470</td>\n",
       "      <td>1854.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>808 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Fmeasure   BF5_n\n",
       "ACH-001344  0.041075    73.0\n",
       "ACH-000581  0.044899    98.0\n",
       "ACH-000498  0.073112    63.0\n",
       "ACH-002017  0.111267   136.0\n",
       "ACH-000680  0.160860   202.0\n",
       "...              ...     ...\n",
       "ACH-001959  0.955614  1896.0\n",
       "ACH-001852  0.957494  1829.0\n",
       "ACH-000406  0.960189  1853.0\n",
       "ACH-000947  0.962265  1766.0\n",
       "ACH-001664  0.962470  1854.0\n",
       "\n",
       "[808 rows x 2 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "stats_single_CCDS.sort_values('Fmeasure')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "fmeasure80 = stats_single_CCDS[stats_single_CCDS['Fmeasure']>=0.80].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "670"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fmeasure80)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read fcfiles\n",
    "cell2reps = dict()\n",
    "fcdata_raw_rep = readcount_mod[['GENE']].copy()\n",
    "for batch in samples:\n",
    "    ctrls = \",\".join(controls[batch])\n",
    "    for depmapid in samples[batch]:\n",
    "        temp = intersect1d(readcount.columns,samples[batch][depmapid])\n",
    "        if len(temp) > 0:\n",
    "            cell2reps[depmapid] = temp\n",
    "        for sample in temp:\n",
    "            fc_ccr = pd.read_csv(f\"Avana_CCDS_filtering.{depmapid}.{sample}.CCR\",sep=\"\\t\",header=0,index_col=0)\n",
    "            fcdata_raw_rep[sample] = fc_ccr['avgFC']  # raw (because to check readcount depletion)\n",
    "            #fout.write(f\"{depmapid}\\t{reps}\\t{ctrls}\\n\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(70981, 1693)"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcdata_raw_rep.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "fcdata_raw_cells = readcount_mod[['GENE']].copy()\n",
    "for cell in cell2reps:\n",
    "    fcdata_raw_cells[cell] = fcdata_raw_rep[cell2reps[cell]].mean(axis=1)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "fcdata_raw_cells_genes  = fcdata_raw_cells.groupby('GENE').mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-002446</th>\n",
       "      <th>ACH-001795</th>\n",
       "      <th>ACH-001556</th>\n",
       "      <th>ACH-000645</th>\n",
       "      <th>ACH-002462</th>\n",
       "      <th>ACH-002508</th>\n",
       "      <th>ACH-001145</th>\n",
       "      <th>ACH-001804</th>\n",
       "      <th>ACH-000161</th>\n",
       "      <th>ACH-001835</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000699</th>\n",
       "      <th>ACH-000827</th>\n",
       "      <th>ACH-000651</th>\n",
       "      <th>ACH-000178</th>\n",
       "      <th>ACH-000579</th>\n",
       "      <th>ACH-000606</th>\n",
       "      <th>ACH-000684</th>\n",
       "      <th>ACH-000588</th>\n",
       "      <th>ACH-000849</th>\n",
       "      <th>ACH-000907</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.143446</td>\n",
       "      <td>0.235384</td>\n",
       "      <td>-0.281499</td>\n",
       "      <td>0.199070</td>\n",
       "      <td>0.024371</td>\n",
       "      <td>0.072926</td>\n",
       "      <td>0.101238</td>\n",
       "      <td>0.206614</td>\n",
       "      <td>0.139476</td>\n",
       "      <td>0.202259</td>\n",
       "      <td>...</td>\n",
       "      <td>0.222260</td>\n",
       "      <td>0.149519</td>\n",
       "      <td>0.388064</td>\n",
       "      <td>0.168876</td>\n",
       "      <td>-0.164640</td>\n",
       "      <td>0.251750</td>\n",
       "      <td>0.185482</td>\n",
       "      <td>0.205807</td>\n",
       "      <td>0.072397</td>\n",
       "      <td>0.081362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A1CF</td>\n",
       "      <td>-0.169845</td>\n",
       "      <td>0.461725</td>\n",
       "      <td>0.290286</td>\n",
       "      <td>0.409529</td>\n",
       "      <td>0.300679</td>\n",
       "      <td>0.237422</td>\n",
       "      <td>-0.111263</td>\n",
       "      <td>0.451126</td>\n",
       "      <td>0.147452</td>\n",
       "      <td>-0.083435</td>\n",
       "      <td>...</td>\n",
       "      <td>0.152002</td>\n",
       "      <td>0.175476</td>\n",
       "      <td>0.083015</td>\n",
       "      <td>0.269811</td>\n",
       "      <td>0.084951</td>\n",
       "      <td>0.124272</td>\n",
       "      <td>-0.088978</td>\n",
       "      <td>0.001478</td>\n",
       "      <td>0.090395</td>\n",
       "      <td>0.144916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A2M</td>\n",
       "      <td>0.033130</td>\n",
       "      <td>0.268733</td>\n",
       "      <td>0.044308</td>\n",
       "      <td>0.111904</td>\n",
       "      <td>0.019797</td>\n",
       "      <td>0.179093</td>\n",
       "      <td>0.280506</td>\n",
       "      <td>0.248023</td>\n",
       "      <td>0.030414</td>\n",
       "      <td>0.152142</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.157897</td>\n",
       "      <td>0.158926</td>\n",
       "      <td>-0.257370</td>\n",
       "      <td>0.129606</td>\n",
       "      <td>-0.063839</td>\n",
       "      <td>0.016948</td>\n",
       "      <td>-0.346891</td>\n",
       "      <td>0.220455</td>\n",
       "      <td>0.159385</td>\n",
       "      <td>-0.075927</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 808 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ACH-002446  ACH-001795  ACH-001556  ACH-000645  ACH-002462  ACH-002508  \\\n",
       "GENE                                                                           \n",
       "A1BG   -0.143446    0.235384   -0.281499    0.199070    0.024371    0.072926   \n",
       "A1CF   -0.169845    0.461725    0.290286    0.409529    0.300679    0.237422   \n",
       "A2M     0.033130    0.268733    0.044308    0.111904    0.019797    0.179093   \n",
       "\n",
       "      ACH-001145  ACH-001804  ACH-000161  ACH-001835  ...  ACH-000699  \\\n",
       "GENE                                                  ...               \n",
       "A1BG    0.101238    0.206614    0.139476    0.202259  ...    0.222260   \n",
       "A1CF   -0.111263    0.451126    0.147452   -0.083435  ...    0.152002   \n",
       "A2M     0.280506    0.248023    0.030414    0.152142  ...   -0.157897   \n",
       "\n",
       "      ACH-000827  ACH-000651  ACH-000178  ACH-000579  ACH-000606  ACH-000684  \\\n",
       "GENE                                                                           \n",
       "A1BG    0.149519    0.388064    0.168876   -0.164640    0.251750    0.185482   \n",
       "A1CF    0.175476    0.083015    0.269811    0.084951    0.124272   -0.088978   \n",
       "A2M     0.158926   -0.257370    0.129606   -0.063839    0.016948   -0.346891   \n",
       "\n",
       "      ACH-000588  ACH-000849  ACH-000907  \n",
       "GENE                                      \n",
       "A1BG    0.205807    0.072397    0.081362  \n",
       "A1CF    0.001478    0.090395    0.144916  \n",
       "A2M     0.220455    0.159385   -0.075927  \n",
       "\n",
       "[3 rows x 808 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcdata_raw_cells_genes.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "fcdata_raw_cells_genes.to_csv(\"table_fc_cellline_genelevel.tsv\",sep=\"\\t\")\n",
    "fcdata_raw_cells.to_csv(\"table_fc_cellline_sgrna.tsv\",sep=\"\\t\")\n",
    "fcdata_raw_rep.to_csv(\"table_fc_rep_sgrna.tsv\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-63-fcf1c6b2b717>, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-63-fcf1c6b2b717>\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    Avana_CCDS_filtering.ACH-000619.PACADD188-311cas9_RepA_p6_batch3.CCR\u001b[0m\n\u001b[0m                                            ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "Avana_CCDS_filtering.ACH-000619.PACADD188-311cas9_RepA_p6_batch3.CCR\n",
    "Avana_CCDS_filtering.ACH-001382.PACADD188-311cas9_RepA_p6_batch3.CCR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sgrna</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>SHOC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>NDUFA12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>SDAD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>FAM98A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>ZNF253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTGGAGAGATGTACGA</td>\n",
       "      <td>LIPH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTGGCACAAATACGGG</td>\n",
       "      <td>RPL10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTCCTCTGAAGTAGT</td>\n",
       "      <td>RPS24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTCCTCTTCTCGAGG</td>\n",
       "      <td>CRISPLD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTGTCTCATTCATGG</td>\n",
       "      <td>C1orf210</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70981 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          GENE\n",
       "sgrna                         \n",
       "AAAAAAATCCAGCAATGCAG     SHOC2\n",
       "AAAAAACCCGTAGATAGCCT   NDUFA12\n",
       "AAAAAAGAAGAAAAAACCAG     SDAD1\n",
       "AAAAAAGCTCAAGAAGGAGG    FAM98A\n",
       "AAAAAAGGCTGTAAAAGCGT    ZNF253\n",
       "...                        ...\n",
       "TTTGTTGGAGAGATGTACGA      LIPH\n",
       "TTTGTTGGCACAAATACGGG    RPL10L\n",
       "TTTGTTTCCTCTGAAGTAGT     RPS24\n",
       "TTTGTTTCCTCTTCTCGAGG  CRISPLD1\n",
       "TTTGTTTGTCTCATTCATGG  C1orf210\n",
       "\n",
       "[70981 rows x 1 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "readcount_mod[['GENE']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write summary files\n",
    "bfdata_single_CCDS.to_csv(\"table_Avana2020Q4_CRISPRcleanR_corrected_all\",sep=\"\\t\")\n",
    "bfdata_single_CCDS[fmeasure80].to_csv(\"table_Avana2020Q4_CRISPRcleanR_corrected_fmeausre80\",sep=\"\\t\")\n",
    "stats_single_CCDS.sort_values('Fmeasure').to_csv(\"table_fmeasure_esscount_bf5.txt\",sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "bfdata_single_CCDS[fmeasure80].dropna().to_csv(\"table_Avana2020Q4_CRISPRcleanR_corrected_fmeausre80.noNA\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "bfdata_single_CCDS.dropna().to_csv(\"table_Avana2020Q4_CRISPRcleanR_corrected_all.noNA\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# load pr files\n",
    "\n",
    "def check_stat(prdata,bfthre):\n",
    "    a = prdata[prdata['BF']>=bfthre].sort_values('BF')\n",
    "    r = a.iloc[0]['Recall']\n",
    "    p = a.iloc[0]['Precision']\n",
    "    fmeasure = 2*r*p / (r+p)\n",
    "    return(len(a.index),fmeasure)\n",
    "    \n",
    "\n",
    "bfdata_single_CCDS_nonmc = pd.DataFrame(index = sorted(guidegenemap_single_CCDS['official_symbol'].unique()))\n",
    "stats_single_CCDS_nonmc = pd.DataFrame(columns=['Fmeasure','BF5_n'])\n",
    "for prfile in glob.glob(\"*.nonmc_pr\"):\n",
    "    cell = prfile.split(\".\")[1]  #Avana_2020Q1_CCDS_filtering.ACH-002204.pr\n",
    "    prdata = pd.read_csv(prfile,sep=\"\\t\",index_col=0,header=0)\n",
    "    esscount, f = check_stat(prdata,5)\n",
    "    \n",
    "    stats_single_CCDS_nonmc.loc[cell] = [f,esscount]\n",
    "    \n",
    "    bfdata_single_CCDS_nonmc[cell] = prdata['BF']\n",
    "    \n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "stats_single_CCDS_nonmc.sort_values('Fmeasure')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fmeasure80_nonmc = stats_single_CCDS_nonmc[stats_single_CCDS_nonmc['Fmeasure']>=0.80].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(len(fmeasure80_nonmc))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write summary files\n",
    "bfdata_single_CCDS_nonmc.to_csv(\"table_Avana2020Q4_CRISPRcleanR_all\",sep=\"\\t\")\n",
    "bfdata_single_CCDS_nonmc[fmeasure80_nonmc].to_csv(\"table_Avana2020Q4_CRISPRcleanR_fmeausre80\",sep=\"\\t\")\n",
    "stats_single_CCDS_nonmc.sort_values('Fmeasure').to_csv(\"table_fmeasure_esscount_bf5_nonmc.txt\",sep=\"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bfdata_single_CCDS_nonmc[fmeasure80_nonmc].dropna().to_csv(\"table_Avana2020Q4_CRISPRcleanR_fmeausre80.noNA\",sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SAMD11_GGGCTGCCGAATATCCTCCC     0.220217\n",
       "SAMD11_GCCAGCGATGGTGACAGCGA     0.149426\n",
       "SAMD11_AAGAGATGTTCACGTCCCAG     0.472329\n",
       "SAMD11_GCCACGGTGAGGACCCACCC    -0.712046\n",
       "NOC2L_GCTGCGGCACATCAGCGTGC      0.383109\n",
       "                                  ...   \n",
       "RPS4Y2_GACAAAGGGAATTCCACACC     0.754422\n",
       "CYorf17_TGTGGACGATATCGTGGCAG    0.202658\n",
       "CYorf17_TATCGTGGCAGTGGTCCAGG   -0.502224\n",
       "CYorf17_GTCCTCATTGCTCTCTCCGG    0.170740\n",
       "CYorf17_TGAGCCAAGGCAGGAAACCC   -0.592677\n",
       "Name: correctedFC, Length: 69981, dtype: float64"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aa = pd.read_csv(\"/export/ekim8/bagelv2/builds/pipeline/temp/HAP1-TKOv3-EXAMPLE.txt.BATCH1_HAP1_T18_STARVED.T18_A_Starved.CCR\",sep=\"\\t\",index_col=0,header=0)\n",
    "bb = pd.read_csv(\"/export/ekim8/bagelv2/builds/pipeline/temp2/HAP1-TKOv3-EXAMPLE.txt.BATCH1_HAP1_T18_STARVED.T18_A_Starved.CCR\",sep=\"\\t\",index_col=0,header=0)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(0, 1000)"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAD/CAYAAAD8MdEiAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjMsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+AADFEAAATU0lEQVR4nO3dfZBd9X3f8fcHYcnYCsRGjCAKhrQGAkaup2zdui14ZTdtiJPY4WkGmMaUEmWGxukfdgA3tVtcphDXSv6QiW15KHgmJOIhHkONa9d2snWcYGdsD44yNkhWgoiJeVRR0APm6ds/7hFeNrqrXd27e/fq937N3EH3fM/v7O/L2dVH5+GeTVUhSWrXEaOegCRptAwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGzTkIkhyd5KNJfm/G8rOTbEnywyR/meSt02qvSPKxJE8l2ZXkE0leMZexkqTFMacgSPKLwE7gyhnLXw18Gvht4FhgA/CHSVZ2q7wX+MfAmcAbgDcB75vjWEnSIphTEFTV3VV1JPChGaW3AQ9V1c1VtbuqbgZ+ALy9q18IXFdV36+q7wMfBi6a41hJ0iIY9BrBqcCOGcu2dssPVJ+tNrMuSVoERw44/lXAvhnL9gAr+9T3AK9KkjmMfUmS9cB6gKOOOuqsE088se+EXnzxRY444vC+Bt5Cj889/wLPz/L0k6NesWzxJrNAWtiP0Eaf49Dj1q1bn6iq4w5UGzQI9gLLZyxbQe8v9APVVwD7qqqSHGzsS6pqE7AJYGJior7xjW/0ndDU1BSTk5PzaGH8tNDjxlvvYsOW/t+eD9zwjkWczcJoYT9CG32OQ49JZp6BecmgEbYVOH3GsjOAbX3qs9Vm1iVJi2DQIPhj4IQk/y7Jq5P8MrAG+FJXvxO4KsmaJD8B/AZwxxzHSpIWwVxvHz03yRPAVcAFSZ5I8u+rajdwPr1bQv8fcDVwflU93Q3dAHwb+G732kLvziHmMFaStAjmdI2gqv43sKpPbYreZwQOVHsW+JXuNa+xkqTFsbQvc0uSFpxBIEmNMwgkqXGDfo5AWnJOvuaeWesPHgafQZCGySMCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaZxBIUuMMAklqnEEgSY0zCCSpcQaBJDXOIJCkxhkEktQ4g0CSGmcQSFLjDAJJapxBIEmNMwgkqXEGgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1LihBEGSk5LcneSpJI8luSnJq7vaeUm2J3kmydeSnDlt3NFJbkuyJ8njSa4bxnwkSXM3rCOCzcAW4ATgTcA/AP5bktcBtwDvAY4DPgvcmSTduA8DK4CTgbOBS5JcPKQ5SZLmYFhBsBb4YlXtq6q/BT4P/BTwzm7556rqaeB64Hjgjd24C4EPVNXjVXU/cCNw0ZDmJEmag2EFwUeAP0jyq0lOo/cX/AbgVGDH/pWq6gVgO3BqkmOB106vA1u7MZKkRZKqGnwjyT8HPgr8BXAecC+9MPgd4JGq+s1p634FuBn4EvAQsLyqnutqbwNurqqTZmx/PbAeYPXq1Wdt3ry571x2797NypUrB+5pKWuhx8d27uLRff3ra9cc07e25eFds257trGLqYX9CG30OQ49rlu37ptVNXGg2pGDbjzJj9M7939WVf11kl+nd13gk8BjwPIZQ1YAe4C93fvlwHMzai9TVZuATQATExM1OTnZdz5TU1PMVj8ctNDjxlvvYsOW/t+eD1462bd22TX3zLrt2cYuphb2I7TR57j3OIxTQ68HllXVXwNU1d8Bn6J3HWArcPr+FZMsA04DtlXVk8DO6XXgDGDbEOYkSZqjYQTBd4Bnklyb5NVJjqd3GufzwN3AZJJzk6wErqJ3lHBfN/ZO4INJViU5BbgSuGMIc5IkzdHAQVBVe4FzgXOAR4BvAfcD76+qHcDl9O4GehJ4F3BB/ejCxNXAs/SuFdwL3AbcOuicJElzN/A1AoCq+hawrk/tduD2PrWngAuGMQdJ0qHxEROS1DiDQJIaZxBIUuMMAklqnEEgSY0zCCSpcQaBJDXOIJCkxhkEktQ4g0CSGmcQSFLjDAJJapxBIEmNMwgkqXEGgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaZxBIUuMMAklqnEEgSY0zCCSpcQaBJDXOIJCkxhkEktS4oQVBkrcm+XKSv0uyN8lruuVXJvl+kn1J/k+Sn5w2Zk2SL3S1h5P82rDmI0mam6EEQZK3AzcDvw0cD5wEPJ3knwEfAs4HTgC2AzdNG3oT8Fdd7Tzg2iRvGcacJElzc+SQtvMR4Iqq+qPu/V6AJBcCn6qqr3fvPwj8oDtaeBF4O7CqqnYBX0/y+8BFwL1Dmpck6SAGPiJI8hPAWmB9kkeTPJnkE0mWA6cCO/avW1WPA7uA13evp7sQ2G9rN0aStEhSVYNtoHcq5/PAzwNfB44D7uj+/EZgc1V9ctr6DwHvBp7vamum1S4H3l1Vb53xNdYD6wFWr1591ubNm/vOZ/fu3axcuXKgnpa6Fnp8bOcuHt3Xv752zTF9a1se3tW3drCxi6mF/Qht9DkOPa5bt+6bVTVxoNowTg0V8MOq+pPu/cNJNgDXAw8Ay2esvwLYA7wwS+3lX6BqE7AJYGJioiYnJ/tOZmpqitnqh4MWetx4611s2NL/2/PBSyf71i675p5Ztz3b2MXUwn6ENvoc9x6HcbF4O3BsktXTlh0JPEnvVM/p+xcmOQ5YBXyve72mW7bfGcC2IcxJkjRHAwdBd97/c8DHk7wmyeuA3wB+D/hD4JIkb05yNPBfgT+uqp3dtYEvAx9KckySCeASeqeVJEmLZFifI3g3sI/eraB/Si8Afreq/gz4L8BngMeA04DLp427AjgFeBS4G7i2qr46pDlJkuZgKLePVtVOev+aP1BtI7CxT+1vgH81jDlIkg6Nj5iQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaZxBIUuMMAklqnEEgSY0zCCSpcQaBJDXOIJCkxhkEktQ4g0CSGmcQSFLjDAJJapxBIEmNMwgkqXEGgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1LihBUGSVyb5v0m+Om3ZeUm2J3kmydeSnDmtdnSS25LsSfJ4kuuGNRdJ0twNJQiSLAM2AyumLXsdcAvwHuA44LPAnUnSrfLhbv2TgbOBS5JcPIz5SJLmblhHBJ8AHgA+Pm3ZO4EvVtXnqupp4HrgeOCNXf1C4ANV9XhV3Q/cCFw0pPlIkuZo4CBI8lvAEVV19YzSqcCO/W+q6gVgO3BqkmOB106vA1u7MZKkRXTkIIOTXA6cBpx/gPKrgEdmLNsDrOxqAPsOUDvQ11kPrAdYvXo1U1NTfee0e/fuWeuHgxZ6XH0UvHft833rs/U/27iDjV1MLexHaKPPce9xoCAA3gD8LLCnO/W/DFiW5BngD4DlM9ZfQe8v/L3d++XAczNqf09VbQI2AUxMTNTk5GTfCU1NTTFb/XDQQo8bb72LDVv6f3s+eOlk39pl19wz67ZnG7uYWtiP0Eaf497jQKeGquq9VfXK/S/gV4A/6/78LeD0/et2F5RPA7ZV1ZPAzul14Axg2yDzkSTN30J+juBuYDLJuUlWAlcBjwH3dfU7gQ8mWZXkFOBK4I4FnI8k6QAWLAiqagdwOb27gZ4E3gVcUFXVrXI18CzwEHAvcBtw60LNR5J0YINeI3iZqrqF3mcH9r+/Hbi9z7pPARcM8+vPxcmznD9+8IZ3LOJMJGlp8BETktQ4g0CSGmcQSFLjDAJJapxBIEmNMwgkqXEGgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaN9RfVTnuZvs1luCvstTC8vtPo+IRgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaZxBIUuMMAklq3MBBkOQ/Jflukr1JHkzyvhn185JsT/JMkq8lOXNa7egktyXZk+TxJNcNOh9J0vwM44iggMuB1wK/ALwvybkASV4H3AK8BzgO+CxwZ5J0Yz8MrABOBs4GLkly8RDmJEmao4GDoKqur6p7q+qZqtoCfAV4U1d+J/DFqvpcVT0NXA8cD7yxq18IfKCqHq+q+4EbgYsGnZMkae6Geo0gyTJgAvhOt+hUYMf+elW9AGwHTk1yLL2jiB3TNrG1GyNJWiSpquFtLPkI8C+Bf1FVLyS5CXikqn5z2jpfAW4GvgQ8BCyvque62tuAm6vqpBnbXQ+sB1i9evVZmzdv7juH3bt3s3Llyr71LQ/vOsTuYO2aYw557DAdrMfDwWM7d/Hovv712fbFwfbxUt2P4zLv+Wrh+3Ucely3bt03q2riQLWh/c7iJNcAvwic0/3LH2AvsHzGqiuAPV2Nrv7cjNrLVNUmYBPAxMRETU5O9p3H1NQUs9UvO8jvhZ3Ng5f23+5iOliPh4ONt97Fhi39vz1n2xcH28dLdT+Oy7znq4Xv13HvcRh3DSXJbwGX0AuBR6aVtwKnT1t3GXAasK2qngR2Tq8DZwDbBp2TJGnuhnGN4Fbgn/D3QwDgbmAyyblJVgJXAY8B93X1O4EPJlmV5BTgSuCOIcxJkjRHwwiCi4FzgCeSPN+9vgdQVTvo3Vp6I/Ak8C7ggvrRhYmrgWfpXSu4F7iNXrBIkhbJwNcIqioHqd8O3N6n9hRwwaBzkCQdOh8xIUmNMwgkqXEGgSQ1ziCQpMYZBJLUOINAkhpnEEhS4wwCSWqcQSBJjTMIJKlxBoEkNc4gkKTGGQSS1DiDQJIaZxBIUuMMAklqnEEgSY0zCCSpcQaBJDXOIJCkxhkEktQ4g0CSGmcQSFLjDAJJapxBIEmNO3LUExgnJ19zT9/agze8YxFnIknD4xGBJDXOIJCkxnlqSJKWiNlOP8PCnYL2iECSGmcQSFLjDAJJapxBIEmNMwgkqXEjv2soyRrgfwLnADuB66vqo6OdlaTFMKq7ZPRyS+GI4Cbgr4ATgPOAa5O8ZbRTkqR2jPSIIMkxwNuBVVW1C/h6kt8HLgLuHeXcJKkVoz419Hrg6S4E9tsK/OyI5qNFNNtpgfeuXcSJSI1LVY3uiydnA5uras20ZZcD766qt05bth5Y3709DXhgls2uAp5YgOkuJfZ4eGihR2ijz3Ho8aSqOu5AhVEfEewFls9YtgLYM31BVW0CNs1lg0m+UVUTw5ne0mSPh4cWeoQ2+hz3Hkd9sfh7wGuSTE+pM4BtI5qPJDVnpEHQXRv4MvChJMckmQAuAe4Y5bwkqSWjPiIAuAI4BXgUuBu4tqq+OsD25nQKaczZ4+GhhR6hjT7HuseRXiyWJI3eUjgikCSN0NgHQZI3JLk3yTNJtie5YJZ1lyXZnGRHkr1Jvp1kyX9mYT49ThuzphtzxWLMcb66+X0hyb4kDyf5tVnWnXf/S8F8euzWX5Xk00muW6w5DsM89+XGJN/rfv4eSPJvF3Ouh2qePd6QZFu37g+SfCTJqO/QnF1Vje0LCPBd4D8DPwb8HPA0vftlD7T+cuB/0PsswnLgAmA3cOyoexlWj92YDcDzQAFXjLqHPnP8PPAx4MeBfwo8CbxlGP0vlddce+zWfU+3z14Erhv13Bewz/8O/KPu5++c7ufvzFH3MOQeL6P3Ydll3X8fBn5h1D3M2t+oJzDgzlkLPE53raNb9r+A/ziPbTzWb4cuhdcgPQJTSzEIgGOA54Bjpi3bCPzOQuzjpd7jjHG3jFMQHGqf09b9c+DiUfexQPvyCODNwA7glFH3Mdtr3E8NnQr8TXX/1ztbu+UHleQfAkfT+zzDUjVQj0tUv0eLHKince1/Pj2Os0Pus3vW2E8D31mguQ3LvHvsnqr8LPAl4D9U1ZL+bNS4B8GrgH0zlu0BVh5sYJIVwKeAjVX1+ALMbVgOucclbD49jWv/4zrv+TqkPpME+CTwhar69gLNbVjm3WNVPcyPTn99fKk/UXlJX8BI8hlgsk/5BnqfQD7oIyoOsN3lwO30Tjm8f7BZDmahelzi5vRokUNYdykZ13nP17z77ELgRuBE4GcWbmpDc0j7sqpeBO7rfsbPZwk/UXlJB0FVvWu2epK1wClJllXVC93iM4A/mmXM0cCn6V0b+OWqen5Y8z0UC9HjGHjp0SLTjsb6PVpkK+PZ/3x6HGfz6jPJK+hdBzkB+Jmq2r0osxzMoPvyx4BdB11rhMb91NBfAo8A70+yMsm/ofev68/AS7d83Z/kzd37VcCfAH8BXDrqEJijefU4Duogjxbp+vmlbvVZ+1+q5tnj2JpPn0leSe/umyOAc8ckBObb408m+Vh3y/OKJD8H/BJL/bE5o75aPYQr+mvp3XnwQ3q/6eyiabWT6d1COdm9n+zePz/jddOo+xhWj92y6+g9Evc5erdaPgH89Kj7mNHTifQupD0D/C3w69NqBVw2l/6X8muePf5qt5+eoXcq4gngX4+6h2H2Oe17debP35dH3cMQezwGuJPeLaN7gfuAnx/1/A/28hETktS4cT81JEkakEEgSY0zCCSpcQaBJDXOIJCkxhkEktQ4g0CSGmcQSFLjDAJJatz/BzH/Efdk3U1PAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "temp = pd.DataFrame()\n",
    "temp['a'] = aa['correctedFC']\n",
    "temp['b'] = bb['correctedFC']\n",
    "(temp['a']-temp['b']).hist(bins=40)\n",
    "ylim(0,1000)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>ACH-000012</th>\n",
       "      <th>ACH-000013</th>\n",
       "      <th>ACH-000014</th>\n",
       "      <th>ACH-000015</th>\n",
       "      <th>ACH-000017</th>\n",
       "      <th>ACH-000018</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001849</th>\n",
       "      <th>ACH-001955</th>\n",
       "      <th>ACH-001956</th>\n",
       "      <th>ACH-001957</th>\n",
       "      <th>ACH-001959</th>\n",
       "      <th>ACH-001960</th>\n",
       "      <th>ACH-002001</th>\n",
       "      <th>ACH-002002</th>\n",
       "      <th>ACH-002004</th>\n",
       "      <th>ACH-002005</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>A1BG</td>\n",
       "      <td>-13.689</td>\n",
       "      <td>-12.704</td>\n",
       "      <td>-27.260</td>\n",
       "      <td>-16.002</td>\n",
       "      <td>-22.922</td>\n",
       "      <td>-23.405</td>\n",
       "      <td>-14.482</td>\n",
       "      <td>-13.137</td>\n",
       "      <td>-14.341</td>\n",
       "      <td>-7.315</td>\n",
       "      <td>...</td>\n",
       "      <td>-23.337</td>\n",
       "      <td>-26.547</td>\n",
       "      <td>-17.751</td>\n",
       "      <td>-25.570</td>\n",
       "      <td>-21.672</td>\n",
       "      <td>-24.316</td>\n",
       "      <td>-13.376</td>\n",
       "      <td>-19.421</td>\n",
       "      <td>-30.341</td>\n",
       "      <td>-13.039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A1CF</td>\n",
       "      <td>-12.656</td>\n",
       "      <td>-9.979</td>\n",
       "      <td>-18.618</td>\n",
       "      <td>-12.640</td>\n",
       "      <td>-38.198</td>\n",
       "      <td>-17.486</td>\n",
       "      <td>-12.782</td>\n",
       "      <td>-23.265</td>\n",
       "      <td>-19.222</td>\n",
       "      <td>-11.960</td>\n",
       "      <td>...</td>\n",
       "      <td>-25.842</td>\n",
       "      <td>-21.578</td>\n",
       "      <td>-36.392</td>\n",
       "      <td>-30.543</td>\n",
       "      <td>-32.026</td>\n",
       "      <td>-20.854</td>\n",
       "      <td>-24.389</td>\n",
       "      <td>-25.703</td>\n",
       "      <td>-26.582</td>\n",
       "      <td>-9.095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>A2M</td>\n",
       "      <td>-6.802</td>\n",
       "      <td>-10.975</td>\n",
       "      <td>-23.707</td>\n",
       "      <td>-8.834</td>\n",
       "      <td>-26.989</td>\n",
       "      <td>-15.247</td>\n",
       "      <td>-13.452</td>\n",
       "      <td>-9.223</td>\n",
       "      <td>-9.010</td>\n",
       "      <td>1.725</td>\n",
       "      <td>...</td>\n",
       "      <td>-15.303</td>\n",
       "      <td>-13.111</td>\n",
       "      <td>-34.422</td>\n",
       "      <td>-27.668</td>\n",
       "      <td>-11.176</td>\n",
       "      <td>-18.758</td>\n",
       "      <td>-16.456</td>\n",
       "      <td>-15.229</td>\n",
       "      <td>-23.869</td>\n",
       "      <td>-9.153</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 580 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ACH-000004  ACH-000007  ACH-000009  ACH-000011  ACH-000012  ACH-000013  \\\n",
       "GENE                                                                           \n",
       "A1BG     -13.689     -12.704     -27.260     -16.002     -22.922     -23.405   \n",
       "A1CF     -12.656      -9.979     -18.618     -12.640     -38.198     -17.486   \n",
       "A2M       -6.802     -10.975     -23.707      -8.834     -26.989     -15.247   \n",
       "\n",
       "      ACH-000014  ACH-000015  ACH-000017  ACH-000018  ...  ACH-001849  \\\n",
       "GENE                                                  ...               \n",
       "A1BG     -14.482     -13.137     -14.341      -7.315  ...     -23.337   \n",
       "A1CF     -12.782     -23.265     -19.222     -11.960  ...     -25.842   \n",
       "A2M      -13.452      -9.223      -9.010       1.725  ...     -15.303   \n",
       "\n",
       "      ACH-001955  ACH-001956  ACH-001957  ACH-001959  ACH-001960  ACH-002001  \\\n",
       "GENE                                                                           \n",
       "A1BG     -26.547     -17.751     -25.570     -21.672     -24.316     -13.376   \n",
       "A1CF     -21.578     -36.392     -30.543     -32.026     -20.854     -24.389   \n",
       "A2M      -13.111     -34.422     -27.668     -11.176     -18.758     -16.456   \n",
       "\n",
       "      ACH-002002  ACH-002004  ACH-002005  \n",
       "GENE                                      \n",
       "A1BG     -19.421     -30.341     -13.039  \n",
       "A1CF     -25.703     -26.582      -9.095  \n",
       "A2M      -15.229     -23.869      -9.153  \n",
       "\n",
       "[3 rows x 580 columns]"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bfdata_old = pd.read_csv(\"/home/ekim8/export/project4/data/avana_2019Q4/CRISPRcleanR/table_Avana2019Q4_CRISPRcleanR_corrected_fmeasure80\",sep=\"\\t\",index_col=0,header=0)\n",
    "bfdata_old.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(18093, 580)"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bfdata_old.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(17852, 739)"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bfdata_single_CCDS.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['AAED1', 'AARS', 'ACPP', 'ADPRHL2', 'ADSS', 'ADSSL1', 'AES',\n",
       "       'ALS2CR12', 'APOPT1', 'ARSE', 'ASNA1', 'ATP5S', 'C11orf74',\n",
       "       'C15orf53', 'C16orf45', 'C17orf51', 'C17orf53', 'C19orf24',\n",
       "       'C19orf66', 'C19orf70', 'C1orf123', 'C21orf2', 'C2orf40',\n",
       "       'C2orf54', 'C2orf70', 'C3orf30', 'C3orf58', 'C5orf30', 'C6orf10',\n",
       "       'C6orf106', 'C6orf203', 'C6orf222', 'C6orf48', 'C7orf43',\n",
       "       'C8orf59', 'C9orf3', 'C9orf66', 'C9orf84', 'CARS', 'CASC10',\n",
       "       'CASC4', 'CCDC129', 'CLLU1OS', 'COL4A3BP', 'CTGF', 'CXorf36',\n",
       "       'CXorf40A', 'CXorf40B', 'CXorf57', 'CYR61', 'DARS', 'DEC1',\n",
       "       'DIRC2', 'DOPEY1', 'DOPEY2', 'ELMSAN1', 'EPRS', 'FAM129A',\n",
       "       'FAM129B', 'FAM129C', 'FAM173A', 'FAM173B', 'FAM192A', 'FAM198A',\n",
       "       'FAM198B', 'FAM19A1', 'FAM19A2', 'FAM19A3', 'FAM19A4', 'FAM19A5',\n",
       "       'FAM206A', 'FAM208A', 'FAM208B', 'FAM213A', 'FAM45A', 'FAM49A',\n",
       "       'FAM49B', 'FAM57A', 'FAM57B', 'FAM69A', 'FAM69B', 'FAM69C',\n",
       "       'FAM84A', 'FAM84B', 'FAM92A', 'FAM92B', 'FGFR1OP', 'FOPNL', 'FUK',\n",
       "       'GARS', 'GIF', 'H1F0', 'H1FNT', 'H1FOO', 'H1FX', 'H2AFJ', 'H2AFV',\n",
       "       'H2AFX', 'H2AFY', 'H2AFY2', 'H2AFZ', 'H2BFM', 'H2BFWT', 'H3F3A',\n",
       "       'H3F3B', 'H3F3C', 'HARS', 'HEXDC', 'HIST1H1A', 'HIST1H1B',\n",
       "       'HIST1H1C', 'HIST1H1D', 'HIST1H1E', 'HIST1H1T', 'HIST1H2AA',\n",
       "       'HIST1H2AB', 'HIST1H2AC', 'HIST1H2AD', 'HIST1H2AE', 'HIST1H2AG',\n",
       "       'HIST1H2AH', 'HIST1H2AI', 'HIST1H2AJ', 'HIST1H2AK', 'HIST1H2AL',\n",
       "       'HIST1H2AM', 'HIST1H2BA', 'HIST1H2BB', 'HIST1H2BC', 'HIST1H2BD',\n",
       "       'HIST1H2BE', 'HIST1H2BF', 'HIST1H2BG', 'HIST1H2BH', 'HIST1H2BI',\n",
       "       'HIST1H2BJ', 'HIST1H2BK', 'HIST1H2BL', 'HIST1H2BM', 'HIST1H2BN',\n",
       "       'HIST1H2BO', 'HIST1H3A', 'HIST1H3B', 'HIST1H3C', 'HIST1H3D',\n",
       "       'HIST1H3E', 'HIST1H3F', 'HIST1H3G', 'HIST1H3H', 'HIST1H3I',\n",
       "       'HIST1H3J', 'HIST1H4A', 'HIST1H4B', 'HIST1H4C', 'HIST1H4D',\n",
       "       'HIST1H4E', 'HIST1H4F', 'HIST1H4G', 'HIST1H4H', 'HIST1H4I',\n",
       "       'HIST1H4L', 'HIST2H2AB', 'HIST2H2AC', 'HIST2H2BE', 'HIST2H2BF',\n",
       "       'HIST2H3D', 'HIST3H2A', 'HIST3H2BB', 'HIST3H3', 'HIST4H4', 'HKR1',\n",
       "       'HRASLS', 'HRASLS2', 'HRASLS5', 'HYPM', 'IARS', 'ICK', 'IMPAD1',\n",
       "       'INSYN2', 'KARS', 'KDELC1', 'KDELC2', 'KIAA0391', 'KIAA1024',\n",
       "       'KIAA1024L', 'KIAA1147', 'KIAA1211', 'KIAA1551', 'KIF1BP', 'LARS',\n",
       "       'LOR', 'MARC1', 'MARC2', 'MARCH1', 'MARCH10', 'MARCH11', 'MARCH2',\n",
       "       'MARCH3', 'MARCH4', 'MARCH5', 'MARCH6', 'MARCH7', 'MARCH8',\n",
       "       'MARCH9', 'MARS', 'METTL13', 'MINOS1', 'MTSS1L', 'MUM1', 'MUM1L1',\n",
       "       'MUT', 'NARS', 'NOV', 'NUPL2', 'PIH1D3', 'PLA2G16', 'PQLC1',\n",
       "       'PQLC2', 'PQLC2L', 'PQLC3', 'PRSS42', 'PRSS45', 'QARS', 'RARRES3',\n",
       "       'RARS', 'RNF219', 'SARS', 'SEPT1', 'SEPT10', 'SEPT11', 'SEPT12',\n",
       "       'SEPT14', 'SEPT2', 'SEPT3', 'SEPT4', 'SEPT5', 'SEPT6', 'SEPT8',\n",
       "       'SEPT9', 'SPERT', 'SSFA2', 'SSSCA1', 'ST5', 'TARS', 'TARSL2',\n",
       "       'TLDC1', 'TMEM136', 'TMEM173', 'TMEM206', 'TMEM246', 'TMEM56',\n",
       "       'TMEM8A', 'TROVE2', 'VARS', 'WARS', 'WISP1', 'WISP2', 'WISP3',\n",
       "       'WRB', 'YARS'], dtype=object)"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "setdiff1d(bfdata_old.index,bfdata_single_CCDS.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1919</td>\n",
       "      <td>AATGATCAGCCTGTTCCCCA</td>\n",
       "      <td>chr17_42961295_+</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23765</td>\n",
       "      <td>CGTGGTCTCCTGCTGTCCCG</td>\n",
       "      <td>chr17_42964209_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28453</td>\n",
       "      <td>GAACATATCTGAAGGCCCCA</td>\n",
       "      <td>chr6_44306300_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40426</td>\n",
       "      <td>GCGTGACAGTTATGCCCGAG</td>\n",
       "      <td>chr17_42964406_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42552</td>\n",
       "      <td>GCTGAGAGTGACTCGCCGTG</td>\n",
       "      <td>chr17_42961299_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48909</td>\n",
       "      <td>GGATGTGGTCAGCCACCACG</td>\n",
       "      <td>chr6_44307320_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50739</td>\n",
       "      <td>GGGCGTATCGGACAGCTCGG</td>\n",
       "      <td>chr16_70268352_+</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55325</td>\n",
       "      <td>GTAGGCCATGTCAATCCCAT</td>\n",
       "      <td>chr16_70269711_+</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62024</td>\n",
       "      <td>TCAGCTCTACCATGTCCAAG</td>\n",
       "      <td>chr6_44305737_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64824</td>\n",
       "      <td>TCTGGCTGTTGGCCACACGT</td>\n",
       "      <td>chr6_44312204_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66140</td>\n",
       "      <td>TGACACCTATGGGTTTCCAG</td>\n",
       "      <td>chr16_70265623_-</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72076</td>\n",
       "      <td>TTATGTTACTTACTTTGGCG</td>\n",
       "      <td>chr16_70276549_-</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment            gene  n_alignments  \\\n",
       "1919   AATGATCAGCCTGTTCCCCA  chr17_42961295_+  AARSD1 (80755)             1   \n",
       "23765  CGTGGTCTCCTGCTGTCCCG  chr17_42964209_-  AARSD1 (80755)             1   \n",
       "28453  GAACATATCTGAAGGCCCCA   chr6_44306300_+   AARS2 (57505)             1   \n",
       "40426  GCGTGACAGTTATGCCCGAG  chr17_42964406_-  AARSD1 (80755)             1   \n",
       "42552  GCTGAGAGTGACTCGCCGTG  chr17_42961299_-  AARSD1 (80755)             1   \n",
       "48909  GGATGTGGTCAGCCACCACG   chr6_44307320_+   AARS2 (57505)             1   \n",
       "50739  GGGCGTATCGGACAGCTCGG  chr16_70268352_+       AARS (16)             1   \n",
       "55325  GTAGGCCATGTCAATCCCAT  chr16_70269711_+       AARS (16)             1   \n",
       "62024  TCAGCTCTACCATGTCCAAG   chr6_44305737_+   AARS2 (57505)             1   \n",
       "64824  TCTGGCTGTTGGCCACACGT   chr6_44312204_+   AARS2 (57505)             1   \n",
       "66140  TGACACCTATGGGTTTCCAG  chr16_70265623_-       AARS (16)             1   \n",
       "72076  TTATGTTACTTACTTTGGCG  chr16_70276549_-       AARS (16)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "1919           AARSD1      80755  \n",
       "23765          AARSD1      80755  \n",
       "28453           AARS2      57505  \n",
       "40426          AARSD1      80755  \n",
       "42552          AARSD1      80755  \n",
       "48909           AARS2      57505  \n",
       "50739           AARS1         16  \n",
       "55325           AARS1         16  \n",
       "62024           AARS2      57505  \n",
       "64824           AARS2      57505  \n",
       "66140           AARS1         16  \n",
       "72076           AARS1         16  "
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap[guidegenemap['gene'].str.contains('AARS')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1919</td>\n",
       "      <td>AATGATCAGCCTGTTCCCCA</td>\n",
       "      <td>chr17_42961295_+</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23765</td>\n",
       "      <td>CGTGGTCTCCTGCTGTCCCG</td>\n",
       "      <td>chr17_42964209_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28453</td>\n",
       "      <td>GAACATATCTGAAGGCCCCA</td>\n",
       "      <td>chr6_44306300_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40426</td>\n",
       "      <td>GCGTGACAGTTATGCCCGAG</td>\n",
       "      <td>chr17_42964406_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42552</td>\n",
       "      <td>GCTGAGAGTGACTCGCCGTG</td>\n",
       "      <td>chr17_42961299_-</td>\n",
       "      <td>AARSD1 (80755)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARSD1</td>\n",
       "      <td>80755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>48909</td>\n",
       "      <td>GGATGTGGTCAGCCACCACG</td>\n",
       "      <td>chr6_44307320_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>50739</td>\n",
       "      <td>GGGCGTATCGGACAGCTCGG</td>\n",
       "      <td>chr16_70268352_+</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55325</td>\n",
       "      <td>GTAGGCCATGTCAATCCCAT</td>\n",
       "      <td>chr16_70269711_+</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62024</td>\n",
       "      <td>TCAGCTCTACCATGTCCAAG</td>\n",
       "      <td>chr6_44305737_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64824</td>\n",
       "      <td>TCTGGCTGTTGGCCACACGT</td>\n",
       "      <td>chr6_44312204_+</td>\n",
       "      <td>AARS2 (57505)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS2</td>\n",
       "      <td>57505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>66140</td>\n",
       "      <td>TGACACCTATGGGTTTCCAG</td>\n",
       "      <td>chr16_70265623_-</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72076</td>\n",
       "      <td>TTATGTTACTTACTTTGGCG</td>\n",
       "      <td>chr16_70276549_-</td>\n",
       "      <td>AARS (16)</td>\n",
       "      <td>1</td>\n",
       "      <td>AARS1</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment            gene  n_alignments  \\\n",
       "1919   AATGATCAGCCTGTTCCCCA  chr17_42961295_+  AARSD1 (80755)             1   \n",
       "23765  CGTGGTCTCCTGCTGTCCCG  chr17_42964209_-  AARSD1 (80755)             1   \n",
       "28453  GAACATATCTGAAGGCCCCA   chr6_44306300_+   AARS2 (57505)             1   \n",
       "40426  GCGTGACAGTTATGCCCGAG  chr17_42964406_-  AARSD1 (80755)             1   \n",
       "42552  GCTGAGAGTGACTCGCCGTG  chr17_42961299_-  AARSD1 (80755)             1   \n",
       "48909  GGATGTGGTCAGCCACCACG   chr6_44307320_+   AARS2 (57505)             1   \n",
       "50739  GGGCGTATCGGACAGCTCGG  chr16_70268352_+       AARS (16)             1   \n",
       "55325  GTAGGCCATGTCAATCCCAT  chr16_70269711_+       AARS (16)             1   \n",
       "62024  TCAGCTCTACCATGTCCAAG   chr6_44305737_+   AARS2 (57505)             1   \n",
       "64824  TCTGGCTGTTGGCCACACGT   chr6_44312204_+   AARS2 (57505)             1   \n",
       "66140  TGACACCTATGGGTTTCCAG  chr16_70265623_-       AARS (16)             1   \n",
       "72076  TTATGTTACTTACTTTGGCG  chr16_70276549_-       AARS (16)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "1919           AARSD1      80755  \n",
       "23765          AARSD1      80755  \n",
       "28453           AARS2      57505  \n",
       "40426          AARSD1      80755  \n",
       "42552          AARSD1      80755  \n",
       "48909           AARS2      57505  \n",
       "50739           AARS1         16  \n",
       "55325           AARS1         16  \n",
       "62024           AARS2      57505  \n",
       "64824           AARS2      57505  \n",
       "66140           AARS1         16  \n",
       "72076           AARS1         16  "
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['gene'].str.contains('AARS')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(18093, 580)"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bfdata_old.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18111"
      ]
     },
     "execution_count": 188,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(guidegenemap_single_CCDS['official_symbol'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [sgrna, genome_alignment, gene, n_alignments, official_symbol, entrez_id]\n",
       "Index: []"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap[guidegenemap['official_symbol'].isin(setdiff1d(bfdata_old.index,guidegenemap_single_CCDS['official_symbol'].unique()))]\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>70</td>\n",
       "      <td>AAAATCGTCCCATTCCCCAG</td>\n",
       "      <td>chrX_119789942_-</td>\n",
       "      <td>RPL39 (6170)</td>\n",
       "      <td>11</td>\n",
       "      <td>RPL39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>820</td>\n",
       "      <td>AAGACGTCATTACTCTACGG</td>\n",
       "      <td>chr16_28348943_-</td>\n",
       "      <td>NPIPB6 (728741)</td>\n",
       "      <td>7</td>\n",
       "      <td>NPIPB6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>942</td>\n",
       "      <td>AAGATAGAGGAGGAACAAGG</td>\n",
       "      <td>chr17_36169971_+</td>\n",
       "      <td>TBC1D3B (414059)</td>\n",
       "      <td>11</td>\n",
       "      <td>TBC1D3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1905</td>\n",
       "      <td>AATGAGATCCTACAGGCACG</td>\n",
       "      <td>chr16_28391775_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>2</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1978</td>\n",
       "      <td>AATGCCTGGACACTGCACAG</td>\n",
       "      <td>chr19_20934092_+</td>\n",
       "      <td>ZNF85 (7639)</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2297</td>\n",
       "      <td>AATTCTCTTATGCTTAGTAA</td>\n",
       "      <td>chr4_272060_+</td>\n",
       "      <td>ZNF732 (654254)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2968</td>\n",
       "      <td>ACACGGGTCTCACATGAGAG</td>\n",
       "      <td>chr4_9210785_-</td>\n",
       "      <td>USP17L10 (100287144)</td>\n",
       "      <td>24</td>\n",
       "      <td>USP17L10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2992</td>\n",
       "      <td>ACACGGGTCTCACATGAGAG</td>\n",
       "      <td>chr4_9339531_-</td>\n",
       "      <td>USP17L5 (728386)</td>\n",
       "      <td>24</td>\n",
       "      <td>USP17L5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3511</td>\n",
       "      <td>ACAGGAGCCACAGTCCCCCT</td>\n",
       "      <td>chr11_71538257_-</td>\n",
       "      <td>KRTAP5-8 (57830)</td>\n",
       "      <td>1</td>\n",
       "      <td>KRTAP5-8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4174</td>\n",
       "      <td>ACCATTTCTAGGCTTCTAGG</td>\n",
       "      <td>chr19_24087322_-</td>\n",
       "      <td>ZNF254 (9534)</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4557</td>\n",
       "      <td>ACGAGAGCTGACCCAGCTAA</td>\n",
       "      <td>chr1_146141771_-</td>\n",
       "      <td>NBPF10 (100132406)</td>\n",
       "      <td>2</td>\n",
       "      <td>NBPF10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5616</td>\n",
       "      <td>ACTGCTGGTTCAGATTGCAG</td>\n",
       "      <td>chr16_28390940_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>2</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7899</td>\n",
       "      <td>AGATGCTGAGAAAAATGAGG</td>\n",
       "      <td>chr16_28400629_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>2</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8205</td>\n",
       "      <td>AGCATGTAGTTGGCAAGTGG</td>\n",
       "      <td>chr4_9210965_-</td>\n",
       "      <td>USP17L10 (100287144)</td>\n",
       "      <td>1</td>\n",
       "      <td>USP17L10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8647</td>\n",
       "      <td>AGCTTTGGGAAGATAAGGAA</td>\n",
       "      <td>chr16_28400969_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>2</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9151</td>\n",
       "      <td>AGGCCTAGAGAACATATATC</td>\n",
       "      <td>chr13_27253813_-</td>\n",
       "      <td>RPL21 (6144)</td>\n",
       "      <td>23</td>\n",
       "      <td>RPL21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10063</td>\n",
       "      <td>AGTCTTCTCACAAGACTTTC</td>\n",
       "      <td>chrX_119789996_-</td>\n",
       "      <td>RPL39 (6170)</td>\n",
       "      <td>2</td>\n",
       "      <td>RPL39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11798</td>\n",
       "      <td>ATAGGCCCAATCATTTCAGG</td>\n",
       "      <td>chrX_49581166_-</td>\n",
       "      <td>GAGE12H (729442)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11802</td>\n",
       "      <td>ATAGGCCCAATCATTTCAGG</td>\n",
       "      <td>chrX_49323256_-</td>\n",
       "      <td>GAGE12J (729396)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11950</td>\n",
       "      <td>ATATAACACTGACATCAGCA</td>\n",
       "      <td>chr3_39408687_-</td>\n",
       "      <td>RPSA (3921)</td>\n",
       "      <td>17</td>\n",
       "      <td>RPSA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12546</td>\n",
       "      <td>ATCTCTATACAAATTCTGCT</td>\n",
       "      <td>chr4_296082_+</td>\n",
       "      <td>ZNF732 (654254)</td>\n",
       "      <td>5</td>\n",
       "      <td>ZNF732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13139</td>\n",
       "      <td>ATGTACAGCAGAGAGCCAGG</td>\n",
       "      <td>chr16_15370047_-</td>\n",
       "      <td>NPIPA5 (100288332)</td>\n",
       "      <td>10</td>\n",
       "      <td>NPIPA5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13165</td>\n",
       "      <td>ATGTCCATCGAATCTTCCAT</td>\n",
       "      <td>chr5_171387951_-</td>\n",
       "      <td>NPM1 (4869)</td>\n",
       "      <td>8</td>\n",
       "      <td>NPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13410</td>\n",
       "      <td>ATTCACTCTCACCTACCTGG</td>\n",
       "      <td>chr19_24106642_-</td>\n",
       "      <td>ZNF254 (9534)</td>\n",
       "      <td>11</td>\n",
       "      <td>ZNF254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14506</td>\n",
       "      <td>CACCAATCTTGACTTCCAGA</td>\n",
       "      <td>chr3_39407750_+</td>\n",
       "      <td>RPSA (3921)</td>\n",
       "      <td>17</td>\n",
       "      <td>RPSA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14548</td>\n",
       "      <td>CACCATTTCTAGGCTTCTAG</td>\n",
       "      <td>chr19_24087323_-</td>\n",
       "      <td>ZNF254 (9534)</td>\n",
       "      <td>2</td>\n",
       "      <td>ZNF254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15568</td>\n",
       "      <td>CAGGAGCCAAATCATCACAG</td>\n",
       "      <td>chr4_9210811_-</td>\n",
       "      <td>USP17L10 (100287144)</td>\n",
       "      <td>28</td>\n",
       "      <td>USP17L10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15576</td>\n",
       "      <td>CAGGAGCCAAATCATCACAG</td>\n",
       "      <td>chr4_9339557_-</td>\n",
       "      <td>USP17L5 (728386)</td>\n",
       "      <td>28</td>\n",
       "      <td>USP17L5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17884</td>\n",
       "      <td>CCACATTCTTCACATGTGAA</td>\n",
       "      <td>chr4_272022_+</td>\n",
       "      <td>ZNF732 (654254)</td>\n",
       "      <td>3</td>\n",
       "      <td>ZNF732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17885</td>\n",
       "      <td>CCACATTCTTCACATGTGAA</td>\n",
       "      <td>chr4_272187_+</td>\n",
       "      <td>ZNF732 (654254)</td>\n",
       "      <td>3</td>\n",
       "      <td>ZNF732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18836</td>\n",
       "      <td>CCAGGGACTGTTGACATTTA</td>\n",
       "      <td>chr19_24105952_+</td>\n",
       "      <td>ZNF254 (9534)</td>\n",
       "      <td>3</td>\n",
       "      <td>ZNF254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21334</td>\n",
       "      <td>CCTGGATGAGGAGGAGGAGG</td>\n",
       "      <td>chr16_28392038_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>1</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21965</td>\n",
       "      <td>CCTTGGTCTAGGCCAATAAT</td>\n",
       "      <td>chrX_49581131_-</td>\n",
       "      <td>GAGE12H (729442)</td>\n",
       "      <td>8</td>\n",
       "      <td>GAGE12H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22154</td>\n",
       "      <td>CGAACTGCAGAAGATCCACA</td>\n",
       "      <td>chr9_67859538_+</td>\n",
       "      <td>ANKRD20A1 (84210)</td>\n",
       "      <td>9</td>\n",
       "      <td>ANKRD20A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22776</td>\n",
       "      <td>CGCCAAAATGACGAACACAA</td>\n",
       "      <td>chr13_27253786_+</td>\n",
       "      <td>RPL21 (6144)</td>\n",
       "      <td>23</td>\n",
       "      <td>RPL21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23449</td>\n",
       "      <td>CGTCATCGTGAGACTGACTT</td>\n",
       "      <td>chr16_28353020_-</td>\n",
       "      <td>NPIPB6 (728741)</td>\n",
       "      <td>7</td>\n",
       "      <td>NPIPB6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23474</td>\n",
       "      <td>CGTCATTCGAAACATAGTGG</td>\n",
       "      <td>chr12_56042554_+</td>\n",
       "      <td>RPS26 (6231)</td>\n",
       "      <td>15</td>\n",
       "      <td>RPS26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23633</td>\n",
       "      <td>CGTGAGACTGACTTTGGTGT</td>\n",
       "      <td>chr16_28353014_-</td>\n",
       "      <td>NPIPB6 (728741)</td>\n",
       "      <td>6</td>\n",
       "      <td>NPIPB6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24510</td>\n",
       "      <td>CTACCTGTGCTGCTTGTCCA</td>\n",
       "      <td>chr9_67859616_-</td>\n",
       "      <td>ANKRD20A1 (84210)</td>\n",
       "      <td>9</td>\n",
       "      <td>ANKRD20A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28323</td>\n",
       "      <td>GAAAGTCTTGTGAGAAGACT</td>\n",
       "      <td>chrX_119790010_+</td>\n",
       "      <td>RPL39 (6170)</td>\n",
       "      <td>1</td>\n",
       "      <td>RPL39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28869</td>\n",
       "      <td>GAATAGGCTGCACGTGGCCG</td>\n",
       "      <td>chr12_56042468_-</td>\n",
       "      <td>RPS26 (6231)</td>\n",
       "      <td>13</td>\n",
       "      <td>RPS26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30694</td>\n",
       "      <td>GAGGCTGGTTGGAGTCACAA</td>\n",
       "      <td>chr1_146139933_+</td>\n",
       "      <td>NBPF10 (100132406)</td>\n",
       "      <td>9</td>\n",
       "      <td>NBPF10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30802</td>\n",
       "      <td>GAGTATTCAGTGTTACCTGG</td>\n",
       "      <td>chr1_146134193_+</td>\n",
       "      <td>NBPF10 (100132406)</td>\n",
       "      <td>22</td>\n",
       "      <td>NBPF10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30806</td>\n",
       "      <td>GAGTATTCAGTGTTACCTGG</td>\n",
       "      <td>chr1_146140484_+</td>\n",
       "      <td>NBPF10 (100132406)</td>\n",
       "      <td>22</td>\n",
       "      <td>NBPF10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31557</td>\n",
       "      <td>GATCCTGCAGCTGCTCAGGA</td>\n",
       "      <td>chrX_49323828_+</td>\n",
       "      <td>GAGE12J (729396)</td>\n",
       "      <td>11</td>\n",
       "      <td>GAGE12J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31631</td>\n",
       "      <td>GATCCTGCAGCTGCTCAGAA</td>\n",
       "      <td>chrX_49581738_+</td>\n",
       "      <td>GAGE12H (729442)</td>\n",
       "      <td>1</td>\n",
       "      <td>GAGE12H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31959</td>\n",
       "      <td>GATGAGCCGGACAAGTCCCA</td>\n",
       "      <td>chr1_146141680_-</td>\n",
       "      <td>NBPF10 (100132406)</td>\n",
       "      <td>17</td>\n",
       "      <td>NBPF10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31995</td>\n",
       "      <td>GATGAGGATGAGGATGAGGA</td>\n",
       "      <td>chr16_28400461_-</td>\n",
       "      <td>EIF3CL (728689)</td>\n",
       "      <td>6</td>\n",
       "      <td>EIF3CL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34140</td>\n",
       "      <td>GCACAGTTAGTGCAGCGAAT</td>\n",
       "      <td>chr12_56042484_-</td>\n",
       "      <td>RPS26 (6231)</td>\n",
       "      <td>13</td>\n",
       "      <td>RPS26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34329</td>\n",
       "      <td>GCACCAGTTATCTTTAAGAA</td>\n",
       "      <td>chr5_171390125_+</td>\n",
       "      <td>NPM1 (4869)</td>\n",
       "      <td>8</td>\n",
       "      <td>NPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35257</td>\n",
       "      <td>GCAGAGGCAATGAATTACGA</td>\n",
       "      <td>chr5_171391366_+</td>\n",
       "      <td>NPM1 (4869)</td>\n",
       "      <td>7</td>\n",
       "      <td>NPM1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36338</td>\n",
       "      <td>GCAGGAGGTGGGCTACTGCA</td>\n",
       "      <td>chr17_36169998_-</td>\n",
       "      <td>TBC1D3B (414059)</td>\n",
       "      <td>18</td>\n",
       "      <td>TBC1D3B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36678</td>\n",
       "      <td>GCAGGTCTCCTGCTACTCAG</td>\n",
       "      <td>chr4_9339612_-</td>\n",
       "      <td>USP17L5 (728386)</td>\n",
       "      <td>30</td>\n",
       "      <td>USP17L5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36691</td>\n",
       "      <td>GCAGGTCTCCTGCTACTCAG</td>\n",
       "      <td>chr4_9210866_-</td>\n",
       "      <td>USP17L10 (100287144)</td>\n",
       "      <td>30</td>\n",
       "      <td>USP17L10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37165</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49323258_-</td>\n",
       "      <td>GAGE12J (729396)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37172</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49581168_-</td>\n",
       "      <td>GAGE12H (729442)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37851</td>\n",
       "      <td>GCCAAAATGACGAACACAAA</td>\n",
       "      <td>chr13_27253787_+</td>\n",
       "      <td>RPL21 (6144)</td>\n",
       "      <td>28</td>\n",
       "      <td>RPL21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38003</td>\n",
       "      <td>GCCACATCCCTAAATGTCAA</td>\n",
       "      <td>chr19_20934032_-</td>\n",
       "      <td>ZNF85 (7639)</td>\n",
       "      <td>5</td>\n",
       "      <td>ZNF85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38342</td>\n",
       "      <td>GCCATGTTCCAGCAGAATAA</td>\n",
       "      <td>chr9_67863251_-</td>\n",
       "      <td>ANKRD20A1 (84210)</td>\n",
       "      <td>6</td>\n",
       "      <td>ANKRD20A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39478</td>\n",
       "      <td>GCGAAGCCCGAAGAACTCAA</td>\n",
       "      <td>chr4_9340139_+</td>\n",
       "      <td>USP17L5 (728386)</td>\n",
       "      <td>1</td>\n",
       "      <td>USP17L5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment                  gene  \\\n",
       "70     AAAATCGTCCCATTCCCCAG  chrX_119789942_-          RPL39 (6170)   \n",
       "820    AAGACGTCATTACTCTACGG  chr16_28348943_-       NPIPB6 (728741)   \n",
       "942    AAGATAGAGGAGGAACAAGG  chr17_36169971_+      TBC1D3B (414059)   \n",
       "1905   AATGAGATCCTACAGGCACG  chr16_28391775_-       EIF3CL (728689)   \n",
       "1978   AATGCCTGGACACTGCACAG  chr19_20934092_+          ZNF85 (7639)   \n",
       "2297   AATTCTCTTATGCTTAGTAA     chr4_272060_+       ZNF732 (654254)   \n",
       "2968   ACACGGGTCTCACATGAGAG    chr4_9210785_-  USP17L10 (100287144)   \n",
       "2992   ACACGGGTCTCACATGAGAG    chr4_9339531_-      USP17L5 (728386)   \n",
       "3511   ACAGGAGCCACAGTCCCCCT  chr11_71538257_-      KRTAP5-8 (57830)   \n",
       "4174   ACCATTTCTAGGCTTCTAGG  chr19_24087322_-         ZNF254 (9534)   \n",
       "4557   ACGAGAGCTGACCCAGCTAA  chr1_146141771_-    NBPF10 (100132406)   \n",
       "5616   ACTGCTGGTTCAGATTGCAG  chr16_28390940_-       EIF3CL (728689)   \n",
       "7899   AGATGCTGAGAAAAATGAGG  chr16_28400629_-       EIF3CL (728689)   \n",
       "8205   AGCATGTAGTTGGCAAGTGG    chr4_9210965_-  USP17L10 (100287144)   \n",
       "8647   AGCTTTGGGAAGATAAGGAA  chr16_28400969_-       EIF3CL (728689)   \n",
       "9151   AGGCCTAGAGAACATATATC  chr13_27253813_-          RPL21 (6144)   \n",
       "10063  AGTCTTCTCACAAGACTTTC  chrX_119789996_-          RPL39 (6170)   \n",
       "11798  ATAGGCCCAATCATTTCAGG   chrX_49581166_-      GAGE12H (729442)   \n",
       "11802  ATAGGCCCAATCATTTCAGG   chrX_49323256_-      GAGE12J (729396)   \n",
       "11950  ATATAACACTGACATCAGCA   chr3_39408687_-           RPSA (3921)   \n",
       "12546  ATCTCTATACAAATTCTGCT     chr4_296082_+       ZNF732 (654254)   \n",
       "13139  ATGTACAGCAGAGAGCCAGG  chr16_15370047_-    NPIPA5 (100288332)   \n",
       "13165  ATGTCCATCGAATCTTCCAT  chr5_171387951_-           NPM1 (4869)   \n",
       "13410  ATTCACTCTCACCTACCTGG  chr19_24106642_-         ZNF254 (9534)   \n",
       "14506  CACCAATCTTGACTTCCAGA   chr3_39407750_+           RPSA (3921)   \n",
       "14548  CACCATTTCTAGGCTTCTAG  chr19_24087323_-         ZNF254 (9534)   \n",
       "15568  CAGGAGCCAAATCATCACAG    chr4_9210811_-  USP17L10 (100287144)   \n",
       "15576  CAGGAGCCAAATCATCACAG    chr4_9339557_-      USP17L5 (728386)   \n",
       "17884  CCACATTCTTCACATGTGAA     chr4_272022_+       ZNF732 (654254)   \n",
       "17885  CCACATTCTTCACATGTGAA     chr4_272187_+       ZNF732 (654254)   \n",
       "18836  CCAGGGACTGTTGACATTTA  chr19_24105952_+         ZNF254 (9534)   \n",
       "21334  CCTGGATGAGGAGGAGGAGG  chr16_28392038_-       EIF3CL (728689)   \n",
       "21965  CCTTGGTCTAGGCCAATAAT   chrX_49581131_-      GAGE12H (729442)   \n",
       "22154  CGAACTGCAGAAGATCCACA   chr9_67859538_+     ANKRD20A1 (84210)   \n",
       "22776  CGCCAAAATGACGAACACAA  chr13_27253786_+          RPL21 (6144)   \n",
       "23449  CGTCATCGTGAGACTGACTT  chr16_28353020_-       NPIPB6 (728741)   \n",
       "23474  CGTCATTCGAAACATAGTGG  chr12_56042554_+          RPS26 (6231)   \n",
       "23633  CGTGAGACTGACTTTGGTGT  chr16_28353014_-       NPIPB6 (728741)   \n",
       "24510  CTACCTGTGCTGCTTGTCCA   chr9_67859616_-     ANKRD20A1 (84210)   \n",
       "28323  GAAAGTCTTGTGAGAAGACT  chrX_119790010_+          RPL39 (6170)   \n",
       "28869  GAATAGGCTGCACGTGGCCG  chr12_56042468_-          RPS26 (6231)   \n",
       "30694  GAGGCTGGTTGGAGTCACAA  chr1_146139933_+    NBPF10 (100132406)   \n",
       "30802  GAGTATTCAGTGTTACCTGG  chr1_146134193_+    NBPF10 (100132406)   \n",
       "30806  GAGTATTCAGTGTTACCTGG  chr1_146140484_+    NBPF10 (100132406)   \n",
       "31557  GATCCTGCAGCTGCTCAGGA   chrX_49323828_+      GAGE12J (729396)   \n",
       "31631  GATCCTGCAGCTGCTCAGAA   chrX_49581738_+      GAGE12H (729442)   \n",
       "31959  GATGAGCCGGACAAGTCCCA  chr1_146141680_-    NBPF10 (100132406)   \n",
       "31995  GATGAGGATGAGGATGAGGA  chr16_28400461_-       EIF3CL (728689)   \n",
       "34140  GCACAGTTAGTGCAGCGAAT  chr12_56042484_-          RPS26 (6231)   \n",
       "34329  GCACCAGTTATCTTTAAGAA  chr5_171390125_+           NPM1 (4869)   \n",
       "35257  GCAGAGGCAATGAATTACGA  chr5_171391366_+           NPM1 (4869)   \n",
       "36338  GCAGGAGGTGGGCTACTGCA  chr17_36169998_-      TBC1D3B (414059)   \n",
       "36678  GCAGGTCTCCTGCTACTCAG    chr4_9339612_-      USP17L5 (728386)   \n",
       "36691  GCAGGTCTCCTGCTACTCAG    chr4_9210866_-  USP17L10 (100287144)   \n",
       "37165  GCATAGGCCCAATCATTTCA   chrX_49323258_-      GAGE12J (729396)   \n",
       "37172  GCATAGGCCCAATCATTTCA   chrX_49581168_-      GAGE12H (729442)   \n",
       "37851  GCCAAAATGACGAACACAAA  chr13_27253787_+          RPL21 (6144)   \n",
       "38003  GCCACATCCCTAAATGTCAA  chr19_20934032_-          ZNF85 (7639)   \n",
       "38342  GCCATGTTCCAGCAGAATAA   chr9_67863251_-     ANKRD20A1 (84210)   \n",
       "39478  GCGAAGCCCGAAGAACTCAA    chr4_9340139_+      USP17L5 (728386)   \n",
       "\n",
       "       n_alignments official_symbol  \n",
       "70               11           RPL39  \n",
       "820               7          NPIPB6  \n",
       "942              11         TBC1D3B  \n",
       "1905              2          EIF3CL  \n",
       "1978              2           ZNF85  \n",
       "2297              1          ZNF732  \n",
       "2968             24        USP17L10  \n",
       "2992             24         USP17L5  \n",
       "3511              1        KRTAP5-8  \n",
       "4174              2          ZNF254  \n",
       "4557              2          NBPF10  \n",
       "5616              2          EIF3CL  \n",
       "7899              2          EIF3CL  \n",
       "8205              1        USP17L10  \n",
       "8647              2          EIF3CL  \n",
       "9151             23           RPL21  \n",
       "10063             2           RPL39  \n",
       "11798            12         GAGE12H  \n",
       "11802            12         GAGE12J  \n",
       "11950            17            RPSA  \n",
       "12546             5          ZNF732  \n",
       "13139            10          NPIPA5  \n",
       "13165             8            NPM1  \n",
       "13410            11          ZNF254  \n",
       "14506            17            RPSA  \n",
       "14548             2          ZNF254  \n",
       "15568            28        USP17L10  \n",
       "15576            28         USP17L5  \n",
       "17884             3          ZNF732  \n",
       "17885             3          ZNF732  \n",
       "18836             3          ZNF254  \n",
       "21334             1          EIF3CL  \n",
       "21965             8         GAGE12H  \n",
       "22154             9       ANKRD20A1  \n",
       "22776            23           RPL21  \n",
       "23449             7          NPIPB6  \n",
       "23474            15           RPS26  \n",
       "23633             6          NPIPB6  \n",
       "24510             9       ANKRD20A1  \n",
       "28323             1           RPL39  \n",
       "28869            13           RPS26  \n",
       "30694             9          NBPF10  \n",
       "30802            22          NBPF10  \n",
       "30806            22          NBPF10  \n",
       "31557            11         GAGE12J  \n",
       "31631             1         GAGE12H  \n",
       "31959            17          NBPF10  \n",
       "31995             6          EIF3CL  \n",
       "34140            13           RPS26  \n",
       "34329             8            NPM1  \n",
       "35257             7            NPM1  \n",
       "36338            18         TBC1D3B  \n",
       "36678            30         USP17L5  \n",
       "36691            30        USP17L10  \n",
       "37165            12         GAGE12J  \n",
       "37172            12         GAGE12H  \n",
       "37851            28           RPL21  \n",
       "38003             5           ZNF85  \n",
       "38342             6       ANKRD20A1  \n",
       "39478             1         USP17L5  "
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap[guidegenemap['official_symbol'].isin(setdiff1d(bfdata_single_CCDS.index,bfdata_old.index))].head(60)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>24255</td>\n",
       "      <td>CTACAGCCAGAAGAGATATG</td>\n",
       "      <td>chr6_73519053_-</td>\n",
       "      <td>EEF1A1 (1915)</td>\n",
       "      <td>8</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33937</td>\n",
       "      <td>GCACAAATGCTACTGTGTCG</td>\n",
       "      <td>chr6_73518995_+</td>\n",
       "      <td>EEF1A1 (1915)</td>\n",
       "      <td>6</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>44135</td>\n",
       "      <td>GCTGGTCTCAAATTTCCACA</td>\n",
       "      <td>chr6_73519428_+</td>\n",
       "      <td>EEF1A1 (1915)</td>\n",
       "      <td>13</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>60469</td>\n",
       "      <td>TATCTCTTCTGGCTGTAGGG</td>\n",
       "      <td>chr6_73519069_+</td>\n",
       "      <td>EEF1A1 (1915)</td>\n",
       "      <td>8</td>\n",
       "      <td>EEF1A1</td>\n",
       "      <td>1915</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna genome_alignment           gene  n_alignments  \\\n",
       "24255  CTACAGCCAGAAGAGATATG  chr6_73519053_-  EEF1A1 (1915)             8   \n",
       "33937  GCACAAATGCTACTGTGTCG  chr6_73518995_+  EEF1A1 (1915)             6   \n",
       "44135  GCTGGTCTCAAATTTCCACA  chr6_73519428_+  EEF1A1 (1915)            13   \n",
       "60469  TATCTCTTCTGGCTGTAGGG  chr6_73519069_+  EEF1A1 (1915)             8   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "24255          EEF1A1       1915  \n",
       "33937          EEF1A1       1915  \n",
       "44135          EEF1A1       1915  \n",
       "60469          EEF1A1       1915  "
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='EEF1A1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CTACAGCCAGAAGAGATATG 8 10\n",
      "GCACAAATGCTACTGTGTCG 6 9\n",
      "GCTGGTCTCAAATTTCCACA 13 6\n",
      "TATCTCTTCTGGCTGTAGGG 8 12\n"
     ]
    }
   ],
   "source": [
    "for sgrna in guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='EEF1A1']['sgrna']:\n",
    "    print(sgrna,\n",
    "          len(str(alignment_summary.loc[sgrna][1]).split(\",\")),\n",
    "          len(str(alignment_summary.loc[sgrna][3]).split(\",\")))\n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>4709</td>\n",
       "      <td>ACGGAGGAGGAGGATGGAGG</td>\n",
       "      <td>chr1_224356928_+</td>\n",
       "      <td>CNIH4 (29097)</td>\n",
       "      <td>1</td>\n",
       "      <td>CNIH4</td>\n",
       "      <td>29097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8972</td>\n",
       "      <td>AGGATACGAAGTAGACCGAG</td>\n",
       "      <td>chr1_224356983_-</td>\n",
       "      <td>CNIH4 (29097)</td>\n",
       "      <td>1</td>\n",
       "      <td>CNIH4</td>\n",
       "      <td>29097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13232</td>\n",
       "      <td>ATGTTGCTCAAAATTAAACA</td>\n",
       "      <td>chr1_224360558_+</td>\n",
       "      <td>CNIH4 (29097)</td>\n",
       "      <td>1</td>\n",
       "      <td>CNIH4</td>\n",
       "      <td>29097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30611</td>\n",
       "      <td>GAGGAGGAGGATGGAGGCGG</td>\n",
       "      <td>chr1_224356931_+</td>\n",
       "      <td>CNIH4 (29097)</td>\n",
       "      <td>1</td>\n",
       "      <td>CNIH4</td>\n",
       "      <td>29097</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment           gene  n_alignments  \\\n",
       "4709   ACGGAGGAGGAGGATGGAGG  chr1_224356928_+  CNIH4 (29097)             1   \n",
       "8972   AGGATACGAAGTAGACCGAG  chr1_224356983_-  CNIH4 (29097)             1   \n",
       "13232  ATGTTGCTCAAAATTAAACA  chr1_224360558_+  CNIH4 (29097)             1   \n",
       "30611  GAGGAGGAGGATGGAGGCGG  chr1_224356931_+  CNIH4 (29097)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "4709            CNIH4      29097  \n",
       "8972            CNIH4      29097  \n",
       "13232           CNIH4      29097  \n",
       "30611           CNIH4      29097  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='CNIH4']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ACGGAGGAGGAGGATGGAGG 1 1\n",
      "AGGATACGAAGTAGACCGAG 1 1\n",
      "ATGTTGCTCAAAATTAAACA 1 1\n",
      "GAGGAGGAGGATGGAGGCGG 1 5\n"
     ]
    }
   ],
   "source": [
    "for sgrna in guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='CNIH4']['sgrna']:\n",
    "    print(sgrna,\n",
    "          len(str(alignment_summary.loc[sgrna][1]).split(\",\")),\n",
    "          len(str(alignment_summary.loc[sgrna][3]).split(\",\")))\n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>8145</td>\n",
       "      <td>AGCAGACTCACTCCAAAGCG</td>\n",
       "      <td>chrX_13662763_+</td>\n",
       "      <td>TCEANC (170082)</td>\n",
       "      <td>1</td>\n",
       "      <td>TCEANC</td>\n",
       "      <td>170082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19538</td>\n",
       "      <td>CCCAGCTACTCGGAAGGCCG</td>\n",
       "      <td>chrX_13659744_-</td>\n",
       "      <td>TCEANC (170082)</td>\n",
       "      <td>35</td>\n",
       "      <td>TCEANC</td>\n",
       "      <td>170082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>73307</td>\n",
       "      <td>TTGTTTCTAGCTCAGTAAGG</td>\n",
       "      <td>chrX_13662597_-</td>\n",
       "      <td>TCEANC (170082)</td>\n",
       "      <td>1</td>\n",
       "      <td>TCEANC</td>\n",
       "      <td>170082</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna genome_alignment             gene  n_alignments  \\\n",
       "8145   AGCAGACTCACTCCAAAGCG  chrX_13662763_+  TCEANC (170082)             1   \n",
       "19538  CCCAGCTACTCGGAAGGCCG  chrX_13659744_-  TCEANC (170082)            35   \n",
       "73307  TTGTTTCTAGCTCAGTAAGG  chrX_13662597_-  TCEANC (170082)             1   \n",
       "\n",
       "      official_symbol  entrez_id  \n",
       "8145           TCEANC     170082  \n",
       "19538          TCEANC     170082  \n",
       "73307          TCEANC     170082  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='TCEANC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AGCAGACTCACTCCAAAGCG 1 1\n",
      "CCCAGCTACTCGGAAGGCCG 34 4071\n",
      "TTGTTTCTAGCTCAGTAAGG 1 1\n"
     ]
    }
   ],
   "source": [
    "for sgrna in guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='TCEANC']['sgrna']:\n",
    "    print(sgrna,\n",
    "          len(str(alignment_summary.loc[sgrna][1]).split(\",\")),\n",
    "          len(str(alignment_summary.loc[sgrna][3]).split(\",\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>GAGATGGTGGAAGAACCCCC</td>\n",
       "      <td>chr19_22193014_-</td>\n",
       "      <td>ZNF676</td>\n",
       "      <td>chr19_24106621_+,chr19_23808792_+,chr19_239200...</td>\n",
       "      <td>ZNF726,AC092279,AC139769,AC011503,ZNF431,ZNF208</td>\n",
       "      <td>chr19_23662112_-,chr19_21530662_+,chr19_224646...</td>\n",
       "      <td>ZNF675,ZNF429,ZNF714,ZNF98,ZNF91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     1       2  \\\n",
       "0                                                \n",
       "GAGATGGTGGAAGAACCCCC  chr19_22193014_-  ZNF676   \n",
       "\n",
       "                                                                      3  \\\n",
       "0                                                                         \n",
       "GAGATGGTGGAAGAACCCCC  chr19_24106621_+,chr19_23808792_+,chr19_239200...   \n",
       "\n",
       "                                                                    4  \\\n",
       "0                                                                       \n",
       "GAGATGGTGGAAGAACCCCC  ZNF726,AC092279,AC139769,AC011503,ZNF431,ZNF208   \n",
       "\n",
       "                                                                      5  \\\n",
       "0                                                                         \n",
       "GAGATGGTGGAAGAACCCCC  chr19_23662112_-,chr19_21530662_+,chr19_224646...   \n",
       "\n",
       "                                                     6  \n",
       "0                                                       \n",
       "GAGATGGTGGAAGAACCCCC  ZNF675,ZNF429,ZNF714,ZNF98,ZNF91  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alignment_summary.loc[guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='ZNF676']['sgrna']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GAGATGGTGGAAGAACCCCC 1 5\n"
     ]
    }
   ],
   "source": [
    "for sgrna in guidegenemap_single_CCDS[guidegenemap_single_CCDS['official_symbol']=='ZNF676']['sgrna']:\n",
    "    print(sgrna,\n",
    "          len(str(alignment_summary.loc[sgrna][1]).split(\",\")),\n",
    "          len(str(alignment_summary.loc[sgrna][3]).split(\",\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# alteranative. multi-protein check - Bowtie align"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964603_+</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr1_173144937_-,chr3_28391460_+</td>\n",
       "      <td>AC098650,ZCWPW2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003598_+</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr17_32870380_-</td>\n",
       "      <td>MYO1D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970350_-</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>chr4_155062892_+,chrX_67580664_-,chr8_26380606...</td>\n",
       "      <td>SDAD1P1,AR</td>\n",
       "      <td>chr6_88829708_+,chr7_158809481_+,chr17_5631315...</td>\n",
       "      <td>PTPN23,SLC25A33,LINC00472,ELK4,AL353604,AC0680...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      1        2  \\\n",
       "0                                                  \n",
       "AAAAAAATCCAGCAATGCAG  chr10_110964603_+    SHOC2   \n",
       "AAAAAACCCGTAGATAGCCT   chr12_95003598_+  NDUFA12   \n",
       "AAAAAAGAAGAAAAAACCAG    chr4_75970350_-    SDAD1   \n",
       "\n",
       "                                                                      3  \\\n",
       "0                                                                         \n",
       "AAAAAAATCCAGCAATGCAG                                                NaN   \n",
       "AAAAAACCCGTAGATAGCCT                                                NaN   \n",
       "AAAAAAGAAGAAAAAACCAG  chr4_155062892_+,chrX_67580664_-,chr8_26380606...   \n",
       "\n",
       "                               4  \\\n",
       "0                                  \n",
       "AAAAAAATCCAGCAATGCAG         NaN   \n",
       "AAAAAACCCGTAGATAGCCT         NaN   \n",
       "AAAAAAGAAGAAAAAACCAG  SDAD1P1,AR   \n",
       "\n",
       "                                                                      5  \\\n",
       "0                                                                         \n",
       "AAAAAAATCCAGCAATGCAG                   chr1_173144937_-,chr3_28391460_+   \n",
       "AAAAAACCCGTAGATAGCCT                                   chr17_32870380_-   \n",
       "AAAAAAGAAGAAAAAACCAG  chr6_88829708_+,chr7_158809481_+,chr17_5631315...   \n",
       "\n",
       "                                                                      6  \n",
       "0                                                                        \n",
       "AAAAAAATCCAGCAATGCAG                                    AC098650,ZCWPW2  \n",
       "AAAAAACCCGTAGATAGCCT                                              MYO1D  \n",
       "AAAAAAGAAGAAAAAACCAG  PTPN23,SLC25A33,LINC00472,ELK4,AL353604,AC0680...  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read alignment file \n",
    "# alignment file is generated by precalc_library_alignment_info.py\n",
    "'''\n",
    "Usage) python3 precalc_library_alignment_info.py [READID2SEQ] [GENOME (bowtie1 indexed)] [GENECODE gtf] (options)\n",
    "python3 precalc_library_alignment_info.py Avana_readid2seq.tsv hg19 gencode.v28lift37.annotation.gtf\n",
    "  - [READID2SEQ] format\n",
    "    READID [tab] GUIDESEQ\n",
    "    READ1  [tab] ATAGATGTCCTGTGGCCCCG\n",
    "    ...\n",
    "    If readid is same with guideseq,\n",
    "\tREADID [tab] GUIDESEQ\n",
    "    ATAGATGTCCTGTGGCCCCG [tab] ATAGATGTCCTGTGGCCCCG\n",
    "    ...    \n",
    "  - Options\n",
    "    --output       : Specify an alignment-info file (Default:{outputalignment})\n",
    "    --outputfastq  : Specify a temporary fastq output file (Default:{outputfastq})\n",
    "    --outputbowtie : Specify a temporary bowtie output file (Default:{outputbowtie})\n",
    "'''\n",
    "\n",
    "\n",
    "# 1: locations(perfect match, hg38) 2: protein coding genes(perfect match, hg38) , 3: locations(1bp mismatch, hg38) 4: protein coding genes(1bp mismatch, hg38) \n",
    "alignment_summary = pd.read_csv(\"Avana_align_summary_hg38.txt\",sep=\"\\t\",header=None,index_col=0)\n",
    "alignment_summary.head(3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(72148, 6)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alignment_summary.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAGGGCTCCAAAAAGG</td>\n",
       "      <td>chr6_26199590_+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX_22262373_-</td>\n",
       "      <td>AC004470</td>\n",
       "      <td>chr19_21162609_-,chr6_26158201_+,chr6_35213800...</td>\n",
       "      <td>ZNF431,SOX5,H2BC11,GTF2IRD1,H2BC5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAACTGGTTTCCAGCAGAG</td>\n",
       "      <td>chr9_111327947_+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr14_76820416_+,chr7_124503785_+</td>\n",
       "      <td>ANGEL1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAGTGTGTGTGGGAGCTG</td>\n",
       "      <td>chr21_30718721_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr9_101362324_+,chr7_130307739_-</td>\n",
       "      <td>BAAT,CPA4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAGTTCTTTCACATCGGA</td>\n",
       "      <td>chr12_48472709_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr15_35237370_-</td>\n",
       "      <td>ANP32AP1,DPH6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAATCTCGTTTCTGCCGAG</td>\n",
       "      <td>chr14_46651660_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGGGTTGGAGGCTCCAGG</td>\n",
       "      <td>chr19_35351974_+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr20_47013194_-</td>\n",
       "      <td>EYA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGGTTTCCTCTTTCCATG</td>\n",
       "      <td>chr19_14941484_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr19_14800117_+,chr2_157039350_+</td>\n",
       "      <td>OR7C1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTAGCTGAGCACCAGCA</td>\n",
       "      <td>chr9_21217254_+,chr9_21166561_+,chr9_21187480_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chr9_21202114_+,chr8_63544124_+</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTCACAGTACCAGGGCG</td>\n",
       "      <td>chrX_37008459_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX_34942880_-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>chrX_37545156_+,chrX_34132206_+</td>\n",
       "      <td>AF241726,AL591501,FAM47A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTGGCACAAATACGGG</td>\n",
       "      <td>chr14_46651514_+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2746 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                      1    2  \\\n",
       "0                                                                              \n",
       "AAAAAAGGGCTCCAAAAAGG                                    chr6_26199590_+  NaN   \n",
       "AAAACTGGTTTCCAGCAGAG                                   chr9_111327947_+  NaN   \n",
       "AAAAGTGTGTGTGGGAGCTG                                   chr21_30718721_-  NaN   \n",
       "AAAAGTTCTTTCACATCGGA                                   chr12_48472709_-  NaN   \n",
       "AAAATCTCGTTTCTGCCGAG                                   chr14_46651660_-  NaN   \n",
       "...                                                                 ...  ...   \n",
       "TTTGGGTTGGAGGCTCCAGG                                   chr19_35351974_+  NaN   \n",
       "TTTGGTTTCCTCTTTCCATG                                   chr19_14941484_-  NaN   \n",
       "TTTGTAGCTGAGCACCAGCA  chr9_21217254_+,chr9_21166561_+,chr9_21187480_...  NaN   \n",
       "TTTGTCACAGTACCAGGGCG                                    chrX_37008459_-  NaN   \n",
       "TTTGTTGGCACAAATACGGG                                   chr14_46651514_+  NaN   \n",
       "\n",
       "                                    3         4  \\\n",
       "0                                                 \n",
       "AAAAAAGGGCTCCAAAAAGG  chrX_22262373_-  AC004470   \n",
       "AAAACTGGTTTCCAGCAGAG              NaN       NaN   \n",
       "AAAAGTGTGTGTGGGAGCTG              NaN       NaN   \n",
       "AAAAGTTCTTTCACATCGGA              NaN       NaN   \n",
       "AAAATCTCGTTTCTGCCGAG              NaN       NaN   \n",
       "...                               ...       ...   \n",
       "TTTGGGTTGGAGGCTCCAGG              NaN       NaN   \n",
       "TTTGGTTTCCTCTTTCCATG              NaN       NaN   \n",
       "TTTGTAGCTGAGCACCAGCA              NaN       NaN   \n",
       "TTTGTCACAGTACCAGGGCG  chrX_34942880_-       NaN   \n",
       "TTTGTTGGCACAAATACGGG              NaN       NaN   \n",
       "\n",
       "                                                                      5  \\\n",
       "0                                                                         \n",
       "AAAAAAGGGCTCCAAAAAGG  chr19_21162609_-,chr6_26158201_+,chr6_35213800...   \n",
       "AAAACTGGTTTCCAGCAGAG                  chr14_76820416_+,chr7_124503785_+   \n",
       "AAAAGTGTGTGTGGGAGCTG                  chr9_101362324_+,chr7_130307739_-   \n",
       "AAAAGTTCTTTCACATCGGA                                   chr15_35237370_-   \n",
       "AAAATCTCGTTTCTGCCGAG                                                NaN   \n",
       "...                                                                 ...   \n",
       "TTTGGGTTGGAGGCTCCAGG                                   chr20_47013194_-   \n",
       "TTTGGTTTCCTCTTTCCATG                  chr19_14800117_+,chr2_157039350_+   \n",
       "TTTGTAGCTGAGCACCAGCA                    chr9_21202114_+,chr8_63544124_+   \n",
       "TTTGTCACAGTACCAGGGCG                    chrX_37545156_+,chrX_34132206_+   \n",
       "TTTGTTGGCACAAATACGGG                                                NaN   \n",
       "\n",
       "                                                      6  \n",
       "0                                                        \n",
       "AAAAAAGGGCTCCAAAAAGG  ZNF431,SOX5,H2BC11,GTF2IRD1,H2BC5  \n",
       "AAAACTGGTTTCCAGCAGAG                             ANGEL1  \n",
       "AAAAGTGTGTGTGGGAGCTG                          BAAT,CPA4  \n",
       "AAAAGTTCTTTCACATCGGA                      ANP32AP1,DPH6  \n",
       "AAAATCTCGTTTCTGCCGAG                                NaN  \n",
       "...                                                 ...  \n",
       "TTTGGGTTGGAGGCTCCAGG                               EYA2  \n",
       "TTTGGTTTCCTCTTTCCATG                              OR7C1  \n",
       "TTTGTAGCTGAGCACCAGCA                                NaN  \n",
       "TTTGTCACAGTACCAGGGCG           AF241726,AL591501,FAM47A  \n",
       "TTTGTTGGCACAAATACGGG                                NaN  \n",
       "\n",
       "[2746 rows x 6 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not-aligned\n",
    "alignment_summary[alignment_summary[2].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>AAAAAAGGGCTCCAAAAAGG</td>\n",
       "      <td>chr6_26199607_+</td>\n",
       "      <td>HIST1H2BF (8343)</td>\n",
       "      <td>1</td>\n",
       "      <td>H2BC7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>AAAACTGGTTTCCAGCAGAG</td>\n",
       "      <td>chr9_111327964_+</td>\n",
       "      <td>OR2K2 (26248)</td>\n",
       "      <td>1</td>\n",
       "      <td>OR2K2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>59</td>\n",
       "      <td>AAAAGTGTGTGTGGGAGCTG</td>\n",
       "      <td>chr21_30718727_-</td>\n",
       "      <td>KRTAP21-3 (100288323)</td>\n",
       "      <td>1</td>\n",
       "      <td>KRTAP21-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>61</td>\n",
       "      <td>AAAAGTTCTTTCACATCGGA</td>\n",
       "      <td>chr12_48472715_-</td>\n",
       "      <td>ANP32D (23519)</td>\n",
       "      <td>1</td>\n",
       "      <td>ANP32D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>72</td>\n",
       "      <td>AAAATCTCGTTTCTGCCGAG</td>\n",
       "      <td>chr14_46651666_-</td>\n",
       "      <td>RPL10L (140801)</td>\n",
       "      <td>1</td>\n",
       "      <td>RPL10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74277</td>\n",
       "      <td>TTTGTAGCTGAGCACCAGCA</td>\n",
       "      <td>chr9_21207063_+</td>\n",
       "      <td>IFNA10 (3446)</td>\n",
       "      <td>5</td>\n",
       "      <td>IFNA10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74278</td>\n",
       "      <td>TTTGTAGCTGAGCACCAGCA</td>\n",
       "      <td>chr9_21217271_+</td>\n",
       "      <td>IFNA16 (3449)</td>\n",
       "      <td>5</td>\n",
       "      <td>IFNA16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74290</td>\n",
       "      <td>TTTGTCACAGTACCAGGGCG</td>\n",
       "      <td>chrX_37008465_-</td>\n",
       "      <td>FAM47C (442444)</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM47C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74338</td>\n",
       "      <td>TTTGTTGGCACAAATACGGG</td>\n",
       "      <td>chr14_46651531_+</td>\n",
       "      <td>RPL10L (140801)</td>\n",
       "      <td>1</td>\n",
       "      <td>RPL10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74354</td>\n",
       "      <td>TGTAATTCCCAATAGCTGAA</td>\n",
       "      <td>chr7_122995203_+</td>\n",
       "      <td>TAS2R16 (50833)</td>\n",
       "      <td>1</td>\n",
       "      <td>TAS2R16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2873 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna  genome_alignment                   gene  \\\n",
       "5      AAAAAAGGGCTCCAAAAAGG   chr6_26199607_+       HIST1H2BF (8343)   \n",
       "34     AAAACTGGTTTCCAGCAGAG  chr9_111327964_+          OR2K2 (26248)   \n",
       "59     AAAAGTGTGTGTGGGAGCTG  chr21_30718727_-  KRTAP21-3 (100288323)   \n",
       "61     AAAAGTTCTTTCACATCGGA  chr12_48472715_-         ANP32D (23519)   \n",
       "72     AAAATCTCGTTTCTGCCGAG  chr14_46651666_-        RPL10L (140801)   \n",
       "...                     ...               ...                    ...   \n",
       "74277  TTTGTAGCTGAGCACCAGCA   chr9_21207063_+          IFNA10 (3446)   \n",
       "74278  TTTGTAGCTGAGCACCAGCA   chr9_21217271_+          IFNA16 (3449)   \n",
       "74290  TTTGTCACAGTACCAGGGCG   chrX_37008465_-        FAM47C (442444)   \n",
       "74338  TTTGTTGGCACAAATACGGG  chr14_46651531_+        RPL10L (140801)   \n",
       "74354  TGTAATTCCCAATAGCTGAA  chr7_122995203_+        TAS2R16 (50833)   \n",
       "\n",
       "       n_alignments official_symbol  \n",
       "5                 1           H2BC7  \n",
       "34                1           OR2K2  \n",
       "59                1       KRTAP21-3  \n",
       "61                1          ANP32D  \n",
       "72                1          RPL10L  \n",
       "...             ...             ...  \n",
       "74277             5          IFNA10  \n",
       "74278             5          IFNA16  \n",
       "74290             1          FAM47C  \n",
       "74338             1          RPL10L  \n",
       "74354             1         TAS2R16  \n",
       "\n",
       "[2873 rows x 5 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# not-aligned (may due to gene annotation difference -> keep)\n",
    "guidegenemap[guidegenemap['sgrna'].isin(alignment_summary[alignment_summary[2].isnull()].index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [],
   "source": [
    "# targeting multiple proteins -> discard\n",
    "multi_proteins = list()\n",
    "for seq in alignment_summary.index:\n",
    "    locations = alignment_summary.loc[seq,1]\n",
    "    genes = alignment_summary.loc[seq,2]\n",
    "    if len(str(genes).split(\",\")) > 1: # targeting more than 1 proteins\n",
    "        multi_proteins.append(seq)\n",
    "single_protein = setdiff1d(alignment_summary.index,multi_proteins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14430"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(multi_proteins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>chr10_110964620_+</td>\n",
       "      <td>SHOC2 (8036)</td>\n",
       "      <td>1</td>\n",
       "      <td>SHOC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>chr12_95003615_+</td>\n",
       "      <td>NDUFA12 (55967)</td>\n",
       "      <td>1</td>\n",
       "      <td>NDUFA12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>chr4_75970356_-</td>\n",
       "      <td>SDAD1 (55153)</td>\n",
       "      <td>1</td>\n",
       "      <td>SDAD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>chr2_33588446_-</td>\n",
       "      <td>FAM98A (25940)</td>\n",
       "      <td>1</td>\n",
       "      <td>FAM98A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>chr19_19891600_+</td>\n",
       "      <td>ZNF253 (56242)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74372</td>\n",
       "      <td>TGTAGTCCTGCACGAACCAG</td>\n",
       "      <td>chr8_11200791_+</td>\n",
       "      <td>XKR6 (286046)</td>\n",
       "      <td>1</td>\n",
       "      <td>XKR6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74373</td>\n",
       "      <td>TGTACAAAGCCCTCTGCGCA</td>\n",
       "      <td>chr19_52504722_-</td>\n",
       "      <td>ZNF578 (147660)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74374</td>\n",
       "      <td>TGTACAAAGCCCTCTGTGCA</td>\n",
       "      <td>chr19_52547579_-</td>\n",
       "      <td>ZNF808 (388558)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74375</td>\n",
       "      <td>TGTACAAAGTCCTCTGAGCG</td>\n",
       "      <td>chr19_53149299_+</td>\n",
       "      <td>ZNF347 (84671)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>74376</td>\n",
       "      <td>TGTACAAATTCTGCTGAGCG</td>\n",
       "      <td>chr11_3371635_+</td>\n",
       "      <td>ZNF195 (7748)</td>\n",
       "      <td>1</td>\n",
       "      <td>ZNF195</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>58051 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna   genome_alignment             gene  n_alignments  \\\n",
       "0      AAAAAAATCCAGCAATGCAG  chr10_110964620_+     SHOC2 (8036)             1   \n",
       "1      AAAAAACCCGTAGATAGCCT   chr12_95003615_+  NDUFA12 (55967)             1   \n",
       "2      AAAAAAGAAGAAAAAACCAG    chr4_75970356_-    SDAD1 (55153)             1   \n",
       "3      AAAAAAGCTCAAGAAGGAGG    chr2_33588446_-   FAM98A (25940)             1   \n",
       "4      AAAAAAGGCTGTAAAAGCGT   chr19_19891600_+   ZNF253 (56242)             1   \n",
       "...                     ...                ...              ...           ...   \n",
       "74372  TGTAGTCCTGCACGAACCAG    chr8_11200791_+    XKR6 (286046)             1   \n",
       "74373  TGTACAAAGCCCTCTGCGCA   chr19_52504722_-  ZNF578 (147660)             1   \n",
       "74374  TGTACAAAGCCCTCTGTGCA   chr19_52547579_-  ZNF808 (388558)             1   \n",
       "74375  TGTACAAAGTCCTCTGAGCG   chr19_53149299_+   ZNF347 (84671)             1   \n",
       "74376  TGTACAAATTCTGCTGAGCG    chr11_3371635_+    ZNF195 (7748)             1   \n",
       "\n",
       "      official_symbol  \n",
       "0               SHOC2  \n",
       "1             NDUFA12  \n",
       "2               SDAD1  \n",
       "3              FAM98A  \n",
       "4              ZNF253  \n",
       "...               ...  \n",
       "74372            XKR6  \n",
       "74373          ZNF578  \n",
       "74374          ZNF808  \n",
       "74375          ZNF347  \n",
       "74376          ZNF195  \n",
       "\n",
       "[58051 rows x 5 columns]"
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "guidegenemap[~guidegenemap['sgrna'].isin(multi_proteins)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sgrna\n",
       "GCATAGGCCCAATCATTTCA    11\n",
       "GTACATCGCGCTCATCACCA     8\n",
       "GCTGCTGCAGGTGGTCACAG     7\n",
       "GCTGCTGCTGAGCATGGGCG     7\n",
       "GGAAATGCCCGGTTTGGCCA     7\n",
       "                        ..\n",
       "GGCCGGCTGAGACTTCAGCG     1\n",
       "GGCCGGCTGACCATCACGCA     1\n",
       "GGCCGGCTCCAGCAGCGCCG     1\n",
       "GGCCGGCGTGACCATGACCA     1\n",
       "AAAAAAATCCAGCAATGCAG     1\n",
       "Length: 57718, dtype: int64"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = guidegenemap[~guidegenemap['sgrna'].isin(multi_proteins)]\n",
    "a.groupby('sgrna').size().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sgrna</th>\n",
       "      <th>genome_alignment</th>\n",
       "      <th>gene</th>\n",
       "      <th>n_alignments</th>\n",
       "      <th>official_symbol</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>37153</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49332802_-</td>\n",
       "      <td>GAGE13 (645051)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37165</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49323258_-</td>\n",
       "      <td>GAGE12J (729396)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37166</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49304921_-</td>\n",
       "      <td>GAGE10 (102724473)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37167</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49571619_-</td>\n",
       "      <td>GAGE12G (645073)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37168</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49552508_-</td>\n",
       "      <td>GAGE12E (729431)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37169</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49533396_-</td>\n",
       "      <td>GAGE12C (729422)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37170</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49562061_-</td>\n",
       "      <td>GAGE12F (100008586)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12F</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37171</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49600239_-</td>\n",
       "      <td>GAGE1 (2543)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37172</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49581168_-</td>\n",
       "      <td>GAGE12H (729442)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12H</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37173</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49542952_-</td>\n",
       "      <td>GAGE12D (100132399)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE12D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37174</td>\n",
       "      <td>GCATAGGCCCAATCATTTCA</td>\n",
       "      <td>chrX_49590712_-</td>\n",
       "      <td>GAGE2A (729447)</td>\n",
       "      <td>12</td>\n",
       "      <td>GAGE2A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      sgrna genome_alignment                 gene  \\\n",
       "37153  GCATAGGCCCAATCATTTCA  chrX_49332802_-      GAGE13 (645051)   \n",
       "37165  GCATAGGCCCAATCATTTCA  chrX_49323258_-     GAGE12J (729396)   \n",
       "37166  GCATAGGCCCAATCATTTCA  chrX_49304921_-   GAGE10 (102724473)   \n",
       "37167  GCATAGGCCCAATCATTTCA  chrX_49571619_-     GAGE12G (645073)   \n",
       "37168  GCATAGGCCCAATCATTTCA  chrX_49552508_-     GAGE12E (729431)   \n",
       "37169  GCATAGGCCCAATCATTTCA  chrX_49533396_-     GAGE12C (729422)   \n",
       "37170  GCATAGGCCCAATCATTTCA  chrX_49562061_-  GAGE12F (100008586)   \n",
       "37171  GCATAGGCCCAATCATTTCA  chrX_49600239_-         GAGE1 (2543)   \n",
       "37172  GCATAGGCCCAATCATTTCA  chrX_49581168_-     GAGE12H (729442)   \n",
       "37173  GCATAGGCCCAATCATTTCA  chrX_49542952_-  GAGE12D (100132399)   \n",
       "37174  GCATAGGCCCAATCATTTCA  chrX_49590712_-      GAGE2A (729447)   \n",
       "\n",
       "       n_alignments official_symbol  \n",
       "37153            12          GAGE13  \n",
       "37165            12         GAGE12J  \n",
       "37166            12          GAGE10  \n",
       "37167            12         GAGE12G  \n",
       "37168            12         GAGE12E  \n",
       "37169            12         GAGE12C  \n",
       "37170            12         GAGE12F  \n",
       "37171            12           GAGE1  \n",
       "37172            12         GAGE12H  \n",
       "37173            12         GAGE12D  \n",
       "37174            12          GAGE2A  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a[a['sgrna']=='GCATAGGCCCAATCATTTCA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1                                      chrX_49304915_-\n",
       "2                                               GAGE10\n",
       "3    chrX_49552502_-,chrX_49533390_-,chrX_49562055_...\n",
       "4    GAGE2A,GAGE12H,GAGE13,GAGE12G,GAGE12J,GAGE1,GA...\n",
       "5                                                  NaN\n",
       "6                                                  NaN\n",
       "Name: GCATAGGCCCAATCATTTCA, dtype: object"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alignment_summary.loc['GCATAGGCCCAATCATTTCA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save filtered\n",
    "guidegenemap[~guidegenemap['sgrna'].isin(multi_proteins)].to_csv(\"Achilles_guide_map_mod.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "temp = guidegenemap.loc[targetsgrna]\n",
    "tempdf = dict()\n",
    "for i in range(len(temp.index)):\n",
    "    gene = temp.iloc[i]['gene'].replace(\")\",\"\").split(\" (\")[0]\n",
    "    entrezid = temp.iloc[i]['gene'].replace(\")\",\"\").split(\" (\")[1]\n",
    "    if gene in old2new and old2new[gene] in finalgenes:\n",
    "        temprow = [ temp.iloc[i]['genome_alignment'] , old2new[gene], entrezid, temp.iloc[i]['n_alignments'] ]\n",
    "        tempdf[temp.index[i]] = temprow\n",
    "        \n",
    "        \n",
    "tempdf = pd.DataFrame.from_dict(tempdf).T.rename(columns={0:\"genome_alignment\",1:\"gene\",2:\"entrezid\",3:\"n_alignments\"})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AGAP4 AKR7L AMY1A AMY1B AMY1C ANKRD20A1 ANKRD20A2 ANKRD20A3 ANKRD20A4 ARL17A ARL17B BOLA2 BOLA2B C4A C4B CBSL CBWD3 CCZ1 CCZ1B CD8B2 CDK11B CGB1 CGB2 CGB3 CGB5 CGB8 CKMT1A CKMT1B CRYAA CRYAA2 CSH1 CSNK2A3 CT45A1 CT45A10 CT45A2 CT45A3 CT45A5 CT45A6 CT45A7 CT45A8 CT45A9 CT47A1 CT47A10 CT47A11 CT47A12 CT47A2 CT47A3 CT47A4 CT47A5 CT47A6 CT47A7 CT47A8 CT47A9 CTAG1A CTAG1B CTAGE4 CTAGE6 CTAGE8 DDT DDTL DEFA1 DEFA1B DEFA3 DEFB103A DEFB103B DEFB104A DEFB104B DEFB105A DEFB105B DEFB106A DEFB106B DEFB107A DEFB107B DEFB130A DEFB130B DEFB131B DMRTC1 DMRTC1B EIF3C ELOA3 ELOA3B F8A1 F8A2 F8A3 FAM153C FAM156A FAM156B FAM25C FAM25G FAM72A FAM72B FAM72C FAM72D FAM86B1 FAM90A1 FAM90A26 FCGR1A FCGR1B FOLR3 FOXD4L3 FOXD4L6 FRG2 GAGE1 GAGE12C GAGE12D GAGE12E GAGE12F GAGE12G GAGE13 GAGE2A GATD3A GATD3B GCOM1 GDF5OS GGTLC3 GOLGA6C GOLGA6L1 GOLGA6L10 GOLGA6L2 GOLGA6L4 GOLGA6L9 GOLGA8A GOLGA8B GOLGA8J GOLGA8K GOLGA8O GOLGA8Q GOLGA8T GPR89A GPR89B GRAP GRAPL GTF2IRD2 GTF2IRD2B GYPB H2AFB1 H2AFB2 H2AFB3 HBA1 HBA2 HIST1H4J HIST1H4K HIST2H2AA3 HIST2H2AA4 HIST2H3A HIST2H3C HIST2H4A HIST2H4B HNRNPCL2 HNRNPCL3 HNRNPCL4 HSFX1 HSFX2 ICOSLG IFNA13 KCNE1 KCNE1B KCNJ18 KIAA0754 KRTAP10-4 KRTAP10-7 KRTAP2-1 KRTAP2-2 KRTAP2-3 KRTAP2-4 KRTAP4-7 KRTAP5-4 KRTAP9-2 KRTAP9-4 KRTAP9-7 LCE1C LCE1F LCE2C LIMS3 LIMS4 LOC102723996 LYZL1 LYZL2 MAGEA2 MAGEA2B MAGEA9 MAGEA9B MAGED4 MAGED4B MBD3L3 MBD3L4 MBD3L5 MFSD14C MYZAP NANOGP8 NBPF11 NBPF12 NBPF15 NBPF19 NBPF4 NBPF6 NBPF9 NOMO3 NOTCH2NLA NOTCH2NLB NOTCH2NLC NPIPA1 NPIPA2 NPIPA3 NPIPA7 NPIPA8 NPIPB13 NPIPB15 NPIPB5 NPIPB8 NPIPB9 NPY4R NPY4R2 NUDT4 NUDT4B NXF2 NXF2B OBP2A OBP2B OPN1MW OPN1MW2 OPN1MW3 OR11H1 OR11H12 OR11H2 OR2A1 OR2A42 OR2J1 OR4F16 OR4F17 OR4F29 OR4F3 OR4F4 PABPC1L2A PABPC1L2B PAGE2B PCDHB9 PCDHGB5 PGA3 PGA4 PGA5 PGBD3 PLGLB1 PLGLB2 POLR2J POLR2J2 POLR2J3 POTEB POTEB2 POTEB3 POTEJ PPIAL4C PPIAL4D PPIAL4E PPIAL4F PPP5D1 PRAMEF10 PRAMEF13 PRAMEF15 PRAMEF17 PRAMEF25 PRAMEF26 PRAMEF27 PRAMEF33 PRAMEF5 PRAMEF6 PRAMEF7 PRAMEF8 PRAMEF9 PRH1 PRR20A PRR20B PRR20C PRR20D PRR20E PRR23D1 PRR23D2 PRSS2 PSG7 RAB6C RAB6D RGPD5 RGPD6 RGPD8 RIMBP3B RIMBP3C SCGB1C1 SCGB1C2 SERF1A SERF1B SIGLEC14 SIK1 SIK1B SLC22A24 SLC35E2A SLC35G4 SLX1A SLX1B SMIM11A SMIM11B SMN1 SMN2 SPACA5 SPACA5B SPAG11A SPAG11B SPANXA1 SPANXA2 SPANXC SPANXD SPATA31A5 SPATA31D3 SPATA31D4 SPDYE16 SPDYE2 SPDYE2B SPDYE5 SPDYE6 SPIN2A SPIN2B SPRR2A SPRR2B SPRR2D SPRR2E SSX2 SSX2B SSX4 SSX4B SULT1A3 SULT1A4 TBC1D3 TBC1D3C TBC1D3D TBC1D3E TBC1D3G TBC1D3H TBC1D3I TBC1D3K TBC1D3L TIMM23B TP53TG3 TP53TG3B TP53TG3C TP53TG3D TP53TG3E TP53TG3F TRAPPC2 TRAPPC2B TRIM49 TRIM49C TRIM64 TRIM64B TRIM64C TRIM74 TUBB8P12 TVP23C TYW1B U2AF1 U2AF1L5 UBE2L5 UGT2A2 UGT2B10 UPK3BL1 UPK3BL2 UQCRHL USP17L1 USP17L11 USP17L12 USP17L15 USP17L17 USP17L18 USP17L19 USP17L2 USP17L20 USP17L21 USP17L22 USP17L24 USP17L25 USP17L26 USP17L27 USP17L28 USP17L29 USP17L3 USP17L30 USP17L4 USP17L7 USP17L8 VCX VCX2 VCX3A XAGE1A XAGE1B ZC3H11B ZDHHC11B ZNF525 ZNF595 ZNF626 ZNF705D ZNF705E ZNF718 ZNF724 ZNF735 ZNF738 ZNF888 ZSCAN5C\n"
     ]
    }
   ],
   "source": [
    "print (\" \".join(setdiff1d(list(finalgenes),tempdf['gene'].unique())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add gene to readcount\n",
    "intst = intersect1d(readcount.index,list(targetsgrna.keys()))\n",
    "readcount = readcount.loc[intst]\n",
    "readcount['GENE'] = pd.Series(targetsgrna)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>143B-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>2313287-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>2313287-311Cas9_RepB_p5_batch3</th>\n",
       "      <th>253J-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>59M-311Cas9_RepA_p4_batch3</th>\n",
       "      <th>59M-311Cas9_RepB_p4_batch3</th>\n",
       "      <th>...</th>\n",
       "      <th>YKG1-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>ZR-75-1-311Cas9_RepA_p5_batch2</th>\n",
       "      <th>ZR-75-1-311Cas9_RepB_p5_batch2</th>\n",
       "      <th>pDNA Avana4_010115_1.5Ex_batch0</th>\n",
       "      <th>pDNA Avana4_010115_1.5Ex_batch1</th>\n",
       "      <th>pDNA Avana4_0101215_0.55Ex_batch1</th>\n",
       "      <th>pDNA Avana4_060115_0.55Ex_batch1</th>\n",
       "      <th>pDNA Avana4_060115_1.5Ex_batch0</th>\n",
       "      <th>pDNA Avana4_060115_1.5Ex_batch1</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>833.0</td>\n",
       "      <td>327.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>683.0</td>\n",
       "      <td>373.0</td>\n",
       "      <td>...</td>\n",
       "      <td>556.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>386.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>324.0</td>\n",
       "      <td>289.0</td>\n",
       "      <td>SHOC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>1089.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>736.0</td>\n",
       "      <td>727.0</td>\n",
       "      <td>565.0</td>\n",
       "      <td>770.0</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>720.0</td>\n",
       "      <td>...</td>\n",
       "      <td>773.0</td>\n",
       "      <td>632.0</td>\n",
       "      <td>673.0</td>\n",
       "      <td>398.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>254.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>445.0</td>\n",
       "      <td>NDUFA12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>95.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>...</td>\n",
       "      <td>57.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>SDAD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>236.0</td>\n",
       "      <td>197.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>438.0</td>\n",
       "      <td>152.0</td>\n",
       "      <td>252.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>161.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>138.0</td>\n",
       "      <td>...</td>\n",
       "      <td>236.0</td>\n",
       "      <td>454.0</td>\n",
       "      <td>214.0</td>\n",
       "      <td>131.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>90.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>135.0</td>\n",
       "      <td>FAM98A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>460.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>125.0</td>\n",
       "      <td>196.0</td>\n",
       "      <td>251.0</td>\n",
       "      <td>204.0</td>\n",
       "      <td>208.0</td>\n",
       "      <td>338.0</td>\n",
       "      <td>254.0</td>\n",
       "      <td>85.0</td>\n",
       "      <td>...</td>\n",
       "      <td>186.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>262.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>153.0</td>\n",
       "      <td>109.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>174.0</td>\n",
       "      <td>193.0</td>\n",
       "      <td>ZNF253</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTGGAGAGATGTACGA</td>\n",
       "      <td>334.0</td>\n",
       "      <td>120.0</td>\n",
       "      <td>266.0</td>\n",
       "      <td>293.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>302.0</td>\n",
       "      <td>486.0</td>\n",
       "      <td>369.0</td>\n",
       "      <td>473.0</td>\n",
       "      <td>577.0</td>\n",
       "      <td>...</td>\n",
       "      <td>276.0</td>\n",
       "      <td>314.0</td>\n",
       "      <td>392.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>199.0</td>\n",
       "      <td>140.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>172.0</td>\n",
       "      <td>141.0</td>\n",
       "      <td>LIPH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTGGCACAAATACGGG</td>\n",
       "      <td>1112.0</td>\n",
       "      <td>590.0</td>\n",
       "      <td>469.0</td>\n",
       "      <td>888.0</td>\n",
       "      <td>963.0</td>\n",
       "      <td>850.0</td>\n",
       "      <td>1005.0</td>\n",
       "      <td>981.0</td>\n",
       "      <td>1202.0</td>\n",
       "      <td>909.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1062.0</td>\n",
       "      <td>1182.0</td>\n",
       "      <td>1285.0</td>\n",
       "      <td>513.0</td>\n",
       "      <td>503.0</td>\n",
       "      <td>356.0</td>\n",
       "      <td>289.0</td>\n",
       "      <td>486.0</td>\n",
       "      <td>486.0</td>\n",
       "      <td>RPL10L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTCCTCTGAAGTAGT</td>\n",
       "      <td>116.0</td>\n",
       "      <td>371.0</td>\n",
       "      <td>381.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>217.0</td>\n",
       "      <td>535.0</td>\n",
       "      <td>523.0</td>\n",
       "      <td>283.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>117.0</td>\n",
       "      <td>...</td>\n",
       "      <td>447.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>128.0</td>\n",
       "      <td>886.0</td>\n",
       "      <td>792.0</td>\n",
       "      <td>530.0</td>\n",
       "      <td>516.0</td>\n",
       "      <td>762.0</td>\n",
       "      <td>771.0</td>\n",
       "      <td>RPS24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTCCTCTTCTCGAGG</td>\n",
       "      <td>1065.0</td>\n",
       "      <td>398.0</td>\n",
       "      <td>345.0</td>\n",
       "      <td>1011.0</td>\n",
       "      <td>640.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>565.0</td>\n",
       "      <td>692.0</td>\n",
       "      <td>301.0</td>\n",
       "      <td>1439.0</td>\n",
       "      <td>...</td>\n",
       "      <td>836.0</td>\n",
       "      <td>481.0</td>\n",
       "      <td>803.0</td>\n",
       "      <td>281.0</td>\n",
       "      <td>259.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>173.0</td>\n",
       "      <td>268.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>CRISPLD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTGTCTCATTCATGG</td>\n",
       "      <td>1748.0</td>\n",
       "      <td>790.0</td>\n",
       "      <td>680.0</td>\n",
       "      <td>1819.0</td>\n",
       "      <td>1048.0</td>\n",
       "      <td>997.0</td>\n",
       "      <td>1440.0</td>\n",
       "      <td>1482.0</td>\n",
       "      <td>1099.0</td>\n",
       "      <td>756.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1931.0</td>\n",
       "      <td>1368.0</td>\n",
       "      <td>974.0</td>\n",
       "      <td>721.0</td>\n",
       "      <td>719.0</td>\n",
       "      <td>485.0</td>\n",
       "      <td>454.0</td>\n",
       "      <td>702.0</td>\n",
       "      <td>666.0</td>\n",
       "      <td>C1orf210</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>70994 rows × 1499 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      143B-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        833.0   \n",
       "AAAAAACCCGTAGATAGCCT                       1089.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         95.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                        236.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                        460.0   \n",
       "...                                           ...   \n",
       "TTTGTTGGAGAGATGTACGA                        334.0   \n",
       "TTTGTTGGCACAAATACGGG                       1112.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                        116.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                       1065.0   \n",
       "TTTGTTTGTCTCATTCATGG                       1748.0   \n",
       "\n",
       "                      2313287-311Cas9_RepA_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                           327.0   \n",
       "AAAAAACCCGTAGATAGCCT                           383.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            32.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                           197.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                           117.0   \n",
       "...                                              ...   \n",
       "TTTGTTGGAGAGATGTACGA                           120.0   \n",
       "TTTGTTGGCACAAATACGGG                           590.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                           371.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                           398.0   \n",
       "TTTGTTTGTCTCATTCATGG                           790.0   \n",
       "\n",
       "                      2313287-311Cas9_RepB_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                           188.0   \n",
       "AAAAAACCCGTAGATAGCCT                           391.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            69.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            59.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                           125.0   \n",
       "...                                              ...   \n",
       "TTTGTTGGAGAGATGTACGA                           266.0   \n",
       "TTTGTTGGCACAAATACGGG                           469.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                           381.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                           345.0   \n",
       "TTTGTTTGTCTCATTCATGG                           680.0   \n",
       "\n",
       "                      253J-311Cas9_RepA_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        546.0   \n",
       "AAAAAACCCGTAGATAGCCT                       1151.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         72.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                        438.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                        196.0   \n",
       "...                                           ...   \n",
       "TTTGTTGGAGAGATGTACGA                        293.0   \n",
       "TTTGTTGGCACAAATACGGG                        888.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                         99.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                       1011.0   \n",
       "TTTGTTTGTCTCATTCATGG                       1819.0   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                            213.0   \n",
       "AAAAAACCCGTAGATAGCCT                            736.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             46.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            152.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                            251.0   \n",
       "...                                               ...   \n",
       "TTTGTTGGAGAGATGTACGA                            301.0   \n",
       "TTTGTTGGCACAAATACGGG                            963.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                            217.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                            640.0   \n",
       "TTTGTTTGTCTCATTCATGG                           1048.0   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepB_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                            349.0   \n",
       "AAAAAACCCGTAGATAGCCT                            727.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             60.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            252.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                            204.0   \n",
       "...                                               ...   \n",
       "TTTGTTGGAGAGATGTACGA                            302.0   \n",
       "TTTGTTGGCACAAATACGGG                            850.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                            535.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                            511.0   \n",
       "TTTGTTTGTCTCATTCATGG                            997.0   \n",
       "\n",
       "                      5637-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        280.0   \n",
       "AAAAAACCCGTAGATAGCCT                        565.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         99.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                        138.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                        208.0   \n",
       "...                                           ...   \n",
       "TTTGTTGGAGAGATGTACGA                        486.0   \n",
       "TTTGTTGGCACAAATACGGG                       1005.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                        523.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                        565.0   \n",
       "TTTGTTTGTCTCATTCATGG                       1440.0   \n",
       "\n",
       "                      5637-311Cas9_RepB_p6_batch3  59M-311Cas9_RepA_p4_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        489.0                       683.0   \n",
       "AAAAAACCCGTAGATAGCCT                        770.0                      1279.0   \n",
       "AAAAAAGAAGAAAAAACCAG                        102.0                        18.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                        161.0                       378.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                        338.0                       254.0   \n",
       "...                                           ...                         ...   \n",
       "TTTGTTGGAGAGATGTACGA                        369.0                       473.0   \n",
       "TTTGTTGGCACAAATACGGG                        981.0                      1202.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                        283.0                       118.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                        692.0                       301.0   \n",
       "TTTGTTTGTCTCATTCATGG                       1482.0                      1099.0   \n",
       "\n",
       "                      59M-311Cas9_RepB_p4_batch3  ...  \\\n",
       "AAAAAAATCCAGCAATGCAG                       373.0  ...   \n",
       "AAAAAACCCGTAGATAGCCT                       720.0  ...   \n",
       "AAAAAAGAAGAAAAAACCAG                        76.0  ...   \n",
       "AAAAAAGCTCAAGAAGGAGG                       138.0  ...   \n",
       "AAAAAAGGCTGTAAAAGCGT                        85.0  ...   \n",
       "...                                          ...  ...   \n",
       "TTTGTTGGAGAGATGTACGA                       577.0  ...   \n",
       "TTTGTTGGCACAAATACGGG                       909.0  ...   \n",
       "TTTGTTTCCTCTGAAGTAGT                       117.0  ...   \n",
       "TTTGTTTCCTCTTCTCGAGG                      1439.0  ...   \n",
       "TTTGTTTGTCTCATTCATGG                       756.0  ...   \n",
       "\n",
       "                      YKG1-311Cas9_RepB_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        556.0   \n",
       "AAAAAACCCGTAGATAGCCT                        773.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         57.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                        236.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                        186.0   \n",
       "...                                           ...   \n",
       "TTTGTTGGAGAGATGTACGA                        276.0   \n",
       "TTTGTTGGCACAAATACGGG                       1062.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                        447.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                        836.0   \n",
       "TTTGTTTGTCTCATTCATGG                       1931.0   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepA_p5_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                           226.0   \n",
       "AAAAAACCCGTAGATAGCCT                           632.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            44.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                           454.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                           301.0   \n",
       "...                                              ...   \n",
       "TTTGTTGGAGAGATGTACGA                           314.0   \n",
       "TTTGTTGGCACAAATACGGG                          1182.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                           301.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                           481.0   \n",
       "TTTGTTTGTCTCATTCATGG                          1368.0   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepB_p5_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                           386.0   \n",
       "AAAAAACCCGTAGATAGCCT                           673.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            39.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                           214.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                           262.0   \n",
       "...                                              ...   \n",
       "TTTGTTGGAGAGATGTACGA                           392.0   \n",
       "TTTGTTGGCACAAATACGGG                          1285.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                           128.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                           803.0   \n",
       "TTTGTTTGTCTCATTCATGG                           974.0   \n",
       "\n",
       "                      pDNA Avana4_010115_1.5Ex_batch0  \\\n",
       "AAAAAAATCCAGCAATGCAG                            305.0   \n",
       "AAAAAACCCGTAGATAGCCT                            398.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             93.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            131.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                            174.0   \n",
       "...                                               ...   \n",
       "TTTGTTGGAGAGATGTACGA                            182.0   \n",
       "TTTGTTGGCACAAATACGGG                            513.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                            886.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                            281.0   \n",
       "TTTGTTTGTCTCATTCATGG                            721.0   \n",
       "\n",
       "                      pDNA Avana4_010115_1.5Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                            300.0   \n",
       "AAAAAACCCGTAGATAGCCT                            396.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             75.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            120.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                            153.0   \n",
       "...                                               ...   \n",
       "TTTGTTGGAGAGATGTACGA                            199.0   \n",
       "TTTGTTGGCACAAATACGGG                            503.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                            792.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                            259.0   \n",
       "TTTGTTTGTCTCATTCATGG                            719.0   \n",
       "\n",
       "                      pDNA Avana4_0101215_0.55Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                              190.0   \n",
       "AAAAAACCCGTAGATAGCCT                              254.0   \n",
       "AAAAAAGAAGAAAAAACCAG                               42.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                               68.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                              109.0   \n",
       "...                                                 ...   \n",
       "TTTGTTGGAGAGATGTACGA                              140.0   \n",
       "TTTGTTGGCACAAATACGGG                              356.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                              530.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                              158.0   \n",
       "TTTGTTTGTCTCATTCATGG                              485.0   \n",
       "\n",
       "                      pDNA Avana4_060115_0.55Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                             176.0   \n",
       "AAAAAACCCGTAGATAGCCT                             230.0   \n",
       "AAAAAAGAAGAAAAAACCAG                              31.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                              90.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                             118.0   \n",
       "...                                                ...   \n",
       "TTTGTTGGAGAGATGTACGA                             118.0   \n",
       "TTTGTTGGCACAAATACGGG                             289.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                             516.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                             173.0   \n",
       "TTTGTTTGTCTCATTCATGG                             454.0   \n",
       "\n",
       "                      pDNA Avana4_060115_1.5Ex_batch0  \\\n",
       "AAAAAAATCCAGCAATGCAG                            324.0   \n",
       "AAAAAACCCGTAGATAGCCT                            421.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             74.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                            135.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                            174.0   \n",
       "...                                               ...   \n",
       "TTTGTTGGAGAGATGTACGA                            172.0   \n",
       "TTTGTTGGCACAAATACGGG                            486.0   \n",
       "TTTGTTTCCTCTGAAGTAGT                            762.0   \n",
       "TTTGTTTCCTCTTCTCGAGG                            268.0   \n",
       "TTTGTTTGTCTCATTCATGG                            702.0   \n",
       "\n",
       "                      pDNA Avana4_060115_1.5Ex_batch1      GENE  \n",
       "AAAAAAATCCAGCAATGCAG                            289.0     SHOC2  \n",
       "AAAAAACCCGTAGATAGCCT                            445.0   NDUFA12  \n",
       "AAAAAAGAAGAAAAAACCAG                             69.0     SDAD1  \n",
       "AAAAAAGCTCAAGAAGGAGG                            135.0    FAM98A  \n",
       "AAAAAAGGCTGTAAAAGCGT                            193.0    ZNF253  \n",
       "...                                               ...       ...  \n",
       "TTTGTTGGAGAGATGTACGA                            141.0      LIPH  \n",
       "TTTGTTGGCACAAATACGGG                            486.0    RPL10L  \n",
       "TTTGTTTCCTCTGAAGTAGT                            771.0     RPS24  \n",
       "TTTGTTTCCTCTTCTCGAGG                            236.0  CRISPLD1  \n",
       "TTTGTTTGTCTCATTCATGG                            666.0  C1orf210  \n",
       "\n",
       "[70994 rows x 1499 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "readcount"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "files = pd.DataFrame(columns=['foldchangefile','col','bffile','prfile','batch'])\n",
    "\n",
    "for batch in samples:\n",
    "    cols = ['GENE']\n",
    "    cell2col = dict()\n",
    "    for cell in samples[batch]:\n",
    "        cell2col[cell] = list()\n",
    "        for repid in samples[batch][cell]:\n",
    "            cell2col[cell].append(len(cols))\n",
    "            cols.append(repid)\n",
    "    cont2col = list()\n",
    "    for repid in controls[batch]:   # to avoid column number change after calculating fc, control is the last\n",
    "        cont2col.append(len(cols))\n",
    "        cols.append(repid)\n",
    "    \n",
    "    readcount_batch_file = 'Avana19Q4_readcount.batch'+str(batch)\n",
    "    readcount[cols].to_csv(readcount_batch_file,sep='\\t')\n",
    "    #print \"./BAGEL_110.py fc -i \" + readcount_batch_file + \" -o \" + readcount_batch_file + \" -c \" + \",\".join(map(str,cont2col))\n",
    "    fcfile = readcount_batch_file + \".foldchange\"\n",
    "    for cell in cell2col:\n",
    "        bffile =  fcfile + \".\" + cell + \".bf\"\n",
    "        prfile =  fcfile + \".\" + cell + \".pr\"\n",
    "        files.loc[cell] = [fcfile,\",\".join(map(str,cell2col[cell])), bffile, prfile, batch]\n",
    "        #print \"./BAGEL_110.py bf -i \" + fcfile + \" -o \" + bffile + \" -c \" + \",\".join(map(str,cell2col[cell])) + \" -e training_ess -n training_noness\"\n",
    "        #print \"./BAGEL_110.py pr -i \" + bffile + \" -o \" + prfile + \" -e training_ess -n training_noness\"\n",
    "    #print\n",
    "    #print\n",
    "    #print"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>foldchangefile</th>\n",
       "      <th>col</th>\n",
       "      <th>bffile</th>\n",
       "      <th>prfile</th>\n",
       "      <th>batch</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>ACH-000601</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange</td>\n",
       "      <td>1,2,3,4</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0006...</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0006...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000496</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange</td>\n",
       "      <td>5,6,7,8</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0004...</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0004...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000222</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange</td>\n",
       "      <td>9,10,11</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0002...</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0002...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000118</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange</td>\n",
       "      <td>12,13,14,15</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0001...</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0001...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000791</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange</td>\n",
       "      <td>16,17,18,19</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0007...</td>\n",
       "      <td>Avana19Q2_readcount.batch0.foldchange.ACH-0007...</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-001791</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange</td>\n",
       "      <td>981,982</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0017...</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0017...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000826</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange</td>\n",
       "      <td>983,984</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0008...</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0008...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000903</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange</td>\n",
       "      <td>985,986</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0009...</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0009...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-000959</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange</td>\n",
       "      <td>987,988</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0009...</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0009...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ACH-001548</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange</td>\n",
       "      <td>989,990</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0015...</td>\n",
       "      <td>Avana19Q2_readcount.batch3.foldchange.ACH-0015...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>689 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   foldchangefile          col  \\\n",
       "ACH-000601  Avana19Q2_readcount.batch0.foldchange      1,2,3,4   \n",
       "ACH-000496  Avana19Q2_readcount.batch0.foldchange      5,6,7,8   \n",
       "ACH-000222  Avana19Q2_readcount.batch0.foldchange      9,10,11   \n",
       "ACH-000118  Avana19Q2_readcount.batch0.foldchange  12,13,14,15   \n",
       "ACH-000791  Avana19Q2_readcount.batch0.foldchange  16,17,18,19   \n",
       "...                                           ...          ...   \n",
       "ACH-001791  Avana19Q2_readcount.batch3.foldchange      981,982   \n",
       "ACH-000826  Avana19Q2_readcount.batch3.foldchange      983,984   \n",
       "ACH-000903  Avana19Q2_readcount.batch3.foldchange      985,986   \n",
       "ACH-000959  Avana19Q2_readcount.batch3.foldchange      987,988   \n",
       "ACH-001548  Avana19Q2_readcount.batch3.foldchange      989,990   \n",
       "\n",
       "                                                       bffile  \\\n",
       "ACH-000601  Avana19Q2_readcount.batch0.foldchange.ACH-0006...   \n",
       "ACH-000496  Avana19Q2_readcount.batch0.foldchange.ACH-0004...   \n",
       "ACH-000222  Avana19Q2_readcount.batch0.foldchange.ACH-0002...   \n",
       "ACH-000118  Avana19Q2_readcount.batch0.foldchange.ACH-0001...   \n",
       "ACH-000791  Avana19Q2_readcount.batch0.foldchange.ACH-0007...   \n",
       "...                                                       ...   \n",
       "ACH-001791  Avana19Q2_readcount.batch3.foldchange.ACH-0017...   \n",
       "ACH-000826  Avana19Q2_readcount.batch3.foldchange.ACH-0008...   \n",
       "ACH-000903  Avana19Q2_readcount.batch3.foldchange.ACH-0009...   \n",
       "ACH-000959  Avana19Q2_readcount.batch3.foldchange.ACH-0009...   \n",
       "ACH-001548  Avana19Q2_readcount.batch3.foldchange.ACH-0015...   \n",
       "\n",
       "                                                       prfile batch  \n",
       "ACH-000601  Avana19Q2_readcount.batch0.foldchange.ACH-0006...     0  \n",
       "ACH-000496  Avana19Q2_readcount.batch0.foldchange.ACH-0004...     0  \n",
       "ACH-000222  Avana19Q2_readcount.batch0.foldchange.ACH-0002...     0  \n",
       "ACH-000118  Avana19Q2_readcount.batch0.foldchange.ACH-0001...     0  \n",
       "ACH-000791  Avana19Q2_readcount.batch0.foldchange.ACH-0007...     0  \n",
       "...                                                       ...   ...  \n",
       "ACH-001791  Avana19Q2_readcount.batch3.foldchange.ACH-0017...     3  \n",
       "ACH-000826  Avana19Q2_readcount.batch3.foldchange.ACH-0008...     3  \n",
       "ACH-000903  Avana19Q2_readcount.batch3.foldchange.ACH-0009...     3  \n",
       "ACH-000959  Avana19Q2_readcount.batch3.foldchange.ACH-0009...     3  \n",
       "ACH-001548  Avana19Q2_readcount.batch3.foldchange.ACH-0015...     3  \n",
       "\n",
       "[689 rows x 5 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Human GENCODE parsing\n",
    "\n",
    "gene2transcript = dict()\n",
    "transcript2exon = dict()\n",
    "transcript2type = dict()\n",
    "transcript2CDS = dict()\n",
    "transcriptlength = dict()\n",
    "gene2location = dict()\n",
    "with open('/home/ekim8/export/project5/gencode.v28lift37.annotation.gtf','r') as fp:\n",
    "    for line in fp:\n",
    "        if line[0] == '#':\n",
    "            continue\n",
    "        linearray = line.rstrip().split(\"\\t\")\n",
    "        #print linearray  # ['chr1', 'HAVANA', 'gene', '3073253', '3074322', '.', '+', '.', 'gene_id \"ENSMUSG00000102693.1\"; gene_type \"TEC\"; gene_name \"RP23-271O17.1\"; level 2; havana_gene \"OTTMUSG00000049935.1\";']\n",
    "        chrom = linearray[0]\n",
    "        featuretype = linearray[2]\n",
    "        start = int(linearray[3])\n",
    "        end = int(linearray[4])\n",
    "        strand = linearray[6]\n",
    "        tags = [x.strip().replace('\"','').split(' ') for x in linearray[8].split(\";\")]\n",
    "        \n",
    "        if featuretype=='exon':\n",
    "            gene=\"\"\n",
    "            transcript=\"\"\n",
    "            #genetype = \"\"\n",
    "            for tag in tags:\n",
    "                if tag[0] == 'gene_name':#'gene_id':\n",
    "                    gene=tag[1].split(\".\")[0]\n",
    "                elif tag[0] == 'transcript_id':\n",
    "                    transcript = tag[1]\n",
    "                elif tag[0] == 'exon_number':\n",
    "                    exon = int(tag[1])\n",
    "                if tag[0] == 'tag' and 'appris_principal' in tag[1]:\n",
    "                    transcript2type[transcript] = tag[1]\n",
    "                \n",
    "                #elif tag[0] == 'tag':\n",
    "            \n",
    "            if gene not in gene2transcript:\n",
    "                gene2transcript[gene]=set()\n",
    "            gene2transcript[gene].add(transcript)\n",
    "            if transcript not in transcript2exon:\n",
    "                transcript2exon[transcript] = dict()\n",
    "                transcriptlength[transcript]=0\n",
    "            transcript2exon[transcript][exon] = (chrom,start,end)\n",
    "            transcriptlength[transcript] += abs(end-start)+1\n",
    "        \n",
    "        \n",
    "        #break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: ('chr17', 1921177, 1921401), 2: ('chr17', 1922479, 1924000)}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript2exon['ENST00000425277.1_1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "# map gene to longest transcript\n",
    "gene2longest = dict()\n",
    "for gene in gene2transcript:\n",
    "    maxt = list(gene2transcript[gene])[0]\n",
    "    maxl = transcriptlength[maxt]\n",
    "    for transcript in gene2transcript[gene]:\n",
    "        if transcript in transcript2type:\n",
    "            if maxl < transcriptlength[transcript]: \n",
    "                maxt = transcript\n",
    "                maxl = transcriptlength[transcript]\n",
    "    gene2longest[gene] = maxt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: ('chr17', 1921177, 1921401), 2: ('chr17', 1922479, 1924000)}"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transcript2exon['ENST00000425277.1_1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CATSPERE\n",
      "ETFRF1\n",
      "PET100\n",
      "OTOL1\n",
      "MACO1\n",
      "DEFB131A\n",
      "RTL3\n",
      "TIMM29\n",
      "MINDY3\n",
      "ARMH2\n",
      "CABCOCO1\n",
      "CALHM6\n",
      "JPT1\n",
      "CNEP1R1\n",
      "JPT2\n",
      "SAXO2\n",
      "PRUNE1\n",
      "N4BP3\n",
      "TIMM29\n",
      "ARMH3\n",
      "SELENOS\n",
      "RTF2\n",
      "VAMP7 Chromosome is different chrX != Ref = chrY\n",
      "AFDN\n",
      "SELENOK\n",
      "SINHCAF\n",
      "DMAC1\n",
      "ATP5MC2\n",
      "NSG2\n",
      "HDGFL2\n",
      "RTL8C\n",
      "MRM2\n",
      "KHDC4\n",
      "VPS26C\n",
      "COA4\n",
      "OTOL1\n",
      "NIPSNAP2\n",
      "MINDY4\n",
      "CLBA1\n",
      "CNEP1R1\n",
      "FBH1\n",
      "TMEM131L\n",
      "CEP83\n",
      "BUD23\n",
      "PNMA8A\n",
      "PPDPFL\n",
      "RAMAC\n",
      "HJV\n",
      "ATP5IF1\n",
      "ELP1\n",
      "EIPR1\n",
      "GUCY1A1\n",
      "SCUBE3\n",
      "YAE1\n",
      "TMEM267\n",
      "HPF1\n",
      "TENT5D\n",
      "HCAR1\n",
      "STPG3\n",
      "BCLAF3\n",
      "TMEM267\n",
      "HIKESHI\n",
      "LMNTD1\n",
      "INSYN2B\n",
      "GRK3\n",
      "LNPK\n",
      "RCC1L\n",
      "EEF1AKMT3\n",
      "ODR4\n",
      "TBXT\n",
      "GSDME\n",
      "ATP5MC1\n",
      "NIPSNAP2\n",
      "C5AR2\n",
      "CASTOR1\n",
      "SELENOK\n",
      "MRNIP\n",
      "RTL10\n",
      "TOX2\n",
      "TERB2\n",
      "SPOUT1\n",
      "TEDC1\n",
      "FAM222B\n",
      "PET100\n",
      "HCAR3\n",
      "TENT5C\n",
      "UTP25\n",
      "CRIPAK\n",
      "SHISAL2A\n",
      "DEUP1\n",
      "DHFR2\n",
      "MESD\n",
      "FAM241B\n",
      "CCDC198\n",
      "VXN\n",
      "INKA2\n",
      "AKAIN1\n",
      "TEX45\n",
      "COQ8A\n",
      "MESD\n",
      "PINX1\n",
      "MYDGF\n",
      "CRYBG1\n",
      "RTF2\n",
      "TMEM273\n",
      "MIGA2\n",
      "EEF1AKMT2\n",
      "CALHM6\n",
      "RTRAF\n",
      "KIRREL1\n",
      "PAXX\n",
      "JCAD\n",
      "PLPPR3\n",
      "MYMK\n",
      "NOP53\n",
      "RAB5IF\n",
      "REXO5\n",
      "VEGFD\n",
      "JCAD\n",
      "WDCP\n",
      "ATP5F1B\n",
      "RCC1L\n",
      "PIP4P2\n",
      "CSKMT\n",
      "SELENOI\n",
      "DELE1\n",
      "ABRAXAS2\n",
      "MARF1\n",
      "ATP5MF\n",
      "CLTRN\n",
      "LLCFC1\n",
      "CPLANE1\n",
      "RETREG1\n",
      "SPEM2\n",
      "ATP5MC1\n",
      "CABCOCO1\n",
      "CYREN\n",
      "TESMIN\n",
      "ATP5MF\n",
      "CLTRN\n",
      "TCTEX1D4\n",
      "RXYLT1\n",
      "NKAPD1\n",
      "N4BP3\n",
      "RTL5\n",
      "DNAAF4\n",
      "RIPOR3\n",
      "RIPOR1\n",
      "KNOP1\n",
      "TMEM94\n",
      "MCEMP1\n",
      "PRKN\n",
      "CFAP299\n",
      "CFAP97D1\n",
      "SELENOH\n",
      "N4BP3\n",
      "KNOP1\n",
      "CAVIN3\n",
      "STUM\n",
      "HYKK\n",
      "INAVA\n",
      "ABRAXAS1\n",
      "LHFPL6\n",
      "SPRY3 Chromosome is different chrX != Ref = chrY\n",
      "CIP2A\n",
      "SINHCAF\n",
      "CBLL2\n",
      "PTPN20\n",
      "BICRAL\n",
      "INKA1\n",
      "MAIP1\n",
      "STPG3\n",
      "CRYBG1\n",
      "YAE1\n",
      "PGGHG\n",
      "MAP3K21\n",
      "TCIM\n",
      "KIAA1107\n",
      "RTL8A\n",
      "TMEM250\n",
      "CALHM5\n",
      "ABRAXAS2\n",
      "TENT5B\n",
      "CENPX\n",
      "FAM243A\n",
      "MYDGF\n",
      "RETREG3\n",
      "DGLUCY\n",
      "ARMH1\n",
      "LHFPL6\n",
      "STN1\n",
      "KHDC4\n",
      "BABAM2\n",
      "SHLD1\n",
      "COA4\n",
      "UQCC3\n",
      "HASPIN\n",
      "REX1BD\n",
      "CFAP97D1\n",
      "SELENOO\n",
      "GUCY1B1\n",
      "AFDN\n",
      "JPT1\n",
      "RTL6\n",
      "HDGFL2\n",
      "LTO1\n",
      "ADM5\n",
      "TMEM268\n",
      "REX1BD\n",
      "CRACR2A\n",
      "INSYN1\n",
      "EEF1AKMT3\n",
      "PLPBP\n",
      "PRKN\n",
      "VPS26C\n",
      "SHISAL1\n",
      "COA3\n",
      "RETREG2\n",
      "TEDC1\n",
      "HCAR1\n",
      "EIPR1\n",
      "MOSMO\n",
      "CYBC1\n",
      "ATP5F1A\n",
      "SCUBE3\n",
      "HPF1\n",
      "TEX47\n",
      "GALNT17\n",
      "MAIP1\n",
      "TUT7\n",
      "HPF1\n",
      "CCAR2\n",
      "CCDC32\n",
      "C2CD6\n",
      "CARMIL3\n",
      "ATP5PO\n",
      "PAK5\n",
      "FAM222B\n",
      "COP1\n",
      "UFD1\n",
      "CNOT9\n",
      "ZBED1 Chromosome is different chrX != Ref = chrY\n",
      "MRM2\n",
      "HDGFL3\n",
      "LMNTD1\n",
      "RMC1\n",
      "PCARE\n",
      "TEDC1\n",
      "TENT5A\n",
      "MESD\n",
      "CASTOR2\n",
      "BCLAF3\n",
      "METTL26\n",
      "SELENOK\n",
      "MAJIN\n",
      "PNMA8A\n",
      "ATP5PO\n",
      "KNL1\n",
      "MARF1\n",
      "PGGHG\n",
      "RFLNA\n",
      "ZUP1\n",
      "PPDPFL\n",
      "WHRN\n",
      "HCAR3\n",
      "RETREG1\n",
      "ETFBKMT\n",
      "ATP5PB\n",
      "BMT2\n",
      "FAM241B\n",
      "LTO1\n",
      "TMEM35A\n",
      "PHETA2\n",
      "SLIRP\n",
      "PAK5\n",
      "KIRREL1\n",
      "SHLD2\n",
      "FAM243A\n",
      "ETFRF1\n",
      "NEMP1\n",
      "GSDME\n",
      "CNOT9\n",
      "CAVIN3\n",
      "SELENOW\n",
      "ATP5F1C\n",
      "GUCY1A1\n",
      "CEP83\n",
      "FMC1\n",
      "HDGFL2\n",
      "MRTFB\n",
      "MRTFB\n",
      "JCAD\n",
      "CSKMT\n",
      "ADM5\n",
      "TEDC2\n",
      "TCTEX1D4\n",
      "DNAAF4\n",
      "NDUFAF8\n",
      "PLPPR3\n",
      "MYDGF\n",
      "CARMIL2\n",
      "HYKK\n",
      "GRAMD2A\n",
      "ATP5MC3\n",
      "BICDL2\n",
      "NKAPD1\n",
      "JCAD\n",
      "UBE2Q2L\n",
      "OGFOD3\n",
      "STMP1\n",
      "DHRSX Chromosome is different chrX != Ref = chrY\n",
      "ARMH4\n",
      "MOSMO\n",
      "FYB2\n",
      "MAJIN\n",
      "KYAT3\n",
      "PIP4P2\n",
      "FAM243A\n",
      "MICALCL\n",
      "CRACR2B\n",
      "PLPBP\n",
      "C5AR2\n",
      "ADA2\n",
      "COLGALT1\n",
      "ARHGAP45\n",
      "BICRA\n",
      "CNEP1R1\n",
      "PHETA1\n",
      "CIAO3\n",
      "DMAC2\n",
      "RTL10\n",
      "ALPG\n",
      "ATP5MC3\n",
      "BMT2\n",
      "SHLD1\n",
      "REXO5\n",
      "GUCY1A1\n",
      "CCDC32\n",
      "ATP5MPL\n",
      "INSYN2\n",
      "NOP53\n",
      "KIRREL1\n",
      "MUCL3\n",
      "RIPOR3\n",
      "RIPOR1\n",
      "RTF2\n",
      "MAJIN\n",
      "VPS35L\n",
      "STPG3\n",
      "ZUP1\n",
      "ABRAXAS2\n",
      "NAA80\n",
      "PRUNE1\n",
      "INSYN1\n",
      "ATP5F1D\n",
      "CAVIN1\n",
      "REX1BD\n",
      "ARMH4\n",
      "LARGE2\n",
      "GRAMD2B\n",
      "ISM1\n",
      "FYB2\n",
      "ODAPH\n",
      "ALKAL2\n",
      "NDUFAF8\n",
      "TMEM121B\n",
      "SMIM29\n",
      "PIMREG\n",
      "YJU2\n",
      "VPS26C\n",
      "ADM5\n",
      "ACP4\n",
      "DMAC2\n",
      "TTI2\n",
      "IQCN\n",
      "MINDY3\n",
      "SRARP\n",
      "CCDC9B\n",
      "TLNRD1\n",
      "SPINDOC\n",
      "STPG4\n",
      "CYBC1\n",
      "RTL5\n",
      "YJU2\n",
      "MRE11\n",
      "HIKESHI\n",
      "DELE1\n",
      "TRMT9B\n",
      "KYAT1\n",
      "MAJIN\n",
      "KNOP1\n",
      "PAXX\n",
      "NEMP1\n",
      "UBE2Q2L\n",
      "FDX2\n",
      "MOSMO\n",
      "TEPSIN\n",
      "MIGA2\n",
      "PNMA8A\n",
      "HASPIN\n",
      "FMC1\n",
      "COA3\n",
      "SELENON\n",
      "TRIR\n",
      "TESMIN\n",
      "RAB5IF\n",
      "PTPA\n",
      "PEAK3\n",
      "ATP5F1E\n",
      "RAB5IF\n",
      "CAVIN4\n",
      "C2CD6\n",
      "FAM92A\n",
      "SELENOS\n",
      "CIAO2B\n",
      "VIRMA\n",
      "VEGFD\n",
      "ESS2\n",
      "CAVIN1\n",
      "ACP4\n",
      "TERB2\n",
      "PEAK3\n",
      "LTO1\n",
      "CDHR2\n",
      "TMEM94\n",
      "TENT5C\n",
      "CAVIN2\n",
      "ELP1\n",
      "TEX45\n",
      "OGFOD3\n",
      "SPACA9\n",
      "PIMREG\n",
      "CAVIN2\n",
      "OGFOD3\n",
      "ZBED1 Chromosome is different chrX != Ref = chrY\n",
      "EMC6\n",
      "LLCFC1\n",
      "JAML\n",
      "KAT14\n",
      "SPART\n",
      "ABRAXAS1\n",
      "STPG4\n",
      "INKA2\n",
      "CSF2RA Chromosome is different chrX != Ref = chrY\n",
      "ALPG\n",
      "CEP128\n",
      "ODAPH\n",
      "ZBED1 Chromosome is different chrX != Ref = chrY\n",
      "PNMA8B\n",
      "STMP1\n",
      "CFAP300\n",
      "RTRAF\n",
      "INKA1\n",
      "FAM241B\n",
      "COP1\n",
      "NIPSNAP2\n",
      "ATP5F1D\n",
      "FNDC11\n",
      "GRK2\n",
      "IL3RA Chromosome is different chrX != Ref = chrY\n",
      "COLGALT1\n",
      "OGA\n",
      "ENTR1\n",
      "RCC1L\n",
      "TRARG1\n",
      "COPS9\n",
      "TMEM131L\n",
      "FAAP24\n",
      "ATP5ME\n",
      "TMEM273\n",
      "SELENOO\n",
      "RIPOR1\n",
      "MCRIP2\n",
      "EIPR1\n",
      "AJM1\n",
      "TLNRD1\n",
      "RMC1\n",
      "TLNRD1\n",
      "ATP5MGL\n",
      "FAM92A\n",
      "PRUNE1\n",
      "CENPX\n",
      "TENT2\n",
      "RTL8B\n",
      "FBH1\n",
      "CCDC9B\n",
      "VIRMA\n",
      "ATP5PD\n",
      "PAK5\n",
      "CARMIL1\n",
      "RIPOR1\n",
      "SHISAL2B\n",
      "TFAP2D\n",
      "REX1BD\n",
      "N4BP3\n",
      "MRM3\n",
      "ATP5F1A\n",
      "MAP3K21\n",
      "ATP5MPL\n",
      "TENT5B\n",
      "TEX47\n",
      "ATP5MC2\n",
      "ARMH1\n",
      "MINDY1\n",
      "SELENOH\n",
      "TENT5A\n",
      "STIMATE\n",
      "ECPAS\n",
      "YAE1\n",
      "ARMH1\n",
      "VAMP7 Chromosome is different chrX != Ref = chrY\n",
      "SCUBE3\n",
      "VIRMA\n",
      "HASPIN\n",
      "ATP5MG\n",
      "SLIRP\n",
      "YJU2\n",
      "COA3\n",
      "COQ8B\n",
      "CEP83\n",
      "MAP3K21\n",
      "ARHGAP45\n",
      "IFT43\n",
      "RFLNA\n",
      "DHFR2\n",
      "MICALCL\n",
      "SPINDOC\n",
      "ATP5PO\n",
      "PHETA2\n",
      "ISM1\n",
      "LLCFC1\n",
      "PIP4P1\n",
      "CGAS\n",
      "KIRREL1\n",
      "FAM241A\n",
      "CNEP1R1\n",
      "CYREN\n",
      "UQCC3\n",
      "INSYN2B\n",
      "SHISAL1\n",
      "TMEM268\n",
      "MCUB\n",
      "SPART\n",
      "SNORC\n",
      "CFAP300\n",
      "MYMK\n",
      "CFAP298\n",
      "MCEMP1\n",
      "TENT5B\n",
      "PRR23A\n",
      "SINHCAF\n",
      "TBXT\n",
      "TENT2\n",
      "PLPPR3\n",
      "HJV\n",
      "REC114\n",
      "ABHD17C\n",
      "TENT5C\n",
      "LRMDA\n",
      "SPACA9\n",
      "GDPGP1\n",
      "ETFRF1\n",
      "CPLANE2\n",
      "CCAR2\n",
      "PAXX\n",
      "TEPSIN\n",
      "RTRAF\n",
      "CARMIL3\n",
      "TMEM35A\n",
      "LRMDA\n",
      "MRM2\n",
      "RETREG2\n",
      "ATP5PB\n",
      "VEGFD\n",
      "AKAIN1\n",
      "COQ8A\n",
      "SPATA48\n",
      "DEUP1\n",
      "DHFR2\n",
      "COP1\n",
      "JHY\n",
      "CAVIN4\n",
      "TMEM267\n",
      "RUBCNL\n",
      "SHISAL2A\n",
      "PHETA1\n",
      "RTL10\n",
      "MACO1\n",
      "ATP5MF\n",
      "ESS2\n",
      "PPP4R3A\n",
      "CRACR2A\n",
      "PIP4K2B\n",
      "VPS26C\n",
      "ATP5PF\n",
      "KIAA1107\n",
      "COQ8B\n",
      "SHISAL2B\n",
      "MCRIP2\n",
      "NEPRO\n",
      "PET100\n",
      "PYCR3\n",
      "CCDC9B\n",
      "FAM92A\n",
      "COA3\n",
      "SELENOV\n",
      "LRRC29\n",
      "SLC49A3\n",
      "ATP5PF\n",
      "RTL8C\n",
      "CIAO3\n",
      "ACP4\n",
      "BICDL2\n",
      "ALPG\n",
      "CFAP299\n",
      "ATP5MG\n",
      "FAM214A\n",
      "SLIRP\n",
      "SPRY3 Chromosome is different chrX != Ref = chrY\n",
      "ATP5F1E\n",
      "PIMREG\n",
      "INKA1\n",
      "GATD1\n",
      "GON7\n",
      "ECPAS\n",
      "RTL5\n",
      "TBXT\n",
      "ODR4\n",
      "GDPGP1\n",
      "ERG28\n",
      "SELENOV\n",
      "CRIPAK\n",
      "CRIPAK\n",
      "CYBC1\n",
      "LRMDA\n",
      "OTOL1\n",
      "DMAC2\n",
      "RETREG2\n",
      "ATP5F1E\n",
      "MCRIP1\n",
      "RHEX\n",
      "RXYLT1\n",
      "FAM222B\n",
      "DOHH\n",
      "DELE1\n",
      "FBH1\n",
      "JHY\n",
      "DNAAF4\n",
      "DMAC2\n",
      "MTREX\n",
      "DMAC1\n",
      "PCARE\n",
      "CALHM5\n",
      "MRE11\n",
      "NKAPD1\n",
      "ATP5PD\n",
      "SHISAL1\n",
      "OGA\n",
      "TOX2\n",
      "CRYBG1\n",
      "GUCY1B1\n",
      "ENTR1\n",
      "UBE2Q2L\n",
      "RXYLT1\n",
      "SPOUT1\n",
      "TRARG1\n",
      "BMT2\n",
      "MCRIP1\n",
      "RTL8B\n",
      "GRK3\n",
      "GRAMD2B\n",
      "ARMH4\n",
      "NSG2\n",
      "LARGE1\n",
      "MCEMP1\n",
      "TRMT9B\n",
      "ATP5MD\n",
      "METTL26\n",
      "CASTOR1\n",
      "STN1\n",
      "SQOR\n",
      "TRARG1\n",
      "COLGALT1\n",
      "JAML\n",
      "MCUB\n",
      "UTP25\n",
      "CIAO2B\n",
      "CCDC198\n",
      "BICRAL\n",
      "GRK2\n",
      "CIP2A\n",
      "HIKESHI\n",
      "RAMAC\n",
      "GSDME\n",
      "EEF1AKMT3\n",
      "C5AR2\n",
      "SPACA9\n",
      "INKA2\n",
      "ATP5IF1\n",
      "TMEM94\n",
      "C15orf53\n",
      "CASTOR3\n",
      "RETREG1\n",
      "MRE11\n",
      "PIP4P2\n",
      "ARMH3\n",
      "TENT5A\n",
      "INKA2\n",
      "ATP5F1B\n",
      "ATP5MC2\n",
      "TEDC1\n",
      "ALPG\n",
      "KAT14\n",
      "FDX2\n",
      "RTL10\n",
      "RTL8B\n",
      "PPP4R3A\n",
      "METTL26\n",
      "CCAR2\n",
      "GDPGP1\n",
      "CALHM5\n",
      "AJM1\n",
      "CALHM6\n",
      "CLBA1\n",
      "BICRA\n",
      "CYBC1\n",
      "RETREG3\n",
      "MINDY1\n",
      "ARHGAP45\n",
      "SHISAL2A\n",
      "MUCL3\n",
      "LRRC29\n",
      "DOHH\n",
      "COQ8A\n",
      "ACP4\n",
      "RTL3\n",
      "TMEM121B\n",
      "CSKMT\n",
      "TMEM131L\n",
      "IFT43\n",
      "SHISAL2B\n",
      "AJM1\n",
      "TMEM94\n",
      "LARGE2\n",
      "INSYN2\n",
      "TEPSIN\n",
      "CCDC198\n",
      "RIPOR3\n",
      "PHETA1\n",
      "PRKN\n",
      "FNDC11\n",
      "AJM1\n",
      "MRNIP\n",
      "PAXX\n",
      "TMEM273\n",
      "NEPRO\n",
      "DMAC1\n",
      "CPLANE2\n",
      "MINDY1\n",
      "MARF1\n",
      "MTERF2\n",
      "ECPAS\n",
      "RTL6\n",
      "GUCY1B1\n",
      "BICRAL\n",
      "KYAT1\n",
      "HCAR3\n",
      "ATP5F1D\n",
      "UFD1\n",
      "INKA1\n",
      "RHEX\n",
      "PPP4R3A\n",
      "FAM214A\n",
      "PIP4P1\n",
      "SPOUT1\n",
      "PIP4P1\n",
      "HDHD5\n",
      "GATD1\n",
      "TENM4\n",
      "FNDC10\n",
      "DGLUCY\n",
      "GALNT17\n",
      "CABCOCO1\n",
      "TRMT9B\n",
      "TERB2\n",
      "CATSPERE\n",
      "ATP5F1A\n",
      "WDCP\n",
      "SNORC\n",
      "TCIM\n",
      "CRACR2B\n",
      "ANGPTL8\n",
      "SELENOV\n",
      "COPS9\n",
      "MAIP1\n",
      "OGFOD3\n",
      "EMC6\n",
      "TEDC2\n",
      "CAVIN3\n",
      "SNORC\n",
      "LARGE1\n",
      "C2CD6\n",
      "DHRSX Chromosome is different chrX != Ref = chrY\n",
      "DGLUCY\n",
      "NDUFAF8\n",
      "TCIM\n",
      "ATP5ME\n",
      "CALHM5\n",
      "DOHH\n",
      "SELENOH\n",
      "PTPA\n",
      "TRIR\n",
      "OGA\n",
      "INSYN2\n",
      "VEGFD\n",
      "CASTOR2\n",
      "KAT14\n",
      "SELENOM\n",
      "ALKAL1\n",
      "RCC1L\n",
      "ABHD17C\n",
      "MESD\n",
      "FNDC10\n",
      "IFT43\n",
      "MRPL58\n",
      "ABHD17C\n",
      "ALKAL1\n",
      "MIGA2\n",
      "GUCY1B1\n",
      "SPOUT1\n",
      "INSYN2\n",
      "SLIRP\n",
      "MRTFB\n",
      "MYMK\n",
      "PLPBP\n",
      "ATP5F1D\n",
      "COP1\n",
      "COQ8B\n",
      "CBLL2\n",
      "GPAT4\n",
      "PPDPFL\n",
      "INSYN1\n",
      "KNL1\n",
      "TEX46\n",
      "BICDL1\n",
      "SLC49A3\n",
      "MRPL58\n",
      "ENTR1\n",
      "ADA2\n",
      "CRACR2A\n",
      "CARMIL3\n",
      "COLGALT1\n",
      "NDUFAF8\n",
      "TSPOAP1\n",
      "FMC1\n",
      "CRYBG1\n",
      "ADA2\n",
      "PCARE\n",
      "METTL27\n",
      "TOX2\n",
      "KAT14\n",
      "FAM241A\n",
      "BICDL1\n",
      "CFAP300\n",
      "TMEM121B\n",
      "ERG28\n",
      "SQOR\n",
      "EEF1AKMT3\n",
      "GALNT17\n",
      "TMEM273\n",
      "TMEM35A\n",
      "RTL6\n",
      "RHEX\n",
      "CIAO2A\n",
      "CARMIL2\n",
      "CLTRN\n",
      "IQCN\n",
      "ARMH2\n",
      "GDPGP1\n",
      "RETREG2\n",
      "MINDY4\n",
      "SHISAL2B\n",
      "SCUBE3\n",
      "TSPOAP1\n",
      "HDHD5\n",
      "RTL5\n",
      "TLNRD1\n",
      "CAVIN1\n",
      "CEP128\n",
      "MAP3K21\n",
      "MICALCL\n",
      "VPS35L\n",
      "FYB2\n",
      "ALKAL2\n",
      "LARGE2\n",
      "PINX1\n",
      "TENT4B\n",
      "TIMM29\n",
      "TMEM268\n",
      "INAVA\n",
      "PYCR3\n",
      "CAVIN2\n",
      "TSPOAP1\n",
      "CPLANE1\n",
      "BICDL1\n",
      "FDX2\n",
      "ABRAXAS2\n",
      "RUBCNL\n",
      "GATD1\n",
      "PNMA8B\n",
      "GON7\n",
      "SMIM29\n",
      "GRK3\n",
      "FBH1\n",
      "PNMA8B\n",
      "TEPSIN\n",
      "SLC49A3\n",
      "BUD23\n",
      "CARMIL1\n",
      "TEX46\n",
      "UTP25\n",
      "ADA2\n",
      "SELENOP\n",
      "VPS35L\n",
      "STIMATE\n",
      "GATD1\n",
      "BICRAL\n",
      "TENT4B\n",
      "ARMH2\n",
      "SPART\n",
      "TEX45\n",
      "RETREG3\n",
      "NOP53\n",
      "NSG2\n",
      "CSF2RA Chromosome is different chrX != Ref = chrY\n",
      "HYKK\n",
      "CEMIP2\n",
      "CIAO2A\n",
      "PIP4P2\n",
      "HDGFL3\n",
      "LNPK\n",
      "PTPN20\n",
      "EIPR1\n",
      "ISM1\n",
      "HIKESHI\n",
      "STIMATE\n",
      "SPACA9\n",
      "RTL4\n",
      "PRR23A\n",
      "TTI2\n",
      "JHY\n",
      "GUCY1A1\n",
      "ATP5F1C\n",
      "PGGHG\n",
      "BABAM2\n",
      "WHRN\n",
      "ESS2\n",
      "PEAK3\n",
      "CIAO2A\n",
      "TENT5A\n",
      "TEX47\n",
      "ALKAL2\n",
      "UBE2Q2L\n",
      "SPEM2\n",
      "PJVK\n",
      "CRACR2A\n",
      "ATP5PO\n",
      "SPRY3 Chromosome is different chrX != Ref = chrY\n",
      "CLBA1\n",
      "TSPOAP1\n",
      "CIAO3\n",
      "JHY\n",
      "VXN\n",
      "ALKAL1\n",
      "INSYN1\n",
      "INAVA\n",
      "TENT4B\n",
      "BICDL1\n",
      "LRRC29\n",
      "SELENOV\n",
      "ALKAL2\n",
      "TENM4\n",
      "CCAR2\n",
      "RAB5IF\n",
      "MINDY2\n",
      "RTL6\n",
      "ABRAXAS1\n",
      "LLCFC1\n",
      "CFAP97D1\n",
      "INAVA\n",
      "TRARG1\n",
      "RUBCNL\n",
      "RTL3\n",
      "CASTOR1\n",
      "FMC1\n",
      "ENTR1\n",
      "PGGHG\n",
      "GPAT4\n",
      "TENT5C\n",
      "SELENON\n",
      "SPINDOC\n",
      "NAA80\n",
      "TMEM270\n",
      "HJV\n",
      "LMNTD1\n",
      "ATP5PD\n",
      "IFT43\n",
      "RELCH\n",
      "RELCH\n",
      "MCUB\n",
      "MRNIP\n",
      "KYAT3\n",
      "ATP5ME\n",
      "CASTOR1\n",
      "FAM222B\n",
      "MRM3\n",
      "NOP53\n",
      "SRARP\n",
      "RTL8A\n",
      "GRAMD2A\n",
      "MOSMO\n",
      "CGAS\n",
      "SHLD2\n",
      "BUD23\n",
      "ATP5F1C\n",
      "RTL8B\n",
      "KYAT1\n",
      "SELENOP\n",
      "MRPL58\n",
      "CPLANE2\n",
      "ISM1\n",
      "MINDY2\n",
      "TFAP2D\n",
      "ATP5MGL\n",
      "SELENOI\n",
      "FAM241A\n",
      "METTL26\n",
      "C2CD6\n",
      "NEPRO\n",
      "TMEM270\n",
      "PIP4K2B\n",
      "TMEM270\n",
      "MTREX\n",
      "SELENOW\n",
      "CARMIL3\n",
      "TEDC2\n",
      "CFAP300\n",
      "ALKAL1\n",
      "TTI2\n",
      "FNDC10\n",
      "CLBA1\n",
      "KYAT1\n",
      "COQ8B\n",
      "MICALCL\n",
      "PET100\n",
      "SPRY3 Chromosome is different chrX != Ref = chrY\n",
      "FAM214A\n",
      "AFG1L\n",
      "SELENOT\n",
      "IL3RA Chromosome is different chrX != Ref = chrY\n",
      "COA4\n",
      "TESMIN\n",
      "RXYLT1\n",
      "CIP2A\n",
      "UFD1\n",
      "CAVIN2\n",
      "BICRA\n",
      "TUT7\n",
      "GRK2\n",
      "EMC6\n",
      "GRAMD2A\n",
      "ANGPTL8\n",
      "IL3RA Chromosome is different chrX != Ref = chrY\n",
      "METTL27\n",
      "DHRSX Chromosome is different chrX != Ref = chrY\n",
      "SELENOS\n",
      "STN1\n",
      "SELENON\n",
      "ETFBKMT\n",
      "SRARP\n",
      "CFAP298\n",
      "SPATA48\n",
      "COA4\n",
      "GON7\n",
      "CEMIP2\n",
      "PRR23A\n",
      "REC114\n",
      "NSG2\n",
      "KHDC4\n",
      "ZBED1 Chromosome is different chrX != Ref = chrY\n",
      "FNDC10\n",
      "PHETA2\n",
      "JPT2\n",
      "COPS9\n",
      "DOHH\n",
      "TCTEX1D4\n",
      "RAMAC\n",
      "REC114\n",
      "MRTFA\n",
      "RHEX\n",
      "PRKN\n",
      "PYCR3\n",
      "CNOT9\n",
      "ANGPTL8\n",
      "TRMT9B\n",
      "CIAO2B\n",
      "TRIR\n",
      "NAA80\n",
      "SRARP\n",
      "MRM3\n",
      "ATP5F1B\n",
      "SPATA48\n",
      "KNL1\n",
      "CIP2A\n",
      "SMIM29\n",
      "ZUP1\n",
      "MRTFA\n",
      "ATP5PB\n",
      "ARHGAP45\n",
      "CARMIL2\n",
      "BABAM2\n",
      "GON7\n",
      "CEMIP2\n",
      "HCAR3\n",
      "LARGE1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRAMD2B\n",
      "TMEM121B\n",
      "MUCL3\n",
      "MCEMP1\n",
      "PPP4R3A\n",
      "CDHR2\n",
      "ATP5PB\n",
      "PIP4K2B\n",
      "DEPP1\n",
      "TIMM29\n",
      "RMC1\n",
      "MINDY4\n",
      "TENT4B\n",
      "BCLAF3\n",
      "DEPP1\n",
      "EEF1AKMT2\n",
      "C5AR2\n",
      "RTL4\n",
      "ATP5MC3\n",
      "CAVIN1\n",
      "SHLD1\n",
      "FAM243A\n",
      "MCUB\n",
      "TENT5D\n",
      "ATP5PF\n",
      "INSYN2B\n",
      "CASTOR3\n",
      "SQOR\n",
      "TESMIN\n",
      "RFLNA\n",
      "MINDY2\n",
      "DHFR2\n",
      "EEF1AKMT2\n",
      "MRPL58\n",
      "IQCN\n",
      "RTL9\n",
      "MCRIP1\n",
      "STUM\n",
      "CFAP299\n",
      "FAM214A\n",
      "TRIR\n",
      "ATP5IF1\n",
      "STUM\n",
      "ABHD17C\n",
      "JPT2\n",
      "JAML\n",
      "HCAR1\n",
      "COPS9\n",
      "FAAP24\n",
      "TUT7\n",
      "PHETA1\n",
      "VXN\n",
      "GPAT4\n",
      "ARMH4\n",
      "METTL27\n",
      "CYREN\n",
      "VAMP7 Chromosome is different chrX != Ref = chrY\n",
      "GRAMD2A\n",
      "SELENOT\n",
      "AFG1L\n",
      "ATP5MGL\n",
      "ATP5MG\n",
      "SELENOK\n",
      "ODR4\n",
      "C15orf53\n",
      "RUBCNL\n",
      "UFD1\n",
      "FDX2\n",
      "C15orf53\n",
      "CFAP298\n",
      "REC114\n",
      "STUM\n",
      "CENPX\n",
      "ARMH3\n",
      "WDCP\n",
      "SELENOO\n",
      "CARMIL2\n",
      "OTULINL\n",
      "STPG3\n",
      "PNMA8A\n",
      "MCRIP1\n",
      "MRTFA\n",
      "GRAMD2B\n",
      "PIP4K2B\n",
      "ATP5MPL\n",
      "CABCOCO1\n",
      "CEP128\n",
      "PJVK\n",
      "SQOR\n",
      "CRACR2B\n",
      "COQ8A\n",
      "MRE11\n",
      "CBLL2\n",
      "ATP5MD\n",
      "SELENOT\n",
      "CAVIN4\n",
      "ATP5MG\n",
      "CCDC32\n",
      "SINHCAF\n",
      "CSF2RA Chromosome is different chrX != Ref = chrY\n",
      "WHRN\n",
      "ODR4\n",
      "ETFRF1\n",
      "RTF2\n",
      "HDGFL3\n",
      "OTULINL\n",
      "CIAO2B\n",
      "TCIM\n",
      "TMEM250\n",
      "CPLANE1\n",
      "DEPP1\n",
      "SPATA48\n",
      "ECPAS\n",
      "DNAAF4\n",
      "CPLANE1\n",
      "CIAO3\n",
      "SPEM2\n",
      "SELENOM\n",
      "HASPIN\n",
      "TEX47\n",
      "PRR23A\n",
      "MIGA2\n",
      "PARPBP\n",
      "ZUP1\n",
      "NKAPD1\n",
      "PTPN20\n",
      "SAXO2\n",
      "CFAP97D1\n",
      "PARPBP\n",
      "AFDN\n",
      "SELENOM\n",
      "RTL4\n",
      "CEMIP2\n",
      "RTL4\n",
      "DMAC1\n",
      "CGAS\n",
      "PJVK\n",
      "EEF1AKMT2\n",
      "AFG1L\n",
      "RELCH\n",
      "MRTFA\n",
      "TUT4\n",
      "TBXT\n",
      "ARMH3\n",
      "SELENOH\n",
      "TUT4\n",
      "RTRAF\n",
      "ODAPH\n",
      "MACO1\n",
      "TEX46\n",
      "WDCP\n",
      "MYDGF\n",
      "ATP5PD\n",
      "HJV\n",
      "AFDN\n",
      "TEDC2\n",
      "SELENOM\n",
      "KIAA1107\n",
      "MYMK\n",
      "MAIP1\n",
      "ARMH2\n",
      "TENT2\n",
      "FAM92A\n",
      "CASTOR3\n",
      "BICDL2\n",
      "ATP5PF\n",
      "PJVK\n",
      "CATSPERE\n",
      "TERB2\n",
      "FNDC11\n",
      "MCRIP2\n",
      "ATP5F1A\n",
      "CRIPAK\n",
      "ATP5IF1\n",
      "TEX46\n",
      "ADM5\n",
      "VIRMA\n",
      "NEMP1\n",
      "TENT5B\n",
      "PLPPR3\n",
      "ATP5MF\n",
      "ESS2\n",
      "LNPK\n",
      "SPART\n",
      "KHDC4\n",
      "HDGFL3\n",
      "RIPOR3\n",
      "SELENOW\n",
      "SLC49A3\n",
      "HDHD5\n",
      "UQCC3\n",
      "TMEM131L\n",
      "RTL3\n",
      "MCRIP2\n",
      "BABAM2\n",
      "TENT5D\n",
      "REXO5\n",
      "ELP1\n",
      "CIAO2A\n",
      "MTERF2\n",
      "PIP4P1\n",
      "NEPRO\n",
      "MINDY3\n",
      "TENM4\n",
      "CPLANE2\n",
      "BMT2\n",
      "SELENOT\n",
      "PPDPFL\n",
      "DGLUCY\n",
      "DEUP1\n",
      "ETFBKMT\n",
      "HDGFL2\n",
      "MRTFB\n",
      "SNORC\n",
      "PNMA8B\n",
      "TOX2\n",
      "LTO1\n",
      "MRM2\n",
      "INSYN2B\n",
      "LRMDA\n",
      "SHISAL1\n",
      "SELENOS\n",
      "CEP128\n",
      "ATP5MC1\n",
      "KYAT3\n",
      "FYB2\n",
      "TUT7\n",
      "ATP5F1B\n",
      "CCDC32\n",
      "STPG4\n",
      "CSF2RA Chromosome is different chrX != Ref = chrY\n",
      "RTL9\n",
      "RTL9\n",
      "FAAP24\n",
      "BICRA\n",
      "CAVIN3\n",
      "SPINDOC\n",
      "SAXO2\n",
      "CARMIL1\n",
      "ETFBKMT\n",
      "MRM3\n",
      "STIMATE\n",
      "TMEM270\n",
      "MINDY2\n",
      "METTL27\n",
      "TFAP2D\n",
      "ABRAXAS1\n",
      "STN1\n",
      "MTREX\n",
      "PARPBP\n",
      "OGA\n",
      "MRNIP\n",
      "MINDY3\n",
      "UTP25\n",
      "JPT1\n",
      "ANGPTL8\n",
      "TMEM250\n",
      "SHISAL2A\n",
      "ATP5F1C\n",
      "CRACR2B\n",
      "FNDC11\n",
      "TENT2\n",
      "HYKK\n",
      "TMEM250\n",
      "PTPA\n",
      "HCAR1\n",
      "ODAPH\n",
      "TMEM267\n",
      "LNPK\n",
      "PYCR3\n",
      "STMP1\n",
      "RFLNA\n",
      "BICDL2\n",
      "FAAP24\n",
      "RELCH\n",
      "SELENOI\n",
      "ATP5ME\n",
      "NAA80\n",
      "DHRSX Chromosome is different chrX != Ref = chrY\n",
      "TTI2\n",
      "SELENOP\n",
      "GPAT4\n",
      "ARMH1\n",
      "CGAS\n",
      "CYREN\n",
      "ATP5MPL\n",
      "PINX1\n",
      "ATP5MD\n",
      "RMC1\n",
      "CSKMT\n",
      "ERG28\n",
      "SMIM29\n",
      "SPEM2\n",
      "GRK3\n",
      "GRK2\n",
      "IQCN\n",
      "JAML\n",
      "KNOP1\n",
      "LHFPL6\n",
      "SELENOO\n",
      "DELE1\n",
      "DEPP1\n",
      "ERG28\n",
      "LRRC29\n",
      "MINDY1\n",
      "CNOT9\n",
      "TENM4\n",
      "NEMP1\n",
      "PTPA\n",
      "CFAP299\n",
      "ATP5MGL\n",
      "LARGE2\n",
      "RETREG3\n",
      "ELP1\n",
      "KYAT3\n",
      "ATP5MC3\n",
      "RTL9\n",
      "LARGE1\n",
      "KNL1\n",
      "PTPN20\n",
      "CCDC9B\n",
      "MTERF2\n",
      "CFAP298\n",
      "YAE1\n",
      "MTERF2\n",
      "PINX1\n",
      "MARF1\n",
      "VPS35L\n",
      "SELENOW\n",
      "CATSPERE\n",
      "GALNT17\n",
      "CAVIN4\n",
      "TMEM268\n",
      "LMNTD1\n",
      "TENT5D\n",
      "PARPBP\n",
      "ATP5MC1\n",
      "BUD23\n",
      "OTOL1\n",
      "JPT1\n",
      "SHLD2\n",
      "UQCC3\n",
      "CBLL2\n",
      "CARMIL1\n",
      "SHLD1\n",
      "TFAP2D\n",
      "CEP83\n",
      "OTULINL\n",
      "ATP5MD\n",
      "CENPX\n",
      "ATP5MC2\n",
      "CALHM6\n",
      "REXO5\n",
      "FAM241B\n",
      "JPT2\n",
      "EMC6\n",
      "OTULINL\n",
      "PHETA2\n",
      "VAMP7 Chromosome is different chrX != Ref = chrY\n",
      "AKAIN1\n",
      "HPF1\n",
      "FAM241A\n",
      "MINDY4\n",
      "MTREX\n",
      "SAXO2\n",
      "PIMREG\n",
      "YJU2\n",
      "DEFB131A\n",
      "ATP5F1E\n",
      "CLTRN\n",
      "PLPBP\n",
      "VXN\n",
      "PAK5\n",
      "CCDC198\n",
      "SELENOI\n",
      "MACO1\n",
      "PCARE\n",
      "STMP1\n",
      "AFG1L\n",
      "PRUNE1\n",
      "C15orf53\n",
      "TUT4\n",
      "CDHR2\n",
      "NIPSNAP2\n",
      "IL3RA Chromosome is different chrX != Ref = chrY\n",
      "SHLD2\n",
      "GSDME\n",
      "TUT4\n",
      "DEUP1\n",
      "TMEM35A\n"
     ]
    }
   ],
   "source": [
    "def check_exon_num(gene,chrom,location):\n",
    "    result = 0\n",
    "    if gene not in gene2longest:\n",
    "        print (gene)\n",
    "        return 0\n",
    "    transcript = gene2longest[gene]\n",
    "    if chrom != transcript2exon[transcript][1][0]:\n",
    "        print (gene, \"Chromosome is different %s != Ref = %s\"%(chrom,transcript2exon[transcript][1][0]))\n",
    "        return 0\n",
    "    \n",
    "    for exon_num in transcript2exon[transcript]:\n",
    "        if transcript2exon[transcript][exon_num][1] <=location and location < transcript2exon[transcript][exon_num][2]:\n",
    "            result = exon_num\n",
    "    '''if result == 0:\n",
    "        print gene, location'''\n",
    "    return result\n",
    "\n",
    "#Generate library map for CRISPRcleanR format\n",
    "guidegenemap_filtered = guidegenemap.loc[intersect1d(list(targetsgrna.keys()),readcount.index)]\n",
    "guidegenemap_CRISPRcleanR = pd.DataFrame(index = intersect1d(list(targetsgrna.keys()),readcount.index),columns=['CODE','GENES','EXONE','CHRM','STRAND','STARTpos','ENDpos'])\n",
    "\n",
    "exones = dict()\n",
    "chroms = dict()\n",
    "strands=dict()\n",
    "starts = dict()\n",
    "ends=dict()\n",
    "for i in range(len(guidegenemap_filtered.index)):\n",
    "    genome_align = guidegenemap_filtered.iloc[i]['genome_alignment']\n",
    "    sgrna = guidegenemap_filtered.index[i]\n",
    "    temp = genome_align.split(\"_\")\n",
    "    chrom = temp[0].replace('chr','')\n",
    "    pos = temp[1]\n",
    "    strand = temp[2]\n",
    "    if strand =='-':\n",
    "        posstart = int(pos)\n",
    "        posend = int(pos) + 19\n",
    "    if strand =='+':\n",
    "        posstart = int(pos) - 19\n",
    "        posend = int(pos)\n",
    "    exones[sgrna] = 'ex' + str(check_exon_num(targetsgrna[sgrna],temp[0],int(pos)))\n",
    "    chroms[sgrna] = chrom\n",
    "    strands[sgrna] = strand\n",
    "    starts[sgrna] = posstart\n",
    "    ends[sgrna] = posend\n",
    "    #guidegenemap_CRISPRcleanR.loc[sgrna] = [targetsgrna[sgrna],targetsgrna[sgrna],'-',chrom,strand,posstart,posend]\n",
    "\n",
    "    \n",
    "guidegenemap_CRISPRcleanR['CODE'] = pd.Series(targetsgrna)\n",
    "guidegenemap_CRISPRcleanR['GENES'] = pd.Series(targetsgrna)\n",
    "guidegenemap_CRISPRcleanR['EXONE'] = pd.Series(exones)\n",
    "guidegenemap_CRISPRcleanR['CHRM'] = pd.Series(chroms)\n",
    "guidegenemap_CRISPRcleanR['STRAND'] = pd.Series(strands)\n",
    "guidegenemap_CRISPRcleanR['STARTpos'] = pd.Series(starts)\n",
    "guidegenemap_CRISPRcleanR['ENDpos'] = pd.Series(starts)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "44967"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene2longest)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "cdstable = dict()\n",
    "for gene in gene2longest:\n",
    "    cdstable[gene] = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "guidegenemap_CRISPRcleanR.to_csv('Avana_library_filtered_forCRISPRcleanR',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "70994 70994\n"
     ]
    }
   ],
   "source": [
    "print (len(intersect1d(list(targetsgrna.keys()),readcount.index)),len(targetsgrna.keys()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "### unmodified raw\n",
    "\n",
    "raw_readcount = pd.read_csv('Achilles_raw_readcounts.csv',index_col=0,header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>143B-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>2313287-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>2313287-311Cas9_RepB_p5_batch3</th>\n",
       "      <th>253J-311Cas9_RepA_p5_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>42-MG-BA-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepA_p6_batch3</th>\n",
       "      <th>5637-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>59M-311Cas9_RepA_p4_batch3</th>\n",
       "      <th>59M-311Cas9_RepB_p4_batch3</th>\n",
       "      <th>...</th>\n",
       "      <th>YKG1-311Cas9_RepB_p6_batch3</th>\n",
       "      <th>ZR-75-1-311Cas9_RepA_p5_batch2</th>\n",
       "      <th>ZR-75-1-311Cas9_RepB_p5_batch2</th>\n",
       "      <th>pDNA Avana4_010115_1.5Ex_batch0</th>\n",
       "      <th>pDNA Avana4_010115_1.5Ex_batch1</th>\n",
       "      <th>pDNA Avana4_0101215_0.55Ex_batch1</th>\n",
       "      <th>pDNA Avana4_060115_0.55Ex_batch1</th>\n",
       "      <th>pDNA Avana4_060115_1.5Ex_batch0</th>\n",
       "      <th>pDNA Avana4_060115_1.5Ex_batch1</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>833.0</td>\n",
       "      <td>327.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>546.0</td>\n",
       "      <td>213.0</td>\n",
       "      <td>349.0</td>\n",
       "      <td>280.0</td>\n",
       "      <td>489.0</td>\n",
       "      <td>683.0</td>\n",
       "      <td>373.0</td>\n",
       "      <td>...</td>\n",
       "      <td>556.0</td>\n",
       "      <td>226.0</td>\n",
       "      <td>386.0</td>\n",
       "      <td>305.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>190.0</td>\n",
       "      <td>176.0</td>\n",
       "      <td>324.0</td>\n",
       "      <td>289.0</td>\n",
       "      <td>SHOC2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>1089.0</td>\n",
       "      <td>383.0</td>\n",
       "      <td>391.0</td>\n",
       "      <td>1151.0</td>\n",
       "      <td>736.0</td>\n",
       "      <td>727.0</td>\n",
       "      <td>565.0</td>\n",
       "      <td>770.0</td>\n",
       "      <td>1279.0</td>\n",
       "      <td>720.0</td>\n",
       "      <td>...</td>\n",
       "      <td>773.0</td>\n",
       "      <td>632.0</td>\n",
       "      <td>673.0</td>\n",
       "      <td>398.0</td>\n",
       "      <td>396.0</td>\n",
       "      <td>254.0</td>\n",
       "      <td>230.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>445.0</td>\n",
       "      <td>NDUFA12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>95.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>60.0</td>\n",
       "      <td>99.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>76.0</td>\n",
       "      <td>...</td>\n",
       "      <td>57.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>69.0</td>\n",
       "      <td>SDAD1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1499 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      143B-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        833.0   \n",
       "AAAAAACCCGTAGATAGCCT                       1089.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         95.0   \n",
       "\n",
       "                      2313287-311Cas9_RepA_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                           327.0   \n",
       "AAAAAACCCGTAGATAGCCT                           383.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            32.0   \n",
       "\n",
       "                      2313287-311Cas9_RepB_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                           188.0   \n",
       "AAAAAACCCGTAGATAGCCT                           391.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            69.0   \n",
       "\n",
       "                      253J-311Cas9_RepA_p5_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        546.0   \n",
       "AAAAAACCCGTAGATAGCCT                       1151.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         72.0   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                            213.0   \n",
       "AAAAAACCCGTAGATAGCCT                            736.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             46.0   \n",
       "\n",
       "                      42-MG-BA-311Cas9_RepB_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                            349.0   \n",
       "AAAAAACCCGTAGATAGCCT                            727.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             60.0   \n",
       "\n",
       "                      5637-311Cas9_RepA_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        280.0   \n",
       "AAAAAACCCGTAGATAGCCT                        565.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         99.0   \n",
       "\n",
       "                      5637-311Cas9_RepB_p6_batch3  59M-311Cas9_RepA_p4_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        489.0                       683.0   \n",
       "AAAAAACCCGTAGATAGCCT                        770.0                      1279.0   \n",
       "AAAAAAGAAGAAAAAACCAG                        102.0                        18.0   \n",
       "\n",
       "                      59M-311Cas9_RepB_p4_batch3  ...  \\\n",
       "AAAAAAATCCAGCAATGCAG                       373.0  ...   \n",
       "AAAAAACCCGTAGATAGCCT                       720.0  ...   \n",
       "AAAAAAGAAGAAAAAACCAG                        76.0  ...   \n",
       "\n",
       "                      YKG1-311Cas9_RepB_p6_batch3  \\\n",
       "AAAAAAATCCAGCAATGCAG                        556.0   \n",
       "AAAAAACCCGTAGATAGCCT                        773.0   \n",
       "AAAAAAGAAGAAAAAACCAG                         57.0   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepA_p5_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                           226.0   \n",
       "AAAAAACCCGTAGATAGCCT                           632.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            44.0   \n",
       "\n",
       "                      ZR-75-1-311Cas9_RepB_p5_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                           386.0   \n",
       "AAAAAACCCGTAGATAGCCT                           673.0   \n",
       "AAAAAAGAAGAAAAAACCAG                            39.0   \n",
       "\n",
       "                      pDNA Avana4_010115_1.5Ex_batch0  \\\n",
       "AAAAAAATCCAGCAATGCAG                            305.0   \n",
       "AAAAAACCCGTAGATAGCCT                            398.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             93.0   \n",
       "\n",
       "                      pDNA Avana4_010115_1.5Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                            300.0   \n",
       "AAAAAACCCGTAGATAGCCT                            396.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             75.0   \n",
       "\n",
       "                      pDNA Avana4_0101215_0.55Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                              190.0   \n",
       "AAAAAACCCGTAGATAGCCT                              254.0   \n",
       "AAAAAAGAAGAAAAAACCAG                               42.0   \n",
       "\n",
       "                      pDNA Avana4_060115_0.55Ex_batch1  \\\n",
       "AAAAAAATCCAGCAATGCAG                             176.0   \n",
       "AAAAAACCCGTAGATAGCCT                             230.0   \n",
       "AAAAAAGAAGAAAAAACCAG                              31.0   \n",
       "\n",
       "                      pDNA Avana4_060115_1.5Ex_batch0  \\\n",
       "AAAAAAATCCAGCAATGCAG                            324.0   \n",
       "AAAAAACCCGTAGATAGCCT                            421.0   \n",
       "AAAAAAGAAGAAAAAACCAG                             74.0   \n",
       "\n",
       "                      pDNA Avana4_060115_1.5Ex_batch1     GENE  \n",
       "AAAAAAATCCAGCAATGCAG                            289.0    SHOC2  \n",
       "AAAAAACCCGTAGATAGCCT                            445.0  NDUFA12  \n",
       "AAAAAAGAAGAAAAAACCAG                             69.0    SDAD1  \n",
       "\n",
       "[3 rows x 1499 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_readcount['GENE'] = pd.Series(sgrna2gene)\n",
    "raw_readcount.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>SW403-311Cas9_RepB_p5_batch2</th>\n",
       "      <th>Avana4pDNA20160601-311cas9 RepG09_batch2</th>\n",
       "      <th>Avana4pDNA20160601-311cas9 RepG10_batch2</th>\n",
       "      <th>Avana4pDNA20160601-311cas9 RepG11_batch2</th>\n",
       "      <th>Avana4pDNA20160601-311cas9 RepG12_batch2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>AAAAAAATCCAGCAATGCAG</td>\n",
       "      <td>SHOC2</td>\n",
       "      <td>431.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>182.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAACCCGTAGATAGCCT</td>\n",
       "      <td>NDUFA12</td>\n",
       "      <td>697.0</td>\n",
       "      <td>262.0</td>\n",
       "      <td>303.0</td>\n",
       "      <td>307.0</td>\n",
       "      <td>275.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGAAGAAAAAACCAG</td>\n",
       "      <td>SDAD1</td>\n",
       "      <td>129.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>37.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGCTCAAGAAGGAGG</td>\n",
       "      <td>FAM98A</td>\n",
       "      <td>171.0</td>\n",
       "      <td>75.0</td>\n",
       "      <td>82.0</td>\n",
       "      <td>84.0</td>\n",
       "      <td>83.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>AAAAAAGGCTGTAAAAGCGT</td>\n",
       "      <td>ZNF253</td>\n",
       "      <td>207.0</td>\n",
       "      <td>101.0</td>\n",
       "      <td>136.0</td>\n",
       "      <td>126.0</td>\n",
       "      <td>109.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTGTTTGTCTCATTCATGG</td>\n",
       "      <td>C1orf210</td>\n",
       "      <td>1366.0</td>\n",
       "      <td>361.0</td>\n",
       "      <td>407.0</td>\n",
       "      <td>440.0</td>\n",
       "      <td>346.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTTACCTTGTTCACATGGA</td>\n",
       "      <td>NA</td>\n",
       "      <td>964.0</td>\n",
       "      <td>260.0</td>\n",
       "      <td>304.0</td>\n",
       "      <td>291.0</td>\n",
       "      <td>266.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTTGACTCTAATCACCGGT</td>\n",
       "      <td>NA</td>\n",
       "      <td>564.0</td>\n",
       "      <td>187.0</td>\n",
       "      <td>227.0</td>\n",
       "      <td>209.0</td>\n",
       "      <td>208.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTTTAATACAAGGTAATCT</td>\n",
       "      <td>NA</td>\n",
       "      <td>1442.0</td>\n",
       "      <td>332.0</td>\n",
       "      <td>381.0</td>\n",
       "      <td>364.0</td>\n",
       "      <td>295.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TTTTTCTCACCCGATGAATC</td>\n",
       "      <td>NA</td>\n",
       "      <td>1668.0</td>\n",
       "      <td>440.0</td>\n",
       "      <td>483.0</td>\n",
       "      <td>464.0</td>\n",
       "      <td>412.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74687 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          GENE  SW403-311Cas9_RepB_p5_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG     SHOC2                         431.0   \n",
       "AAAAAACCCGTAGATAGCCT   NDUFA12                         697.0   \n",
       "AAAAAAGAAGAAAAAACCAG     SDAD1                         129.0   \n",
       "AAAAAAGCTCAAGAAGGAGG    FAM98A                         171.0   \n",
       "AAAAAAGGCTGTAAAAGCGT    ZNF253                         207.0   \n",
       "...                        ...                           ...   \n",
       "TTTGTTTGTCTCATTCATGG  C1orf210                        1366.0   \n",
       "TTTTACCTTGTTCACATGGA        NA                         964.0   \n",
       "TTTTGACTCTAATCACCGGT        NA                         564.0   \n",
       "TTTTTAATACAAGGTAATCT        NA                        1442.0   \n",
       "TTTTTCTCACCCGATGAATC        NA                        1668.0   \n",
       "\n",
       "                      Avana4pDNA20160601-311cas9 RepG09_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                                     187.0   \n",
       "AAAAAACCCGTAGATAGCCT                                     262.0   \n",
       "AAAAAAGAAGAAAAAACCAG                                      46.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                                      75.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                                     101.0   \n",
       "...                                                        ...   \n",
       "TTTGTTTGTCTCATTCATGG                                     361.0   \n",
       "TTTTACCTTGTTCACATGGA                                     260.0   \n",
       "TTTTGACTCTAATCACCGGT                                     187.0   \n",
       "TTTTTAATACAAGGTAATCT                                     332.0   \n",
       "TTTTTCTCACCCGATGAATC                                     440.0   \n",
       "\n",
       "                      Avana4pDNA20160601-311cas9 RepG10_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                                     205.0   \n",
       "AAAAAACCCGTAGATAGCCT                                     303.0   \n",
       "AAAAAAGAAGAAAAAACCAG                                      29.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                                      82.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                                     136.0   \n",
       "...                                                        ...   \n",
       "TTTGTTTGTCTCATTCATGG                                     407.0   \n",
       "TTTTACCTTGTTCACATGGA                                     304.0   \n",
       "TTTTGACTCTAATCACCGGT                                     227.0   \n",
       "TTTTTAATACAAGGTAATCT                                     381.0   \n",
       "TTTTTCTCACCCGATGAATC                                     483.0   \n",
       "\n",
       "                      Avana4pDNA20160601-311cas9 RepG11_batch2  \\\n",
       "AAAAAAATCCAGCAATGCAG                                     194.0   \n",
       "AAAAAACCCGTAGATAGCCT                                     307.0   \n",
       "AAAAAAGAAGAAAAAACCAG                                      46.0   \n",
       "AAAAAAGCTCAAGAAGGAGG                                      84.0   \n",
       "AAAAAAGGCTGTAAAAGCGT                                     126.0   \n",
       "...                                                        ...   \n",
       "TTTGTTTGTCTCATTCATGG                                     440.0   \n",
       "TTTTACCTTGTTCACATGGA                                     291.0   \n",
       "TTTTGACTCTAATCACCGGT                                     209.0   \n",
       "TTTTTAATACAAGGTAATCT                                     364.0   \n",
       "TTTTTCTCACCCGATGAATC                                     464.0   \n",
       "\n",
       "                      Avana4pDNA20160601-311cas9 RepG12_batch2  \n",
       "AAAAAAATCCAGCAATGCAG                                     182.0  \n",
       "AAAAAACCCGTAGATAGCCT                                     275.0  \n",
       "AAAAAAGAAGAAAAAACCAG                                      37.0  \n",
       "AAAAAAGCTCAAGAAGGAGG                                      83.0  \n",
       "AAAAAAGGCTGTAAAAGCGT                                     109.0  \n",
       "...                                                        ...  \n",
       "TTTGTTTGTCTCATTCATGG                                     346.0  \n",
       "TTTTACCTTGTTCACATGGA                                     266.0  \n",
       "TTTTGACTCTAATCACCGGT                                     208.0  \n",
       "TTTTTAATACAAGGTAATCT                                     295.0  \n",
       "TTTTTCTCACCCGATGAATC                                     412.0  \n",
       "\n",
       "[74687 rows x 6 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_readcount[['GENE'] + samples[2]['ACH-000820'] + controls[2]].fillna('NA')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "# modfy gene expression data\n",
    "\n",
    "expdata = pd.read_csv('CCLE_expression.csv',index_col=0,header=0).T\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "ncbi_id2symbol = pd.read_csv(\"Homo_sapiens.gene_info.id2symbol\",index_col=0,header=0,sep=\"\\t\")['Symbol']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-001097</th>\n",
       "      <th>ACH-001804</th>\n",
       "      <th>ACH-000534</th>\n",
       "      <th>ACH-000742</th>\n",
       "      <th>ACH-000836</th>\n",
       "      <th>ACH-000545</th>\n",
       "      <th>ACH-001850</th>\n",
       "      <th>ACH-000470</th>\n",
       "      <th>ACH-001959</th>\n",
       "      <th>ACH-000079</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000305</th>\n",
       "      <th>ACH-000603</th>\n",
       "      <th>ACH-000916</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-000978</th>\n",
       "      <th>ACH-000904</th>\n",
       "      <th>ACH-000110</th>\n",
       "      <th>ACH-000261</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000682</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TSPAN6 (7105)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.934988</td>\n",
       "      <td>0.839960</td>\n",
       "      <td>3.722466</td>\n",
       "      <td>4.632268</td>\n",
       "      <td>4.251719</td>\n",
       "      <td>4.844988</td>\n",
       "      <td>3.681449</td>\n",
       "      <td>3.321928</td>\n",
       "      <td>3.788686</td>\n",
       "      <td>...</td>\n",
       "      <td>4.152183</td>\n",
       "      <td>0.910733</td>\n",
       "      <td>4.997744</td>\n",
       "      <td>5.382321</td>\n",
       "      <td>5.002252</td>\n",
       "      <td>4.316870</td>\n",
       "      <td>5.227279</td>\n",
       "      <td>4.447579</td>\n",
       "      <td>4.714795</td>\n",
       "      <td>5.976364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TNMD (64102)</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150560</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.124328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DPM1 (8813)</td>\n",
       "      <td>4.667324</td>\n",
       "      <td>6.635464</td>\n",
       "      <td>5.376082</td>\n",
       "      <td>6.029674</td>\n",
       "      <td>6.704180</td>\n",
       "      <td>5.651052</td>\n",
       "      <td>8.068993</td>\n",
       "      <td>7.294896</td>\n",
       "      <td>7.357288</td>\n",
       "      <td>6.626293</td>\n",
       "      <td>...</td>\n",
       "      <td>6.024364</td>\n",
       "      <td>5.740118</td>\n",
       "      <td>5.898450</td>\n",
       "      <td>6.551670</td>\n",
       "      <td>6.523719</td>\n",
       "      <td>5.990955</td>\n",
       "      <td>6.764341</td>\n",
       "      <td>6.748864</td>\n",
       "      <td>6.745910</td>\n",
       "      <td>5.983450</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1249 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ACH-001097  ACH-001804  ACH-000534  ACH-000742  ACH-000836  \\\n",
       "TSPAN6 (7105)    0.000000    4.934988    0.839960    3.722466    4.632268   \n",
       "TNMD (64102)     0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "DPM1 (8813)      4.667324    6.635464    5.376082    6.029674    6.704180   \n",
       "\n",
       "               ACH-000545  ACH-001850  ACH-000470  ACH-001959  ACH-000079  \\\n",
       "TSPAN6 (7105)    4.251719    4.844988    3.681449    3.321928    3.788686   \n",
       "TNMD (64102)     0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "DPM1 (8813)      5.651052    8.068993    7.294896    7.357288    6.626293   \n",
       "\n",
       "               ...  ACH-000305  ACH-000603  ACH-000916  ACH-000296  \\\n",
       "TSPAN6 (7105)  ...    4.152183    0.910733    4.997744    5.382321   \n",
       "TNMD (64102)   ...    0.000000    0.000000    0.000000    0.150560   \n",
       "DPM1 (8813)    ...    6.024364    5.740118    5.898450    6.551670   \n",
       "\n",
       "               ACH-000978  ACH-000904  ACH-000110  ACH-000261  ACH-000031  \\\n",
       "TSPAN6 (7105)    5.002252    4.316870    5.227279    4.447579    4.714795   \n",
       "TNMD (64102)     0.042644    0.028569    0.070389    0.028569    0.028569   \n",
       "DPM1 (8813)      6.523719    5.990955    6.764341    6.748864    6.745910   \n",
       "\n",
       "               ACH-000682  \n",
       "TSPAN6 (7105)    5.976364  \n",
       "TNMD (64102)     0.124328  \n",
       "DPM1 (8813)      5.983450  \n",
       "\n",
       "[3 rows x 1249 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expdata.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp_xref= dict()\n",
    "for g in expdata.index:\n",
    "    entrez = g.split(\"(\")[1].replace(\")\",\"\")\n",
    "    exp_xref[g] = int(entrez)\n",
    "expdata = expdata.rename(index=exp_xref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-001097</th>\n",
       "      <th>ACH-001804</th>\n",
       "      <th>ACH-000534</th>\n",
       "      <th>ACH-000742</th>\n",
       "      <th>ACH-000836</th>\n",
       "      <th>ACH-000545</th>\n",
       "      <th>ACH-001850</th>\n",
       "      <th>ACH-000470</th>\n",
       "      <th>ACH-001959</th>\n",
       "      <th>ACH-000079</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000305</th>\n",
       "      <th>ACH-000603</th>\n",
       "      <th>ACH-000916</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-000978</th>\n",
       "      <th>ACH-000904</th>\n",
       "      <th>ACH-000110</th>\n",
       "      <th>ACH-000261</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000682</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>7105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.934988</td>\n",
       "      <td>0.839960</td>\n",
       "      <td>3.722466</td>\n",
       "      <td>4.632268</td>\n",
       "      <td>4.251719</td>\n",
       "      <td>4.844988</td>\n",
       "      <td>3.681449</td>\n",
       "      <td>3.321928</td>\n",
       "      <td>3.788686</td>\n",
       "      <td>...</td>\n",
       "      <td>4.152183</td>\n",
       "      <td>0.910733</td>\n",
       "      <td>4.997744</td>\n",
       "      <td>5.382321</td>\n",
       "      <td>5.002252</td>\n",
       "      <td>4.316870</td>\n",
       "      <td>5.227279</td>\n",
       "      <td>4.447579</td>\n",
       "      <td>4.714795</td>\n",
       "      <td>5.976364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64102</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150560</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.124328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8813</td>\n",
       "      <td>4.667324</td>\n",
       "      <td>6.635464</td>\n",
       "      <td>5.376082</td>\n",
       "      <td>6.029674</td>\n",
       "      <td>6.704180</td>\n",
       "      <td>5.651052</td>\n",
       "      <td>8.068993</td>\n",
       "      <td>7.294896</td>\n",
       "      <td>7.357288</td>\n",
       "      <td>6.626293</td>\n",
       "      <td>...</td>\n",
       "      <td>6.024364</td>\n",
       "      <td>5.740118</td>\n",
       "      <td>5.898450</td>\n",
       "      <td>6.551670</td>\n",
       "      <td>6.523719</td>\n",
       "      <td>5.990955</td>\n",
       "      <td>6.764341</td>\n",
       "      <td>6.748864</td>\n",
       "      <td>6.745910</td>\n",
       "      <td>5.983450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57147</td>\n",
       "      <td>1.761285</td>\n",
       "      <td>1.929791</td>\n",
       "      <td>2.687061</td>\n",
       "      <td>2.192194</td>\n",
       "      <td>2.792855</td>\n",
       "      <td>3.440952</td>\n",
       "      <td>1.280956</td>\n",
       "      <td>2.400538</td>\n",
       "      <td>1.691534</td>\n",
       "      <td>2.469886</td>\n",
       "      <td>...</td>\n",
       "      <td>2.361768</td>\n",
       "      <td>2.505891</td>\n",
       "      <td>3.814550</td>\n",
       "      <td>2.592158</td>\n",
       "      <td>2.613532</td>\n",
       "      <td>3.185867</td>\n",
       "      <td>3.537296</td>\n",
       "      <td>2.650765</td>\n",
       "      <td>2.776104</td>\n",
       "      <td>2.939227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55732</td>\n",
       "      <td>3.554589</td>\n",
       "      <td>2.503349</td>\n",
       "      <td>4.440288</td>\n",
       "      <td>2.533563</td>\n",
       "      <td>4.079805</td>\n",
       "      <td>3.275007</td>\n",
       "      <td>2.757023</td>\n",
       "      <td>2.849999</td>\n",
       "      <td>4.328406</td>\n",
       "      <td>3.094236</td>\n",
       "      <td>...</td>\n",
       "      <td>1.659925</td>\n",
       "      <td>3.161888</td>\n",
       "      <td>3.446256</td>\n",
       "      <td>4.352617</td>\n",
       "      <td>3.243364</td>\n",
       "      <td>4.846493</td>\n",
       "      <td>4.491212</td>\n",
       "      <td>4.607626</td>\n",
       "      <td>3.621759</td>\n",
       "      <td>2.918386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110599564</td>\n",
       "      <td>2.980025</td>\n",
       "      <td>2.283922</td>\n",
       "      <td>1.333424</td>\n",
       "      <td>1.526069</td>\n",
       "      <td>2.983678</td>\n",
       "      <td>2.899176</td>\n",
       "      <td>1.521051</td>\n",
       "      <td>3.711495</td>\n",
       "      <td>3.211012</td>\n",
       "      <td>1.485427</td>\n",
       "      <td>...</td>\n",
       "      <td>4.432959</td>\n",
       "      <td>3.229588</td>\n",
       "      <td>3.852998</td>\n",
       "      <td>4.080658</td>\n",
       "      <td>3.765535</td>\n",
       "      <td>3.831877</td>\n",
       "      <td>4.644433</td>\n",
       "      <td>2.615887</td>\n",
       "      <td>4.476382</td>\n",
       "      <td>2.871844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6905</td>\n",
       "      <td>5.132988</td>\n",
       "      <td>4.288359</td>\n",
       "      <td>4.841470</td>\n",
       "      <td>4.494416</td>\n",
       "      <td>4.743623</td>\n",
       "      <td>5.453847</td>\n",
       "      <td>3.477677</td>\n",
       "      <td>4.500802</td>\n",
       "      <td>5.351911</td>\n",
       "      <td>5.096768</td>\n",
       "      <td>...</td>\n",
       "      <td>4.421560</td>\n",
       "      <td>4.266787</td>\n",
       "      <td>5.120186</td>\n",
       "      <td>5.814038</td>\n",
       "      <td>5.315059</td>\n",
       "      <td>5.882888</td>\n",
       "      <td>4.605850</td>\n",
       "      <td>4.613532</td>\n",
       "      <td>4.425594</td>\n",
       "      <td>3.783457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112577462</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.056584</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.422233</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89839</td>\n",
       "      <td>0.443607</td>\n",
       "      <td>0.475085</td>\n",
       "      <td>3.904966</td>\n",
       "      <td>0.941106</td>\n",
       "      <td>2.650765</td>\n",
       "      <td>3.950468</td>\n",
       "      <td>1.959770</td>\n",
       "      <td>1.157044</td>\n",
       "      <td>1.372952</td>\n",
       "      <td>0.367371</td>\n",
       "      <td>...</td>\n",
       "      <td>1.350497</td>\n",
       "      <td>1.176323</td>\n",
       "      <td>2.720278</td>\n",
       "      <td>1.137504</td>\n",
       "      <td>2.384050</td>\n",
       "      <td>3.063503</td>\n",
       "      <td>1.906891</td>\n",
       "      <td>2.411426</td>\n",
       "      <td>2.014355</td>\n",
       "      <td>1.526069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112577461</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.765535</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084064</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189034</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.084064</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19144 rows × 1249 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ACH-001097  ACH-001804  ACH-000534  ACH-000742  ACH-000836  \\\n",
       "7105         0.000000    4.934988    0.839960    3.722466    4.632268   \n",
       "64102        0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "8813         4.667324    6.635464    5.376082    6.029674    6.704180   \n",
       "57147        1.761285    1.929791    2.687061    2.192194    2.792855   \n",
       "55732        3.554589    2.503349    4.440288    2.533563    4.079805   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "110599564    2.980025    2.283922    1.333424    1.526069    2.983678   \n",
       "6905         5.132988    4.288359    4.841470    4.494416    4.743623   \n",
       "112577462    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "89839        0.443607    0.475085    3.904966    0.941106    2.650765   \n",
       "112577461    0.000000    0.000000    0.000000    0.000000    0.765535   \n",
       "\n",
       "           ACH-000545  ACH-001850  ACH-000470  ACH-001959  ACH-000079  ...  \\\n",
       "7105         4.251719    4.844988    3.681449    3.321928    3.788686  ...   \n",
       "64102        0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "8813         5.651052    8.068993    7.294896    7.357288    6.626293  ...   \n",
       "57147        3.440952    1.280956    2.400538    1.691534    2.469886  ...   \n",
       "55732        3.275007    2.757023    2.849999    4.328406    3.094236  ...   \n",
       "...               ...         ...         ...         ...         ...  ...   \n",
       "110599564    2.899176    1.521051    3.711495    3.211012    1.485427  ...   \n",
       "6905         5.453847    3.477677    4.500802    5.351911    5.096768  ...   \n",
       "112577462    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "89839        3.950468    1.959770    1.157044    1.372952    0.367371  ...   \n",
       "112577461    0.000000    0.084064    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "           ACH-000305  ACH-000603  ACH-000916  ACH-000296  ACH-000978  \\\n",
       "7105         4.152183    0.910733    4.997744    5.382321    5.002252   \n",
       "64102        0.000000    0.000000    0.000000    0.150560    0.042644   \n",
       "8813         6.024364    5.740118    5.898450    6.551670    6.523719   \n",
       "57147        2.361768    2.505891    3.814550    2.592158    2.613532   \n",
       "55732        1.659925    3.161888    3.446256    4.352617    3.243364   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "110599564    4.432959    3.229588    3.852998    4.080658    3.765535   \n",
       "6905         4.421560    4.266787    5.120186    5.814038    5.315059   \n",
       "112577462    0.000000    0.056584    0.000000    0.000000    0.111031   \n",
       "89839        1.350497    1.176323    2.720278    1.137504    2.384050   \n",
       "112577461    0.000000    0.000000    0.000000    0.189034    0.042644   \n",
       "\n",
       "           ACH-000904  ACH-000110  ACH-000261  ACH-000031  ACH-000682  \n",
       "7105         4.316870    5.227279    4.447579    4.714795    5.976364  \n",
       "64102        0.028569    0.070389    0.028569    0.028569    0.124328  \n",
       "8813         5.990955    6.764341    6.748864    6.745910    5.983450  \n",
       "57147        3.185867    3.537296    2.650765    2.776104    2.939227  \n",
       "55732        4.846493    4.491212    4.607626    3.621759    2.918386  \n",
       "...               ...         ...         ...         ...         ...  \n",
       "110599564    3.831877    4.644433    2.615887    4.476382    2.871844  \n",
       "6905         5.882888    4.605850    4.613532    4.425594    3.783457  \n",
       "112577462    0.000000    0.000000    0.000000    0.422233    0.000000  \n",
       "89839        3.063503    1.906891    2.411426    2.014355    1.526069  \n",
       "112577461    0.000000    0.000000    0.042644    0.084064    0.000000  \n",
       "\n",
       "[19144 rows x 1249 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expdata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "expdata_filtering = expdata.rename(index=ncbi_id2symbol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([ 10638,  10896,  11217, 114299, 284083, 387638])"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "setdiff1d(expdata.index,list(ncbi_id2symbol.index))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-001097</th>\n",
       "      <th>ACH-001804</th>\n",
       "      <th>ACH-000534</th>\n",
       "      <th>ACH-000742</th>\n",
       "      <th>ACH-000836</th>\n",
       "      <th>ACH-000545</th>\n",
       "      <th>ACH-001850</th>\n",
       "      <th>ACH-000470</th>\n",
       "      <th>ACH-001959</th>\n",
       "      <th>ACH-000079</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000305</th>\n",
       "      <th>ACH-000603</th>\n",
       "      <th>ACH-000916</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-000978</th>\n",
       "      <th>ACH-000904</th>\n",
       "      <th>ACH-000110</th>\n",
       "      <th>ACH-000261</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000682</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.934988</td>\n",
       "      <td>0.839960</td>\n",
       "      <td>3.722466</td>\n",
       "      <td>4.632268</td>\n",
       "      <td>4.251719</td>\n",
       "      <td>4.844988</td>\n",
       "      <td>3.681449</td>\n",
       "      <td>3.321928</td>\n",
       "      <td>3.788686</td>\n",
       "      <td>...</td>\n",
       "      <td>4.152183</td>\n",
       "      <td>0.910733</td>\n",
       "      <td>4.997744</td>\n",
       "      <td>5.382321</td>\n",
       "      <td>5.002252</td>\n",
       "      <td>4.316870</td>\n",
       "      <td>5.227279</td>\n",
       "      <td>4.447579</td>\n",
       "      <td>4.714795</td>\n",
       "      <td>5.976364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TNMD</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150560</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.124328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>DPM1</td>\n",
       "      <td>4.667324</td>\n",
       "      <td>6.635464</td>\n",
       "      <td>5.376082</td>\n",
       "      <td>6.029674</td>\n",
       "      <td>6.704180</td>\n",
       "      <td>5.651052</td>\n",
       "      <td>8.068993</td>\n",
       "      <td>7.294896</td>\n",
       "      <td>7.357288</td>\n",
       "      <td>6.626293</td>\n",
       "      <td>...</td>\n",
       "      <td>6.024364</td>\n",
       "      <td>5.740118</td>\n",
       "      <td>5.898450</td>\n",
       "      <td>6.551670</td>\n",
       "      <td>6.523719</td>\n",
       "      <td>5.990955</td>\n",
       "      <td>6.764341</td>\n",
       "      <td>6.748864</td>\n",
       "      <td>6.745910</td>\n",
       "      <td>5.983450</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1249 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ACH-001097  ACH-001804  ACH-000534  ACH-000742  ACH-000836  \\\n",
       "TSPAN6    0.000000    4.934988    0.839960    3.722466    4.632268   \n",
       "TNMD      0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "DPM1      4.667324    6.635464    5.376082    6.029674    6.704180   \n",
       "\n",
       "        ACH-000545  ACH-001850  ACH-000470  ACH-001959  ACH-000079  ...  \\\n",
       "TSPAN6    4.251719    4.844988    3.681449    3.321928    3.788686  ...   \n",
       "TNMD      0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "DPM1      5.651052    8.068993    7.294896    7.357288    6.626293  ...   \n",
       "\n",
       "        ACH-000305  ACH-000603  ACH-000916  ACH-000296  ACH-000978  \\\n",
       "TSPAN6    4.152183    0.910733    4.997744    5.382321    5.002252   \n",
       "TNMD      0.000000    0.000000    0.000000    0.150560    0.042644   \n",
       "DPM1      6.024364    5.740118    5.898450    6.551670    6.523719   \n",
       "\n",
       "        ACH-000904  ACH-000110  ACH-000261  ACH-000031  ACH-000682  \n",
       "TSPAN6    4.316870    5.227279    4.447579    4.714795    5.976364  \n",
       "TNMD      0.028569    0.070389    0.028569    0.028569    0.124328  \n",
       "DPM1      5.990955    6.764341    6.748864    6.745910    5.983450  \n",
       "\n",
       "[3 rows x 1249 columns]"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expdata_filtering.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expdata_filtering.index.is_unique"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# removed because id changed to entrez\n",
    "#ensembl2entrez = pd.read_table('../avana_2018Q3/gene2ensembl.human',index_col=2,header=None)[1]"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "'''ensembl2entrez_rmdup = dict()\n",
    "for i in range(len(ensembl2entrez.index)):\n",
    "    ensembl = ensembl2entrez.index[i]\n",
    "    entrez = ensembl2entrez.iloc[i]\n",
    "    ensembl2entrez_rmdup[ensembl]=entrez\n",
    "'''    "
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# update genes in exptable\n",
    "'''xref = dict()\n",
    "filtering = list()\n",
    "for ensembl_v in expdata.index:\n",
    "    ensembl = ensembl_v.split(\".\")[0]  # remove version   XXXXX.3\n",
    "    if ensembl in ensembl2entrez_rmdup:\n",
    "        entrez = ensembl2entrez_rmdup[ensembl]\n",
    "        if entrez in id2gene_ccds:\n",
    "            filtering.append(ensembl_v)\n",
    "            xref[ensembl_v] = id2gene_ccds[entrez]\n",
    "expdata_filtering = expdata.loc[filtering].rename(index=xref)'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-001097</th>\n",
       "      <th>ACH-001804</th>\n",
       "      <th>ACH-000534</th>\n",
       "      <th>ACH-000742</th>\n",
       "      <th>ACH-000836</th>\n",
       "      <th>ACH-000545</th>\n",
       "      <th>ACH-001850</th>\n",
       "      <th>ACH-000470</th>\n",
       "      <th>ACH-001959</th>\n",
       "      <th>ACH-000079</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-000305</th>\n",
       "      <th>ACH-000603</th>\n",
       "      <th>ACH-000916</th>\n",
       "      <th>ACH-000296</th>\n",
       "      <th>ACH-000978</th>\n",
       "      <th>ACH-000904</th>\n",
       "      <th>ACH-000110</th>\n",
       "      <th>ACH-000261</th>\n",
       "      <th>ACH-000031</th>\n",
       "      <th>ACH-000682</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>7105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.934988</td>\n",
       "      <td>0.839960</td>\n",
       "      <td>3.722466</td>\n",
       "      <td>4.632268</td>\n",
       "      <td>4.251719</td>\n",
       "      <td>4.844988</td>\n",
       "      <td>3.681449</td>\n",
       "      <td>3.321928</td>\n",
       "      <td>3.788686</td>\n",
       "      <td>...</td>\n",
       "      <td>4.152183</td>\n",
       "      <td>0.910733</td>\n",
       "      <td>4.997744</td>\n",
       "      <td>5.382321</td>\n",
       "      <td>5.002252</td>\n",
       "      <td>4.316870</td>\n",
       "      <td>5.227279</td>\n",
       "      <td>4.447579</td>\n",
       "      <td>4.714795</td>\n",
       "      <td>5.976364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64102</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150560</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.070389</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.028569</td>\n",
       "      <td>0.124328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8813</td>\n",
       "      <td>4.667324</td>\n",
       "      <td>6.635464</td>\n",
       "      <td>5.376082</td>\n",
       "      <td>6.029674</td>\n",
       "      <td>6.704180</td>\n",
       "      <td>5.651052</td>\n",
       "      <td>8.068993</td>\n",
       "      <td>7.294896</td>\n",
       "      <td>7.357288</td>\n",
       "      <td>6.626293</td>\n",
       "      <td>...</td>\n",
       "      <td>6.024364</td>\n",
       "      <td>5.740118</td>\n",
       "      <td>5.898450</td>\n",
       "      <td>6.551670</td>\n",
       "      <td>6.523719</td>\n",
       "      <td>5.990955</td>\n",
       "      <td>6.764341</td>\n",
       "      <td>6.748864</td>\n",
       "      <td>6.745910</td>\n",
       "      <td>5.983450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>57147</td>\n",
       "      <td>1.761285</td>\n",
       "      <td>1.929791</td>\n",
       "      <td>2.687061</td>\n",
       "      <td>2.192194</td>\n",
       "      <td>2.792855</td>\n",
       "      <td>3.440952</td>\n",
       "      <td>1.280956</td>\n",
       "      <td>2.400538</td>\n",
       "      <td>1.691534</td>\n",
       "      <td>2.469886</td>\n",
       "      <td>...</td>\n",
       "      <td>2.361768</td>\n",
       "      <td>2.505891</td>\n",
       "      <td>3.814550</td>\n",
       "      <td>2.592158</td>\n",
       "      <td>2.613532</td>\n",
       "      <td>3.185867</td>\n",
       "      <td>3.537296</td>\n",
       "      <td>2.650765</td>\n",
       "      <td>2.776104</td>\n",
       "      <td>2.939227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>55732</td>\n",
       "      <td>3.554589</td>\n",
       "      <td>2.503349</td>\n",
       "      <td>4.440288</td>\n",
       "      <td>2.533563</td>\n",
       "      <td>4.079805</td>\n",
       "      <td>3.275007</td>\n",
       "      <td>2.757023</td>\n",
       "      <td>2.849999</td>\n",
       "      <td>4.328406</td>\n",
       "      <td>3.094236</td>\n",
       "      <td>...</td>\n",
       "      <td>1.659925</td>\n",
       "      <td>3.161888</td>\n",
       "      <td>3.446256</td>\n",
       "      <td>4.352617</td>\n",
       "      <td>3.243364</td>\n",
       "      <td>4.846493</td>\n",
       "      <td>4.491212</td>\n",
       "      <td>4.607626</td>\n",
       "      <td>3.621759</td>\n",
       "      <td>2.918386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>110599564</td>\n",
       "      <td>2.980025</td>\n",
       "      <td>2.283922</td>\n",
       "      <td>1.333424</td>\n",
       "      <td>1.526069</td>\n",
       "      <td>2.983678</td>\n",
       "      <td>2.899176</td>\n",
       "      <td>1.521051</td>\n",
       "      <td>3.711495</td>\n",
       "      <td>3.211012</td>\n",
       "      <td>1.485427</td>\n",
       "      <td>...</td>\n",
       "      <td>4.432959</td>\n",
       "      <td>3.229588</td>\n",
       "      <td>3.852998</td>\n",
       "      <td>4.080658</td>\n",
       "      <td>3.765535</td>\n",
       "      <td>3.831877</td>\n",
       "      <td>4.644433</td>\n",
       "      <td>2.615887</td>\n",
       "      <td>4.476382</td>\n",
       "      <td>2.871844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6905</td>\n",
       "      <td>5.132988</td>\n",
       "      <td>4.288359</td>\n",
       "      <td>4.841470</td>\n",
       "      <td>4.494416</td>\n",
       "      <td>4.743623</td>\n",
       "      <td>5.453847</td>\n",
       "      <td>3.477677</td>\n",
       "      <td>4.500802</td>\n",
       "      <td>5.351911</td>\n",
       "      <td>5.096768</td>\n",
       "      <td>...</td>\n",
       "      <td>4.421560</td>\n",
       "      <td>4.266787</td>\n",
       "      <td>5.120186</td>\n",
       "      <td>5.814038</td>\n",
       "      <td>5.315059</td>\n",
       "      <td>5.882888</td>\n",
       "      <td>4.605850</td>\n",
       "      <td>4.613532</td>\n",
       "      <td>4.425594</td>\n",
       "      <td>3.783457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112577462</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.056584</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.111031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.422233</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>89839</td>\n",
       "      <td>0.443607</td>\n",
       "      <td>0.475085</td>\n",
       "      <td>3.904966</td>\n",
       "      <td>0.941106</td>\n",
       "      <td>2.650765</td>\n",
       "      <td>3.950468</td>\n",
       "      <td>1.959770</td>\n",
       "      <td>1.157044</td>\n",
       "      <td>1.372952</td>\n",
       "      <td>0.367371</td>\n",
       "      <td>...</td>\n",
       "      <td>1.350497</td>\n",
       "      <td>1.176323</td>\n",
       "      <td>2.720278</td>\n",
       "      <td>1.137504</td>\n",
       "      <td>2.384050</td>\n",
       "      <td>3.063503</td>\n",
       "      <td>1.906891</td>\n",
       "      <td>2.411426</td>\n",
       "      <td>2.014355</td>\n",
       "      <td>1.526069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>112577461</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.765535</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084064</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.189034</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.042644</td>\n",
       "      <td>0.084064</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19144 rows × 1249 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ACH-001097  ACH-001804  ACH-000534  ACH-000742  ACH-000836  \\\n",
       "7105         0.000000    4.934988    0.839960    3.722466    4.632268   \n",
       "64102        0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "8813         4.667324    6.635464    5.376082    6.029674    6.704180   \n",
       "57147        1.761285    1.929791    2.687061    2.192194    2.792855   \n",
       "55732        3.554589    2.503349    4.440288    2.533563    4.079805   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "110599564    2.980025    2.283922    1.333424    1.526069    2.983678   \n",
       "6905         5.132988    4.288359    4.841470    4.494416    4.743623   \n",
       "112577462    0.000000    0.000000    0.000000    0.000000    0.000000   \n",
       "89839        0.443607    0.475085    3.904966    0.941106    2.650765   \n",
       "112577461    0.000000    0.000000    0.000000    0.000000    0.765535   \n",
       "\n",
       "           ACH-000545  ACH-001850  ACH-000470  ACH-001959  ACH-000079  ...  \\\n",
       "7105         4.251719    4.844988    3.681449    3.321928    3.788686  ...   \n",
       "64102        0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "8813         5.651052    8.068993    7.294896    7.357288    6.626293  ...   \n",
       "57147        3.440952    1.280956    2.400538    1.691534    2.469886  ...   \n",
       "55732        3.275007    2.757023    2.849999    4.328406    3.094236  ...   \n",
       "...               ...         ...         ...         ...         ...  ...   \n",
       "110599564    2.899176    1.521051    3.711495    3.211012    1.485427  ...   \n",
       "6905         5.453847    3.477677    4.500802    5.351911    5.096768  ...   \n",
       "112577462    0.000000    0.000000    0.000000    0.000000    0.000000  ...   \n",
       "89839        3.950468    1.959770    1.157044    1.372952    0.367371  ...   \n",
       "112577461    0.000000    0.084064    0.000000    0.000000    0.000000  ...   \n",
       "\n",
       "           ACH-000305  ACH-000603  ACH-000916  ACH-000296  ACH-000978  \\\n",
       "7105         4.152183    0.910733    4.997744    5.382321    5.002252   \n",
       "64102        0.000000    0.000000    0.000000    0.150560    0.042644   \n",
       "8813         6.024364    5.740118    5.898450    6.551670    6.523719   \n",
       "57147        2.361768    2.505891    3.814550    2.592158    2.613532   \n",
       "55732        1.659925    3.161888    3.446256    4.352617    3.243364   \n",
       "...               ...         ...         ...         ...         ...   \n",
       "110599564    4.432959    3.229588    3.852998    4.080658    3.765535   \n",
       "6905         4.421560    4.266787    5.120186    5.814038    5.315059   \n",
       "112577462    0.000000    0.056584    0.000000    0.000000    0.111031   \n",
       "89839        1.350497    1.176323    2.720278    1.137504    2.384050   \n",
       "112577461    0.000000    0.000000    0.000000    0.189034    0.042644   \n",
       "\n",
       "           ACH-000904  ACH-000110  ACH-000261  ACH-000031  ACH-000682  \n",
       "7105         4.316870    5.227279    4.447579    4.714795    5.976364  \n",
       "64102        0.028569    0.070389    0.028569    0.028569    0.124328  \n",
       "8813         5.990955    6.764341    6.748864    6.745910    5.983450  \n",
       "57147        3.185867    3.537296    2.650765    2.776104    2.939227  \n",
       "55732        4.846493    4.491212    4.607626    3.621759    2.918386  \n",
       "...               ...         ...         ...         ...         ...  \n",
       "110599564    3.831877    4.644433    2.615887    4.476382    2.871844  \n",
       "6905         5.882888    4.605850    4.613532    4.425594    3.783457  \n",
       "112577462    0.000000    0.000000    0.000000    0.422233    0.000000  \n",
       "89839        3.063503    1.906891    2.411426    2.014355    1.526069  \n",
       "112577461    0.000000    0.000000    0.042644    0.084064    0.000000  \n",
       "\n",
       "[19144 rows x 1249 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expdata_filtering.loc['BAZ2B']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19144\n",
      "19144\n",
      "C11orf65    1\n",
      "FABP9       1\n",
      "RDH14       1\n",
      "FAM102A     1\n",
      "SH3GL1      1\n",
      "ALDOB       1\n",
      "ZNF587B     1\n",
      "TBC1D22A    1\n",
      "R3HDM4      1\n",
      "ABCF3       1\n",
      "ZNF564      1\n",
      "NOS1AP      1\n",
      "OR4D11      1\n",
      "AMMECR1     1\n",
      "LHX4        1\n",
      "MBNL1       1\n",
      "HAND2       1\n",
      "TLR9        1\n",
      "TCERG1      1\n",
      "NSG2        1\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print (len(expdata_filtering.index))\n",
    "print (len(expdata_filtering.index.unique()))\n",
    "print (expdata_filtering.index.value_counts().head(20))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# group for duplicates # disabled due to uniqueness\n",
    "expdata_filtering_sum = expdata_filtering.groupby(expdata_filtering.index).sum()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "print (len(expdata_filtering_sum.index))\n",
    "print (len(expdata_filtering_sum.index.unique()))"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "print (len(intersect1d(expdata_filtering_sum.index,readcount['GENE'].values)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "expdata_filtering.to_csv('CCLE_expression.all.tsv',sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "expdata_filtering[intersect1d(expdata_filtering.dtypes.index,files.index)].to_csv('CCLE_expression.mod.tsv',sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "# modify cndata\n",
    "\n",
    "cndata = pd.read_csv('CCLE_gene_cn.csv',index_col=0,header=0).T\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000003</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001498</th>\n",
       "      <th>ACH-001528</th>\n",
       "      <th>ACH-001617</th>\n",
       "      <th>ACH-001653</th>\n",
       "      <th>ACH-001670</th>\n",
       "      <th>ACH-001819</th>\n",
       "      <th>ACH-002029</th>\n",
       "      <th>ACH-002045</th>\n",
       "      <th>ACH-002059</th>\n",
       "      <th>ACH-002062</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>A1BG (1)</td>\n",
       "      <td>1.179621</td>\n",
       "      <td>1.009801</td>\n",
       "      <td>1.022828</td>\n",
       "      <td>1.232225</td>\n",
       "      <td>1.151880</td>\n",
       "      <td>1.020698</td>\n",
       "      <td>0.980042</td>\n",
       "      <td>1.151756</td>\n",
       "      <td>0.975527</td>\n",
       "      <td>1.071332</td>\n",
       "      <td>...</td>\n",
       "      <td>1.048173</td>\n",
       "      <td>0.609990</td>\n",
       "      <td>1.060877</td>\n",
       "      <td>0.883109</td>\n",
       "      <td>0.789867</td>\n",
       "      <td>1.372514</td>\n",
       "      <td>0.829472</td>\n",
       "      <td>0.867459</td>\n",
       "      <td>1.074234</td>\n",
       "      <td>1.144470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NAT2 (10)</td>\n",
       "      <td>1.098639</td>\n",
       "      <td>0.981249</td>\n",
       "      <td>1.067509</td>\n",
       "      <td>1.061091</td>\n",
       "      <td>1.112505</td>\n",
       "      <td>1.095915</td>\n",
       "      <td>0.979670</td>\n",
       "      <td>0.821120</td>\n",
       "      <td>0.910024</td>\n",
       "      <td>0.844566</td>\n",
       "      <td>...</td>\n",
       "      <td>1.041048</td>\n",
       "      <td>1.026103</td>\n",
       "      <td>1.055120</td>\n",
       "      <td>0.909599</td>\n",
       "      <td>0.793262</td>\n",
       "      <td>1.187271</td>\n",
       "      <td>0.798488</td>\n",
       "      <td>0.831747</td>\n",
       "      <td>1.017901</td>\n",
       "      <td>1.156380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ADA (100)</td>\n",
       "      <td>1.199020</td>\n",
       "      <td>1.012577</td>\n",
       "      <td>1.524481</td>\n",
       "      <td>0.836336</td>\n",
       "      <td>0.816603</td>\n",
       "      <td>0.977753</td>\n",
       "      <td>0.984000</td>\n",
       "      <td>1.372350</td>\n",
       "      <td>1.719741</td>\n",
       "      <td>1.097651</td>\n",
       "      <td>...</td>\n",
       "      <td>1.024727</td>\n",
       "      <td>1.014170</td>\n",
       "      <td>1.039297</td>\n",
       "      <td>1.422202</td>\n",
       "      <td>1.137351</td>\n",
       "      <td>1.466707</td>\n",
       "      <td>1.311691</td>\n",
       "      <td>1.136317</td>\n",
       "      <td>1.008777</td>\n",
       "      <td>1.124883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1682 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           ACH-000001  ACH-000002  ACH-000003  ACH-000004  ACH-000005  \\\n",
       "A1BG (1)     1.179621    1.009801    1.022828    1.232225    1.151880   \n",
       "NAT2 (10)    1.098639    0.981249    1.067509    1.061091    1.112505   \n",
       "ADA (100)    1.199020    1.012577    1.524481    0.836336    0.816603   \n",
       "\n",
       "           ACH-000006  ACH-000007  ACH-000008  ACH-000009  ACH-000011  ...  \\\n",
       "A1BG (1)     1.020698    0.980042    1.151756    0.975527    1.071332  ...   \n",
       "NAT2 (10)    1.095915    0.979670    0.821120    0.910024    0.844566  ...   \n",
       "ADA (100)    0.977753    0.984000    1.372350    1.719741    1.097651  ...   \n",
       "\n",
       "           ACH-001498  ACH-001528  ACH-001617  ACH-001653  ACH-001670  \\\n",
       "A1BG (1)     1.048173    0.609990    1.060877    0.883109    0.789867   \n",
       "NAT2 (10)    1.041048    1.026103    1.055120    0.909599    0.793262   \n",
       "ADA (100)    1.024727    1.014170    1.039297    1.422202    1.137351   \n",
       "\n",
       "           ACH-001819  ACH-002029  ACH-002045  ACH-002059  ACH-002062  \n",
       "A1BG (1)     1.372514    0.829472    0.867459    1.074234    1.144470  \n",
       "NAT2 (10)    1.187271    0.798488    0.831747    1.017901    1.156380  \n",
       "ADA (100)    1.466707    1.311691    1.136317    1.008777    1.124883  \n",
       "\n",
       "[3 rows x 1682 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cndata.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ACH-000001</th>\n",
       "      <th>ACH-000002</th>\n",
       "      <th>ACH-000003</th>\n",
       "      <th>ACH-000004</th>\n",
       "      <th>ACH-000005</th>\n",
       "      <th>ACH-000006</th>\n",
       "      <th>ACH-000007</th>\n",
       "      <th>ACH-000008</th>\n",
       "      <th>ACH-000009</th>\n",
       "      <th>ACH-000011</th>\n",
       "      <th>...</th>\n",
       "      <th>ACH-001498</th>\n",
       "      <th>ACH-001528</th>\n",
       "      <th>ACH-001617</th>\n",
       "      <th>ACH-001653</th>\n",
       "      <th>ACH-001670</th>\n",
       "      <th>ACH-001819</th>\n",
       "      <th>ACH-002029</th>\n",
       "      <th>ACH-002045</th>\n",
       "      <th>ACH-002059</th>\n",
       "      <th>ACH-002062</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>A1BG</td>\n",
       "      <td>1.179621</td>\n",
       "      <td>1.009801</td>\n",
       "      <td>1.022828</td>\n",
       "      <td>1.232225</td>\n",
       "      <td>1.151880</td>\n",
       "      <td>1.020698</td>\n",
       "      <td>0.980042</td>\n",
       "      <td>1.151756</td>\n",
       "      <td>0.975527</td>\n",
       "      <td>1.071332</td>\n",
       "      <td>...</td>\n",
       "      <td>1.048173</td>\n",
       "      <td>0.609990</td>\n",
       "      <td>1.060877</td>\n",
       "      <td>0.883109</td>\n",
       "      <td>0.789867</td>\n",
       "      <td>1.372514</td>\n",
       "      <td>0.829472</td>\n",
       "      <td>0.867459</td>\n",
       "      <td>1.074234</td>\n",
       "      <td>1.144470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>NAT2</td>\n",
       "      <td>1.098639</td>\n",
       "      <td>0.981249</td>\n",
       "      <td>1.067509</td>\n",
       "      <td>1.061091</td>\n",
       "      <td>1.112505</td>\n",
       "      <td>1.095915</td>\n",
       "      <td>0.979670</td>\n",
       "      <td>0.821120</td>\n",
       "      <td>0.910024</td>\n",
       "      <td>0.844566</td>\n",
       "      <td>...</td>\n",
       "      <td>1.041048</td>\n",
       "      <td>1.026103</td>\n",
       "      <td>1.055120</td>\n",
       "      <td>0.909599</td>\n",
       "      <td>0.793262</td>\n",
       "      <td>1.187271</td>\n",
       "      <td>0.798488</td>\n",
       "      <td>0.831747</td>\n",
       "      <td>1.017901</td>\n",
       "      <td>1.156380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>ADA</td>\n",
       "      <td>1.199020</td>\n",
       "      <td>1.012577</td>\n",
       "      <td>1.524481</td>\n",
       "      <td>0.836336</td>\n",
       "      <td>0.816603</td>\n",
       "      <td>0.977753</td>\n",
       "      <td>0.984000</td>\n",
       "      <td>1.372350</td>\n",
       "      <td>1.719741</td>\n",
       "      <td>1.097651</td>\n",
       "      <td>...</td>\n",
       "      <td>1.024727</td>\n",
       "      <td>1.014170</td>\n",
       "      <td>1.039297</td>\n",
       "      <td>1.422202</td>\n",
       "      <td>1.137351</td>\n",
       "      <td>1.466707</td>\n",
       "      <td>1.311691</td>\n",
       "      <td>1.136317</td>\n",
       "      <td>1.008777</td>\n",
       "      <td>1.124883</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 1682 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      ACH-000001  ACH-000002  ACH-000003  ACH-000004  ACH-000005  ACH-000006  \\\n",
       "A1BG    1.179621    1.009801    1.022828    1.232225    1.151880    1.020698   \n",
       "NAT2    1.098639    0.981249    1.067509    1.061091    1.112505    1.095915   \n",
       "ADA     1.199020    1.012577    1.524481    0.836336    0.816603    0.977753   \n",
       "\n",
       "      ACH-000007  ACH-000008  ACH-000009  ACH-000011  ...  ACH-001498  \\\n",
       "A1BG    0.980042    1.151756    0.975527    1.071332  ...    1.048173   \n",
       "NAT2    0.979670    0.821120    0.910024    0.844566  ...    1.041048   \n",
       "ADA     0.984000    1.372350    1.719741    1.097651  ...    1.024727   \n",
       "\n",
       "      ACH-001528  ACH-001617  ACH-001653  ACH-001670  ACH-001819  ACH-002029  \\\n",
       "A1BG    0.609990    1.060877    0.883109    0.789867    1.372514    0.829472   \n",
       "NAT2    1.026103    1.055120    0.909599    0.793262    1.187271    0.798488   \n",
       "ADA     1.014170    1.039297    1.422202    1.137351    1.466707    1.311691   \n",
       "\n",
       "      ACH-002045  ACH-002059  ACH-002062  \n",
       "A1BG    0.867459    1.074234    1.144470  \n",
       "NAT2    0.831747    1.017901    1.156380  \n",
       "ADA     1.136317    1.008777    1.124883  \n",
       "\n",
       "[3 rows x 1682 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cn_xref= dict()\n",
    "for g in cndata.index:\n",
    "    sym = g.split(\" (\")[0]\n",
    "    entrez = int(g.split(\" (\")[1].replace(\")\",\"\"))\n",
    "    '''if sym in old2new:\n",
    "        sym = old2new[sym]'''\n",
    "    cn_xref[g] = entrez\n",
    "cndata = cndata.rename(index=cn_xref)\n",
    "\n",
    "cndata_filtering = cndata.rename(index=ncbi_id2symbol)\n",
    "cndata_filtering.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cndata_filtering.index.is_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "cndata_filtering.to_csv('CCLE_gene_cn.all.tsv',sep=\"\\t\")\n",
    "cndata_filtering[intersect1d(cndata_filtering.dtypes.index,files.index)].to_csv('CCLE_gene_cn.mod.tsv',sep=\"\\t\")\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ekim8/anaconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (4,20,26,27,28,29,30,31,32) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n",
      "/home/ekim8/anaconda3/lib/python3.7/site-packages/numpy/lib/arraysetops.py:568: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  mask |= (ar1 == a)\n"
     ]
    }
   ],
   "source": [
    "# modify mafdata\n",
    "\n",
    "mafdata = pd.read_csv('CCLE_mutations.csv',index_col=0,header=0)\n",
    "\n",
    "'''maf_xref = dict()\n",
    "mafgenes = mafdata['Hugo_Symbol']\n",
    "for i in mafdata.index:\n",
    "    g = mafgenes.loc[i]\n",
    "    if g in old2new:\n",
    "        maf_xref[i] = old2new[g]\n",
    "        #if mafdata.loc[i]['Hugo_Symbol'] != old2new[mafdata.loc[i]['Hugo_Symbol']]:\n",
    "        #    print mafdata.loc[i]['Hugo_Symbol'], old2new[mafdata.loc[i]['Hugo_Symbol']]\n",
    "    else:\n",
    "        maf_xref[i] = g\n",
    "    \n",
    "mafdata['Hugo_Symbol']  = pd.Series(maf_xref)'''\n",
    "\n",
    "\n",
    "temp = dict()\n",
    "for i in mafdata.index:\n",
    "    entrez = mafdata.loc[i,'Entrez_Gene_Id']\n",
    "    if entrez in ncbi_id2symbol:\n",
    "        symbol = ncbi_id2symbol[entrez]\n",
    "    else:\n",
    "        symbol = str(entrez)\n",
    "    temp[i] = symbol\n",
    "mafdata['Entrez_Symbol'] = pd.Series(temp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "mafdata.to_csv('CCLE_mutations.all.tsv',sep=\"\\t\",header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "mafdata[['Hugo_Symbol','Entrez_Symbol','Chromosome','Start_position','End_position','Strand','Variant_Classification','Variant_Type','Protein_Change','isDeleterious','isTCGAhotspot','isCOSMIChotspot','Variant_annotation','DepMap_ID']].to_csv('CCLE_mutations.crop.all.tsv',sep=\"\\t\",header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4.1928547297199997e-06"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cndata['ACH-000437']['ATAD1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bfdata' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-59-d15911872031>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mbfdata\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'ACH-000389'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'bfdata' is not defined"
     ]
    }
   ],
   "source": [
    "bfdata['ACH-000389']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
